<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">
<head>
<meta charset="UTF-8"/>
<title>BoneJ - ImageJ</title>
<script>document.documentElement.className = document.documentElement.className.replace( /(^|\s)client-nojs(\s|$)/, "$1client-js$2" );</script>
<script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":0,"wgPageName":"BoneJ","wgTitle":"BoneJ","wgCurRevisionId":45837,"wgRevisionId":45837,"wgArticleId":12143,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":[],"wgBreakFrames":false,"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRelevantPageName":"BoneJ","wgRelevantArticleId":12143,"wgRequestId":"7e2978ded23f9735e0159b2d","wgIsProbablyEditable":false,"wgRestrictionEdit":[],"wgRestrictionMove":[],"wgPreferredVariant":"en","fancytree_path":"/extensions/TreeAndMenu/fancytree"});mw.loader.state({"site.styles":"ready","noscript":"ready","user.styles":"ready","user.cssprefs":"ready","user":"ready","user.options":"loading","user.tokens":"loading","ext.math.styles":"ready","mediawiki.legacy.shared":"ready","mediawiki.legacy.commonPrint":"ready","mediawiki.sectionAnchor":"ready","skins.erudite":"ready"});mw.loader.implement("user.options@0j3lz3q",function($,jQuery,require,module){mw.user.options.set({"variant":"en"});});mw.loader.implement("user.tokens@1ku9xth",function ( $, jQuery, require, module ) {
mw.user.tokens.set({"editToken":"+\\","patrolToken":"+\\","watchToken":"+\\","csrfToken":"+\\"});/*@nomin*/;

});mw.loader.load(["mediawiki.page.startup"]);});</script>
<link rel="stylesheet" href="load.php%3Fdebug=false&amp;lang=en&amp;modules=ext.math.styles%257Cmediawiki.legacy.commonPrint%252Cshared%257Cmediawiki.sectionAnchor%257Cskins.erudite&amp;only=styles&amp;skin=erudite.css"/>
<script async="" src="load.php%3Fdebug=false&amp;lang=en&amp;modules=startup&amp;only=scripts&amp;skin=erudite"></script>
<link rel="stylesheet" href="extensions/TreeAndMenu/fancytree/fancytree.css"/><link rel="stylesheet" href="extensions/TreeAndMenu/suckerfish/suckerfish.css"/>
<meta name="ResourceLoaderDynamicStyles" content=""/>
<link rel="stylesheet" href="load.php%3Fdebug=false&amp;lang=en&amp;modules=site.styles&amp;only=styles&amp;skin=erudite.css"/>
<meta name="generator" content="MediaWiki 1.28.0"/>
<meta name="description" content="BoneJ is a collection of skeletal biology plug-ins for ImageJ.&#10;This is the new, modernized version of the software available through the ImageJ updater. Its update site is called BoneJ. For the old ImageJ1 version, see BoneJ1."/>
<link rel="shortcut icon" href="skins/ij2.ico"/>
<link rel="search" type="application/opensearchdescription+xml" href="opensearch_desc.php" title="ImageJ (en)"/>
<link rel="EditURI" type="application/rsd+xml" href="api.php?action=rsd"/>
<link rel="alternate" type="application/atom+xml" title="ImageJ Atom feed" href="index.php?title=Special:RecentChanges&amp;feed=atom"/>

<script type="text/javascript" src="extensions/SyntaxHighlighter/syntaxhighlighter/scripts/shCore.js"></script>
<script type="text/javascript" src="extensions/SyntaxHighlighter/syntaxhighlighter/scripts/shBrushJScript.js"></script>
<script type="text/javascript" src="extensions/SyntaxHighlighter/syntaxhighlighter/scripts/shBrushPlain.js"></script>
<script type="text/javascript">
SyntaxHighlighter.all();
</script>
<link rel="stylesheet" type="text/css" media="screen" href="extensions/SyntaxHighlighter/syntaxhighlighter/styles/shCoreMinit.css" />

	<meta property="og:type" content="article"/>

	<meta property="og:site_name" content="ImageJ"/>

	<meta property="og:title" content="BoneJ"/>

	<meta property="og:description" content="BoneJ is a collection of skeletal biology plug-ins for ImageJ.&#10;This is the new, modernized version of the software available through the ImageJ updater. Its update site is called BoneJ. For the old ImageJ1 version, see BoneJ1."/>


<meta name="viewport" content="width=device-width, initial-scale=1" />
</head>
<body class="mediawiki ltr sitedir-ltr mw-hide-empty-elt ns-0 ns-subject page-BoneJ rootpage-BoneJ skin-erudite action-view">
		<div class="mw-jump">
			<a href="BoneJ_experimental.html#bodyContent">Skip to content</a>, 			<a href="BoneJ_experimental.html#search">Skip to search</a>
		</div>

		<div id="top-wrap" role="banner">
			<h1><a href="Welcome" title="ImageJ" rel="home">ImageJ</a></h1>
			<div id="tagline">From ImageJ</div>

			<a id="menubutton" href="BoneJ_experimental.html#menu">Menu</a>
			<div id="nav" role="navigation">
			<ul id='menu'>
<li id="menu-item-n-About"><a href="ImageJ">About</a></li>
<li id="menu-item-n-Downloads"><a href="Downloads">Downloads</a></li>
<li id="menu-item-n-Learn"><a href="Learn">Learn</a></li>
<li id="menu-item-n-Develop"><a href="Development">Develop</a></li>
<li id="menu-item-n-News"><a href="News">News</a></li>
<li id="menu-item-n-Events"><a href="Events">Events</a></li>
<li id="menu-item-n-Help"><a href="Help">Help</a></li>
</ul>
			</div>
		</div>

		<div id="mw-js-message"></div>
		
		<div id="main" role="main">
			<div id="nav-meta">
			<span id="ca-nstab-main" class="selected"><a href="BoneJ_experimental.html" title="View the content page [c]" accesskey="c">Page</a></span><span class="meta-sep">|</span><span id="ca-talk" class="new"><a href="index.php?title=Talk:BoneJ&amp;action=edit&amp;redlink=1" rel="discussion" title="Discussion about the content page [t]" accesskey="t">Discussion</a></span><span class="meta-sep">|</span><span id="ca-viewsource"><a href="index.php?title=BoneJ&amp;action=edit" title="This page is protected.&#10;You can view its source [e]" accesskey="e">View source</a></span><span class="meta-sep">|</span><span id="ca-history"><a href="index.php?title=BoneJ&amp;action=history" title="Past revisions of this page [h]" accesskey="h">History</a></span><span class="meta-sep">|</span>			</div>

			<div id="bodyContent">
				<h1>BoneJ</h1>
				
				<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><table class="infobox" cellspacing="5" style="max-width: 31em; font-size: 90%; text-align: left; float: right; border: 1px solid #a0a0a0;">
<tr>
<th colspan="2" style="text-align: center; font-size: 130%;"> <div style="float: left"><a href="File:Bonej-icon.png" class="image"><img alt="Bonej-icon.png" src="_images/5/57/Bonej-icon.png" width="96" height="108" /></a></div>BoneJ (ImageJ)<div style="clear: left;"></div>
</th></tr>
<tr>
<th> Author
</th>
<td> <a href="User:Mdoube" title="User:Mdoube">Michael Doube</a>, <a href="User:Rdom" title="User:Rdom">Richard Domander</a>, <a href="User:Alessandrofelder" title="User:Alessandrofelder">Alessandro Felder</a>
</td></tr>




<tr>
<th> Maintainer
</th>
<td> <a href="User:Mdoube" title="User:Mdoube">Michael Doube</a>, <a href="User:Alessandrofelder" title="User:Alessandrofelder">Alessandro Felder</a>
</td></tr>


<tr>
<th> Source
</th>
<td> <a rel="nofollow" class="external text" href="https://github.com/bonej-org/BoneJ2/">on GitHub</a>, <a rel="nofollow" class="external text" href="https://doi.org/10.5281/zenodo.1427262">doi:10.5281/zenodo.1427262</a>
</td></tr>
<tr>
<th> Initial release
</th>
<td> Dec 11<sup>th</sup>, 2017
</td></tr>
<tr>
<th> Latest version
</th>
<td> <a rel="nofollow" class="external text" href="https://github.com/bonej-org/BoneJ2/releases">styloid-r7</a>
</td></tr>
<tr>
<th> Development status
</th>
<td> Active
</td></tr>


</table>
<p><br />
BoneJ is a collection of skeletal biology plug-ins for ImageJ.
This is the new, modernized version of the software available through the ImageJ <a rel="nofollow" class="external text" href="Updater">updater</a>. Its update site is called <a rel="nofollow" class="external text" href="http://sites.imagej.net/BoneJ">BoneJ</a>. For the old ImageJ1 version, see <a href="BoneJ1" title="BoneJ1">BoneJ1</a>.
</p><p>This version works with the latest Fiji, and complies with the modern ImageJ <a href="Architecture" title="Architecture">architecture</a>. Most plug-ins also now support hyperstacks, i.e. images with multiple channels or time frames.
</p><p>Below is the documentation for the plug-ins included in BoneJ.
</p>
<div id="toc" class="toc"><div id="toctitle"><h2>Contents</h2></div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="BoneJ_experimental.html#Installation"><span class="tocnumber">1</span> <span class="toctext">Installation</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="BoneJ_experimental.html#Analyse_skeleton"><span class="tocnumber">2</span> <span class="toctext">Analyse skeleton</span></a>
<ul>
<li class="toclevel-2 tocsection-3"><a href="BoneJ_experimental.html#Suitable_images"><span class="tocnumber">2.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-4"><a href="BoneJ_experimental.html#Differences_to_BoneJ1"><span class="tocnumber">2.2</span> <span class="toctext">Differences to BoneJ1</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-5"><a href="BoneJ_experimental.html#Anisotropy"><span class="tocnumber">3</span> <span class="toctext">Anisotropy</span></a>
<ul>
<li class="toclevel-2 tocsection-6"><a href="BoneJ_experimental.html#Suitable_images_2"><span class="tocnumber">3.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-7"><a href="BoneJ_experimental.html#Parameters"><span class="tocnumber">3.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-8"><a href="BoneJ_experimental.html#Convergence_Analysis"><span class="tocnumber">3.3</span> <span class="toctext">Convergence Analysis</span></a></li>
<li class="toclevel-2 tocsection-9"><a href="BoneJ_experimental.html#Results"><span class="tocnumber">3.4</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-10"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_2"><span class="tocnumber">3.5</span> <span class="toctext">Differences to BoneJ1</span></a></li>
<li class="toclevel-2 tocsection-11"><a href="BoneJ_experimental.html#Related_publications"><span class="tocnumber">3.6</span> <span class="toctext">Related publications</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-12"><a href="BoneJ_experimental.html#Area.2FVolume_fraction"><span class="tocnumber">4</span> <span class="toctext">Area/Volume fraction</span></a>
<ul>
<li class="toclevel-2 tocsection-13"><a href="BoneJ_experimental.html#Suitable_images_3"><span class="tocnumber">4.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-14"><a href="BoneJ_experimental.html#Results_2"><span class="tocnumber">4.2</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-15"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_3"><span class="tocnumber">4.3</span> <span class="toctext">Differences to BoneJ1</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-16"><a href="BoneJ_experimental.html#Calibrate_SCANCO_.28WIP.29"><span class="tocnumber">5</span> <span class="toctext">Calibrate SCANCO (WIP)</span></a>
<ul>
<li class="toclevel-2 tocsection-17"><a href="BoneJ_experimental.html#Suitable_images_4"><span class="tocnumber">5.1</span> <span class="toctext">Suitable images</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-18"><a href="BoneJ_experimental.html#Check_voxel_depth_.28WIP.29"><span class="tocnumber">6</span> <span class="toctext">Check voxel depth (WIP)</span></a>
<ul>
<li class="toclevel-2 tocsection-19"><a href="BoneJ_experimental.html#Suitable_images_5"><span class="tocnumber">6.1</span> <span class="toctext">Suitable images</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-20"><a href="BoneJ_experimental.html#Connectivity"><span class="tocnumber">7</span> <span class="toctext">Connectivity</span></a>
<ul>
<li class="toclevel-2 tocsection-21"><a href="BoneJ_experimental.html#Suitable_images_6"><span class="tocnumber">7.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-22"><a href="BoneJ_experimental.html#Results_3"><span class="tocnumber">7.2</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-23"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_4"><span class="tocnumber">7.3</span> <span class="toctext">Differences to BoneJ1</span></a></li>
<li class="toclevel-2 tocsection-24"><a href="BoneJ_experimental.html#Related_publications_2"><span class="tocnumber">7.4</span> <span class="toctext">Related publications</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-25"><a href="BoneJ_experimental.html#Delete_slice_range_.28WIP.29"><span class="tocnumber">8</span> <span class="toctext">Delete slice range (WIP)</span></a>
<ul>
<li class="toclevel-2 tocsection-26"><a href="BoneJ_experimental.html#Suitable_images_7"><span class="tocnumber">8.1</span> <span class="toctext">Suitable images</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-27"><a href="BoneJ_experimental.html#Ellipsoid_factor"><span class="tocnumber">9</span> <span class="toctext">Ellipsoid factor</span></a>
<ul>
<li class="toclevel-2 tocsection-28"><a href="BoneJ_experimental.html#Suitable_images_8"><span class="tocnumber">9.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-29"><a href="BoneJ_experimental.html#Parameters_2"><span class="tocnumber">9.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-30"><a href="BoneJ_experimental.html#Results_4"><span class="tocnumber">9.3</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-31"><a href="BoneJ_experimental.html#Related_publications_3"><span class="tocnumber">9.4</span> <span class="toctext">Related publications</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-32"><a href="BoneJ_experimental.html#Fit_ellipsoid"><span class="tocnumber">10</span> <span class="toctext">Fit ellipsoid</span></a>
<ul>
<li class="toclevel-2 tocsection-33"><a href="BoneJ_experimental.html#Suitable_images_9"><span class="tocnumber">10.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-34"><a href="BoneJ_experimental.html#Parameters_3"><span class="tocnumber">10.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-35"><a href="BoneJ_experimental.html#Results_5"><span class="tocnumber">10.3</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-36"><a href="BoneJ_experimental.html#Differences_from_BoneJ1"><span class="tocnumber">10.4</span> <span class="toctext">Differences from BoneJ1</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-37"><a href="BoneJ_experimental.html#Fit_sphere_.28WIP.29"><span class="tocnumber">11</span> <span class="toctext">Fit sphere (WIP)</span></a>
<ul>
<li class="toclevel-2 tocsection-38"><a href="BoneJ_experimental.html#Suitable_images_10"><span class="tocnumber">11.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-39"><a href="BoneJ_experimental.html#Parameters_4"><span class="tocnumber">11.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-40"><a href="BoneJ_experimental.html#Results_6"><span class="tocnumber">11.3</span> <span class="toctext">Results</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-41"><a href="BoneJ_experimental.html#Fractal_dimension"><span class="tocnumber">12</span> <span class="toctext">Fractal dimension</span></a>
<ul>
<li class="toclevel-2 tocsection-42"><a href="BoneJ_experimental.html#Suitable_images_11"><span class="tocnumber">12.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-43"><a href="BoneJ_experimental.html#Parameters_5"><span class="tocnumber">12.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-44"><a href="BoneJ_experimental.html#Results_7"><span class="tocnumber">12.3</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-45"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_5"><span class="tocnumber">12.4</span> <span class="toctext">Differences to BoneJ1</span></a></li>
<li class="toclevel-2 tocsection-46"><a href="BoneJ_experimental.html#Related_publications_4"><span class="tocnumber">12.5</span> <span class="toctext">Related publications</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-47"><a href="BoneJ_experimental.html#Inter-trabecular_angles"><span class="tocnumber">13</span> <span class="toctext">Inter-trabecular angles</span></a>
<ul>
<li class="toclevel-2 tocsection-48"><a href="BoneJ_experimental.html#Suitable_images_12"><span class="tocnumber">13.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-49"><a href="BoneJ_experimental.html#Parameters_6"><span class="tocnumber">13.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-50"><a href="BoneJ_experimental.html#Results_8"><span class="tocnumber">13.3</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-51"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_6"><span class="tocnumber">13.4</span> <span class="toctext">Differences to BoneJ1</span></a></li>
<li class="toclevel-2 tocsection-52"><a href="BoneJ_experimental.html#Related_publications_5"><span class="tocnumber">13.5</span> <span class="toctext">Related publications</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-53"><a href="BoneJ_experimental.html#Thickness"><span class="tocnumber">14</span> <span class="toctext">Thickness</span></a>
<ul>
<li class="toclevel-2 tocsection-54"><a href="BoneJ_experimental.html#Suitable_images_13"><span class="tocnumber">14.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-55"><a href="BoneJ_experimental.html#Parameters_7"><span class="tocnumber">14.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-56"><a href="BoneJ_experimental.html#Results_9"><span class="tocnumber">14.3</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-57"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_7"><span class="tocnumber">14.4</span> <span class="toctext">Differences to BoneJ1</span></a></li>
<li class="toclevel-2 tocsection-58"><a href="BoneJ_experimental.html#Related_publications_6"><span class="tocnumber">14.5</span> <span class="toctext">Related publications</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-59"><a href="BoneJ_experimental.html#Moments_of_inertia"><span class="tocnumber">15</span> <span class="toctext">Moments of inertia</span></a>
<ul>
<li class="toclevel-2 tocsection-60"><a href="BoneJ_experimental.html#Suitable_images_14"><span class="tocnumber">15.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-61"><a href="BoneJ_experimental.html#Parameters_8"><span class="tocnumber">15.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-62"><a href="BoneJ_experimental.html#Results_10"><span class="tocnumber">15.3</span> <span class="toctext">Results</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-63"><a href="BoneJ_experimental.html#Orientation"><span class="tocnumber">16</span> <span class="toctext">Orientation</span></a>
<ul>
<li class="toclevel-2 tocsection-64"><a href="BoneJ_experimental.html#Suitable_images_15"><span class="tocnumber">16.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-65"><a href="BoneJ_experimental.html#Parameters_9"><span class="tocnumber">16.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-66"><a href="BoneJ_experimental.html#Results_11"><span class="tocnumber">16.3</span> <span class="toctext">Results</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-67"><a href="BoneJ_experimental.html#Particle_Analyser"><span class="tocnumber">17</span> <span class="toctext">Particle Analyser</span></a>
<ul>
<li class="toclevel-2 tocsection-68"><a href="BoneJ_experimental.html#Suitable_images_16"><span class="tocnumber">17.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-69"><a href="BoneJ_experimental.html#Parameters_10"><span class="tocnumber">17.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-70"><a href="BoneJ_experimental.html#Output"><span class="tocnumber">17.3</span> <span class="toctext">Output</span></a></li>
<li class="toclevel-2 tocsection-71"><a href="BoneJ_experimental.html#Citation"><span class="tocnumber">17.4</span> <span class="toctext">Citation</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-72"><a href="BoneJ_experimental.html#Purify"><span class="tocnumber">18</span> <span class="toctext">Purify</span></a>
<ul>
<li class="toclevel-2 tocsection-73"><a href="BoneJ_experimental.html#Suitable_images_17"><span class="tocnumber">18.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-74"><a href="BoneJ_experimental.html#Parameters_11"><span class="tocnumber">18.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-75"><a href="BoneJ_experimental.html#Results_12"><span class="tocnumber">18.3</span> <span class="toctext">Results</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-76"><a href="BoneJ_experimental.html#Skeletonise"><span class="tocnumber">19</span> <span class="toctext">Skeletonise</span></a>
<ul>
<li class="toclevel-2 tocsection-77"><a href="BoneJ_experimental.html#Suitable_images_18"><span class="tocnumber">19.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-78"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_8"><span class="tocnumber">19.2</span> <span class="toctext">Differences to BoneJ1</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-79"><a href="BoneJ_experimental.html#Slice_geometry"><span class="tocnumber">20</span> <span class="toctext">Slice geometry</span></a>
<ul>
<li class="toclevel-2 tocsection-80"><a href="BoneJ_experimental.html#Suitable_images_19"><span class="tocnumber">20.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-81"><a href="BoneJ_experimental.html#Parameters_12"><span class="tocnumber">20.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-82"><a href="BoneJ_experimental.html#Results_13"><span class="tocnumber">20.3</span> <span class="toctext">Results</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-83"><a href="BoneJ_experimental.html#Surface_area"><span class="tocnumber">21</span> <span class="toctext">Surface area</span></a>
<ul>
<li class="toclevel-2 tocsection-84"><a href="BoneJ_experimental.html#Suitable_images_20"><span class="tocnumber">21.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-85"><a href="BoneJ_experimental.html#Parameters_13"><span class="tocnumber">21.2</span> <span class="toctext">Parameters</span></a></li>
<li class="toclevel-2 tocsection-86"><a href="BoneJ_experimental.html#Results_14"><span class="tocnumber">21.3</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-87"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_9"><span class="tocnumber">21.4</span> <span class="toctext">Differences to BoneJ1</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-88"><a href="BoneJ_experimental.html#Surface_fraction"><span class="tocnumber">22</span> <span class="toctext">Surface fraction</span></a>
<ul>
<li class="toclevel-2 tocsection-89"><a href="BoneJ_experimental.html#Suitable_images_21"><span class="tocnumber">22.1</span> <span class="toctext">Suitable images</span></a></li>
<li class="toclevel-2 tocsection-90"><a href="BoneJ_experimental.html#Results_15"><span class="tocnumber">22.2</span> <span class="toctext">Results</span></a></li>
<li class="toclevel-2 tocsection-91"><a href="BoneJ_experimental.html#Differences_to_BoneJ1_10"><span class="tocnumber">22.3</span> <span class="toctext">Differences to BoneJ1</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-92"><a href="BoneJ_experimental.html#Results_table"><span class="tocnumber">23</span> <span class="toctext">Results table</span></a></li>
<li class="toclevel-1 tocsection-93"><a href="BoneJ_experimental.html#Usage_reporting"><span class="tocnumber">24</span> <span class="toctext">Usage reporting</span></a>
<ul>
<li class="toclevel-2 tocsection-94"><a href="BoneJ_experimental.html#What_is_collected.3F"><span class="tocnumber">24.1</span> <span class="toctext">What is collected?</span></a></li>
<li class="toclevel-2 tocsection-95"><a href="BoneJ_experimental.html#What_does_the_report_look_like.3F"><span class="tocnumber">24.2</span> <span class="toctext">What does the report look like?</span></a></li>
<li class="toclevel-2 tocsection-96"><a href="BoneJ_experimental.html#For_what_is_the_data_used.3F"><span class="tocnumber">24.3</span> <span class="toctext">For what is the data used?</span></a></li>
<li class="toclevel-2 tocsection-97"><a href="BoneJ_experimental.html#That_is_noble_but_my_work_is_top_secret"><span class="tocnumber">24.4</span> <span class="toctext">That is noble but my work is top secret</span></a></li>
<li class="toclevel-2 tocsection-98"><a href="BoneJ_experimental.html#Can_I_see_the_report.3F"><span class="tocnumber">24.5</span> <span class="toctext">Can I see the report?</span></a></li>
<li class="toclevel-2 tocsection-99"><a href="BoneJ_experimental.html#How_is_it_implemented.3F"><span class="tocnumber">24.6</span> <span class="toctext">How is it implemented?</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-100"><a href="BoneJ_experimental.html#Where_is_my_favourite_plug-in.3F"><span class="tocnumber">25</span> <span class="toctext">Where is my favourite plug-in?</span></a></li>
<li class="toclevel-1 tocsection-101"><a href="BoneJ_experimental.html#Licence"><span class="tocnumber">26</span> <span class="toctext">Licence</span></a></li>
<li class="toclevel-1 tocsection-102"><a href="BoneJ_experimental.html#Citation_2"><span class="tocnumber">27</span> <span class="toctext">Citation</span></a></li>
<li class="toclevel-1 tocsection-103"><a href="BoneJ_experimental.html#Funding"><span class="tocnumber">28</span> <span class="toctext">Funding</span></a></li>
</ul>
</div>

<h2><span class="mw-headline" id="Installation">Installation</span></h2>
<p><a href="File:Install-bonej.png" class="image" title="Installation steps"><img alt="Installation steps" src="_images/7/71/Install-bonej.png" width="400" height="257" /></a>
</p>
<ol><li> <a rel="nofollow" class="external text" href="Downloads">Download</a> the latest version of Fiji for your operating system</li>
<li> Launch Fiji</li>
<li> Select in the menu <i>Help</i> &gt; <i>Update...</i></li>
<li> Click <i>Manage update sites</i></li>
<li> Check <i>BoneJ</i></li>
<li> Click <i>Close</i></li>
<li> Click <i>Apply changes</i></li></ol>
<p>After the downloads have finished, close and restart Fiji.
</p>
<h2><span class="mw-headline" id="Analyse_skeleton">Analyse skeleton</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Analyse skeleton</i>.
</p><p>This plug-in simply includes <a href="AnalyzeSkeleton" title="AnalyzeSkeleton">AnalyzeSkeleton</a> in BoneJ. It adds some additional validation to check that your image suits the tool. It also skeletonizes your image by calling <a href="Skeletonize3D" title="Skeletonize3D">Skeletonize3D</a> if needed.
</p>
<h4><span class="mw-headline" id="Suitable_images">Suitable images</span></h4>
<p>The input image must be 2D or 3D, 8-bit and binary. Hyperstacks are not supported.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1">Differences to BoneJ1</span></h4>
<p>Calls the latest version of <a href="AnalyzeSkeleton" title="AnalyzeSkeleton">AnalyzeSkeleton</a>.
</p>
<h2><span class="mw-headline" id="Anisotropy">Anisotropy</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Anisotropy</i>.
</p><p>Anisotropy is used to quantify the directionality of trabecular bone. It tells whether the trabeculae have a certain orientation, or if they're randomly aligned. The method to measure anisotropy is fairly complex and consists of multiple steps:
</p>
<ol><li> Find mean intercept length (MIL) vectors from <img class="mwe-math-fallback-image-inline tex" alt="n" src="_images/math/7/b/8/7b8b965ad4bca0e41ab51de7b31363a1.png" /> directions</li>
<li> Plot MIL vectors into a point cloud</li>
<li> Solve the equation of an ellipsoid that best fits the point cloud</li>
<li> Calculate the degree of anisotropy from the radii of the ellipsoid</li></ol>
<p>It's important to note that algorithm is stochastic and does not guarantee exact results. Thus it's recommended to run it several times to establish the degree of anisotropy in your image.
</p>
<div class="floatleft"><a href="File:PhaseChanges.png" class="image" title="The red dots mark phase changes"><img alt="The red dots mark phase changes" src="_images/8/8a/PhaseChanges.png" width="150" height="148" /></a></div>
<p>In the first step the algorithm draws parallel lines over the input image in direction <img class="mwe-math-fallback-image-inline tex" alt="\mathbf{v}" src="_images/math/0/a/a/0aa3ec374bdc0d6a17aecbb6bcda6a89.png" />. The direction is chosen randomly. Each line segment in the image stack is sampled to find points where it changes from background to foreground, i.e. where the line enters an object. The points are called <i>phase changes</i>, in the adjacent figure they're marked with red dots. After the sampling is complete, the algorithm forms a <i>MIL vector</i>, whose length is the total length of the line segments divided by the total number of phase changes found. The MIL vector has the same direction as <img class="mwe-math-fallback-image-inline tex" alt="\mathbf{v}" src="_images/math/0/a/a/0aa3ec374bdc0d6a17aecbb6bcda6a89.png" />. Drawing and sampling the lines is repeated for <img class="mwe-math-fallback-image-inline tex" alt="n" src="_images/math/7/b/8/7b8b965ad4bca0e41ab51de7b31363a1.png" /> directions, and the method creates <img class="mwe-math-fallback-image-inline tex" alt="n" src="_images/math/7/b/8/7b8b965ad4bca0e41ab51de7b31363a1.png" /> MIL vectors.
</p><p>After the MIL vectors have been calculated, they are added to a point cloud (a collection of points) around the origin. Then the method tries solve the equation of an ellipsoid that would fit the cloud. There may be no solution, especially if there are few points. That is, the fitting may fail at which point the plug-in stops. The radii of this ellipsoid determine the degree of anisotropy (see <a rel="nofollow" class="external text" href="index.php?title=BoneJ_experimental&amp;action=submit#Results">results</a>).
</p>
<div class="floatright"><a href="File:MilCube.png" class="image" title="Projecting lines from a plane"><img alt="Projecting lines from a plane" src="_images/c/c5/MilCube.png" width="200" height="196" /></a></div>
<p>In more detail, the lines in the first step are projected from a <img class="mwe-math-fallback-image-inline tex" alt="d * d" src="_images/math/a/4/b/a4bd906ee23757091273a971abde4667.png" /> plane with normal <img class="mwe-math-fallback-image-inline tex" alt="\mathbf{v}" src="_images/math/0/a/a/0aa3ec374bdc0d6a17aecbb6bcda6a89.png" /> (see the adjacent figure). The size <img class="mwe-math-fallback-image-inline tex" alt="d = \sqrt{w^{2} + h^{2} + d^{2}}" src="_images/math/f/5/0/f502c5a5be1cc092e614f8eb0c3f67cb.png" />, where <img class="mwe-math-fallback-image-inline tex" alt="w, h, d" src="_images/math/7/0/3/70388378f6d2646c58cbab636ec62d19.png" /> are the dimensions of the image stack. Each of the lines goes through a random point <img class="mwe-math-fallback-image-inline tex" alt="\mathbf{o}" src="_images/math/a/b/0/ab08120b5c856534253f627146b7238a.png" /> on the plane. The points <img class="mwe-math-fallback-image-inline tex" alt="\mathbf{o}" src="_images/math/a/b/0/ab08120b5c856534253f627146b7238a.png" /> are random, but evenly distributed across the plane. The lines are drawn until they intercept the stack edges at points <img class="mwe-math-fallback-image-inline tex" alt="\mathbf{o} + t_{min}\mathbf{v}" src="_images/math/b/b/1/bb1acc8e224eb1384ddd3fe5236c47fc.png" />,<img class="mwe-math-fallback-image-inline tex" alt="\mathbf{o} + t_{max}\mathbf{v}" src="_images/math/9/3/8/9388100a8a0690baed91915c58a9fede.png" /> (the algorithm solves for <img class="mwe-math-fallback-image-inline tex" alt="t_{min}" src="_images/math/1/5/a/15a2a3abf3920ff2f598b237380a578f.png" />, <img class="mwe-math-fallback-image-inline tex" alt="t_{max}" src="_images/math/a/9/a/a9ab2f28b21d25555dd13dc7285a98ba.png" />). These line segments within the stack are then sampled for phase changes. In this drawing method some lines may miss the image stack completely, but conversely there aren't any areas in the stack that don't have a chance of being sampled.
</p>
<h4><span class="mw-headline" id="Suitable_images_2">Suitable images</span></h4>
<p>A 3D binary image.
</p><p>The plug-in is intended to analyse the "texture" of an object thus it suits samples of a larger whole, e.g. a trabecular cube that fills the whole stack. The results for whole objects such as the pre-packaged <i>Bat Cochlea Volume</i> ImageJ sample image are not really meaningful.
</p>
<h4><span class="mw-headline" id="Parameters">Parameters</span></h4>
<ul><li> <b>Directions</b>: number of times sampling is performed from different directions. The minimum is <img class="mwe-math-fallback-image-inline tex" alt="9" src="_images/math/4/5/c/45c48cce2e2d7fbdea1afc51c7c6ad26.png" />, because that's how many independent variables the algorithm needs to solve the "shape" of the orientation in the image.</li>
<li> <b>Lines per direction</b>: controls how many parallel lines are drawn per each direction. Each line has the length <img class="mwe-math-fallback-image-inline tex" alt="d = \sqrt{width^2 + height^2 + depth^2}" src="_images/math/a/5/b/a5be268e5fe82161ded33c7cab7f30a3.png" />.</li>
<li> <b>Sampling increment</b>: controls the distance between sampling points along a line. The default and minimum is &lt;math\sqrt{3}&lt;/math&gt;. The number of samples taken per line depends on the length of the line segment within the image stack.</li>
<li> <b>Recommended minimum</b>: if checked, then the above three parameters are set to the recommended minimum values. In our tests we found that with these values the results are quite stable, and fitting unlikely to fail. However, these minimums are not guaranteed to be the best settings for your image.</li>
<li> <b>Show radii</b>: if checked, then the radii of the fitted ellipsoid are shown in the results table.</li>
<li> <b>Show Eigens</b> if checked, then the eigenvectors and values of the fitted ellipsoid are shown in the results table.</li>
<li> <b>Display MIL vectors</b> if checked, shows the MIL vectors plotted in a point cloud on 3D Viewer</li></ul>
<p><br />
</p>
<h4><span class="mw-headline" id="Convergence_Analysis">Convergence Analysis</span></h4>
<p>It is helpful to run a convergence analysis to determine settings that lead to a stable result in a reasonable amount of time. We recommend running the following macro on a typical image from your experiments.
</p>
<pre class="brush:javascript">
 //number of directions to draw probes
 nDirsMax = 32768; //&lt;- edit to suit your needs
 //number of lines per direction
 nLinesMax = 1024; //&lt;- edit to suit your needs
 
 // --- No need to edit the rest
 row = 0;
 setBatchMode(true);
 for (nDirs = 16; nDirs &lt;= nDirsMax; nDirs *= 2){
    for (nLines = 1; nLines &lt;= nLinesMax; nLines *= 2){
 	startTime = getTime();
 	run("Anisotropy", "inputimage=net.imagej.ImgPlus@73956688 directions="+nDirs+" lines="+nLines+" samplingincrement=1.73 recommendedmin=true printradii=true printeigens=true displaymilvectors=false instruction=\"\"");
     	endTime = getTime();
     	duration = endTime - startTime;
     	setResult("nDirs", row, nDirs);
     	setResult("nLines", row, nLines);
     	setResult("Duration", row, duration);
     	updateResults();
     	row++;
     }
 }
</pre>
<p>This macro will output two tables, one with anisotropy results and one logging directions, lines, and duration. To complete the convergence analysis the two tables have to be copy-pasted into a spreadsheet so that anisotropy results can be plotted alongside timings and input parameters.
</p>
<h4><span class="mw-headline" id="Results">Results</span></h4>
<ul><li> <b>Degree of anisotropy</b>: how much orientation there is in the structure. <img class="mwe-math-fallback-image-inline tex" alt="0.0" src="_images/math/3/0/5/30565a8911a6bb487e3745c0ea3c8224.png" /> means the image is completely isotropic, the sample has no directionality whatsoever. <img class="mwe-math-fallback-image-inline tex" alt="1.0" src="_images/math/e/4/c/e4c2e8edac362acab7123654b9e73432.png" /> means there is an extreme prevailing orientation in the structure of the image.</li>
<li> <b>Radii of fitted ellipsoid</b> (optional): the lengths of the radii <img class="mwe-math-fallback-image-inline tex" alt="a \leq b \leq c" src="_images/math/f/2/8/f280272369287127c5295d5b0deac01f.png" /> of the ellipsoid fitted on the MIL points. Degree of anisotropy equals <img class="mwe-math-fallback-image-inline tex" alt="= 1.0 - \frac{1}{c^2}/\frac{1}{a^2}" src="_images/math/9/4/9/949a770c358815d21f88af00089e99cb.png" />.</li>
<li> <b>Eigenvectors and values</b> (optional): the values <img class="mwe-math-fallback-image-inline tex" alt="m00, m01, m02 ... m22" src="_images/math/8/1/c/81c31f1de47f1996824a135f0bb16c31.png" /> correspond to the <img class="mwe-math-fallback-image-inline tex" alt="x, y, z" src="_images/math/7/8/b/78b70da0fb6369f45abaccaaef4cabe9.png" /> components of the three eigenvectors of the fitted ellipsoid. The eigenvalues <img class="mwe-math-fallback-image-inline tex" alt="D1, D2, D3" src="_images/math/4/a/6/4a60f019a2df1858143a6895c0f14b9c.png" /> correspond to <img class="mwe-math-fallback-image-inline tex" alt="\frac{1}{c^2}, \frac{1}{b^2}, \frac{1}{a^2}" src="_images/math/8/3/e/83e9069bd833800c56cd8c49499655b0.png" /> respectively, where <img class="mwe-math-fallback-image-inline tex" alt="a, b, c" src="_images/math/a/4/4/a44c56c8177e32d3613988f4dba7962e.png" /> are the radii of the ellipsoid.</li></ul>
<p>The measures are reported separately for each 3D subspace in the image, i.e. for each channel and time frame.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_2">Differences to BoneJ1</span></h4>
<ul><li> This Anisotropy is not repeated until results stabilise like in BoneJ1. </li>
<li> MIL vectors are drawn differently. In BoneJ1 they're drawn in sphere-shapes around random seed points. Here parallel lines from different directions are drawn through the whole stack. We think this method produces more uniform sampling, i.e. there's less chance of a directional bias.</li></ul>
<h4><span class="mw-headline" id="Related_publications">Related publications</span></h4>
<ul><li> Odgaard A (1997), <i>Three-dimensional methods for quantification of cancellous bone architecture</i>, Bone, 20, 315-328, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016/S8756-3282(97)00007-0">doi:10.1016/S8756-3282(97)00007-0</a></li>
<li> Harrigan TP, Mann RW (1984), <i>Characterization of microstructural anisotropy in orthotropic materials using a second rank tensor</i>, J Mater Sci, 19, 761-767, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1007/BF00540446">doi:10.1007/BF00540446</a></li></ul>
<h2><span class="mw-headline" id="Area.2FVolume_fraction">Area/Volume fraction</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Fraction &gt; Area/Volume fraction</i>
</p><p><i>Area/Volume fraction</i> calculates the fraction of bone in an image by it to the whole image. It counts all the foreground voxels, which it assumes represent bone, and compares them to the total number of voxels in the image. More formally defined, the plug-in calculates the fraction <i>BV/TV</i>, which is the volume of mineralised bone <i>BV</i> per unit volume of the sample <i>TV</i>. In case of a 2D image, it calculates the fraction <i>BA/TA</i>, which is the area of bone per unit area of the sample.
</p>
<h4><span class="mw-headline" id="Suitable_images_3">Suitable images</span></h4>
<p>A 2D or 3D binary image
</p>
<h4><span class="mw-headline" id="Results_2">Results</span></h4>
<ul><li> <b>Bone volume</b>: volume of the bone voxels.</li>
<li> <b>Total volume</b>: volume of the whole image.</li>
<li> <b>Volume ratio</b>: ratio of bone to total volume.</li></ul>
<p>The measures are reported separately for each 2D/3D subspace in the image, i.e. for each channel and time frame. Results will be for <i>area</i> if image is 2D.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_3">Differences to BoneJ1</span></h4>
<ul><li> In BoneJ1 the plug-in was called <i>Volume fraction</i></li>
<li> Can process 2D images</li>
<li> Supports hyperstacks </li></ul>
<h2><span class="mw-headline" id="Calibrate_SCANCO_.28WIP.29">Calibrate SCANCO (WIP)</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Analyze &gt; Calibrate SCANCO</i>
</p><p>Applies the <i>mg HA/ccm</i> pixel value calibration, i.e. <i>HU</i> or Hounfield unit calibration  stored in the .isq format metadata to the image.
</p>
<h4><span class="mw-headline" id="Suitable_images_4">Suitable images</span></h4>
<p>An .isq format image generated by Scanco X-ray microtomography scanners.
</p>
<h2><span class="mw-headline" id="Check_voxel_depth_.28WIP.29">Check voxel depth (WIP)</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Stacks &gt; Check Voxel Depth</i>
</p><p>Checks whether slice spacing has been calibrated correctly. Some DICOM images contain slice thickness data in the header information, but thickness is not necessarily the same as the physical distance between consecutive slices' positions.
</p>
<h4><span class="mw-headline" id="Suitable_images_5">Suitable images</span></h4>
<p>A 3D image.
</p>
<h2><span class="mw-headline" id="Connectivity">Connectivity</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Connectivity</i>.
</p><p>The Connectivity plug-in is designed to estimate the number of connected structures i.e. trabeculae in a trabecular network. This <i>connectivity</i> measure is related to a topological number <img class="mwe-math-fallback-image-inline tex" alt="\chi" src="_images/math/7/9/c/79c40dd2b43f3c03eaf88b5fc4c199b8.png" /> known as <i>Euler characteristic</i>, <i>Euler number</i> or <i>Euler-Poincaré characteristic</i>. Mathematically defined connectivity is <img class="mwe-math-fallback-image-inline tex" alt="= 1 - (\chi + \Delta\chi)" src="_images/math/0/1/6/01675d9b16bfa07f7dc066f1471d817d.png" />. Roughly speaking, <img class="mwe-math-fallback-image-inline tex" alt="\chi" src="_images/math/7/9/c/79c40dd2b43f3c03eaf88b5fc4c199b8.png" /> describes the shape or structure of a topological space. It can also be expressed as <img class="mwe-math-fallback-image-inline tex" alt="\chi = objects - handles + cavities" src="_images/math/6/0/c/60c43352598b2a33e0e35729ce2f8632.png" />, where a handle is a hole that goes through an object (e.g the hole in a doughnut, or the ear of a coffee mug), and a cavity is enclosed inside one. When measuring trabecular cubes, you need to add <img class="mwe-math-fallback-image-inline tex" alt="\Delta\chi" src="_images/math/7/9/2/792881dbe2c1da8ebbebf1bbfe484e9e.png" /> to <img class="mwe-math-fallback-image-inline tex" alt="\chi" src="_images/math/7/9/c/79c40dd2b43f3c03eaf88b5fc4c199b8.png" /> to get a more accurate estimate of the connectivity of the whole network. The term <img class="mwe-math-fallback-image-inline tex" alt="\Delta\chi" src="_images/math/7/9/2/792881dbe2c1da8ebbebf1bbfe484e9e.png" /> corrects for the change in the topology of an object, when it's cut to pieces. 
</p><p>NB some other Euler characteristic implementations report <img class="mwe-math-fallback-image-inline tex" alt="\chi + \Delta\chi" src="_images/math/b/e/c/bec379ec70dca49ef79c634ec1c4a908.png" /> as <img class="mwe-math-fallback-image-inline tex" alt="\chi" src="_images/math/7/9/c/79c40dd2b43f3c03eaf88b5fc4c199b8.png" />, i.e. in them the correction <img class="mwe-math-fallback-image-inline tex" alt="\Delta\chi" src="_images/math/7/9/2/792881dbe2c1da8ebbebf1bbfe484e9e.png" /> is implicit.
</p>
<h4><span class="mw-headline" id="Suitable_images_6">Suitable images</span></h4>
<p>The input image must be 3D and binary. The plug-in assumes that there is only one particle in the foreground; to achieve this, run <i>Purify</i>. Having more than one object often leads to negative connectivity.
</p>
<h4><span class="mw-headline" id="Results_3">Results</span></h4>
<ul><li> <b>Euler characteristic (χ)</b>: describes the shape of the object(s) in the image, <img class="mwe-math-fallback-image-inline tex" alt="\chi = objects - handles + cavities" src="_images/math/6/0/c/60c43352598b2a33e0e35729ce2f8632.png" />.</li>
<li> <b>Corrected Euler (χ + Δχ)</b>: the Euler characteristic of the part corrected for edge effects to fit the whole.</li>
<li> <b>Connectivity</b>: gives an estimate of the number of connected trabeculae in the image. Equal to <img class="mwe-math-fallback-image-inline tex" alt="1 - (\chi + \Delta\chi)" src="_images/math/a/6/d/a6df405cbac2f0f3a655bf9fd3a833f2.png" />.</li>
<li> <b>Conn. density</b>: connectivity divided by unit volume. </li></ul>
<p>The measures are reported separately for each 3D subspace in the image, i.e. for each channel and time frame.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_4">Differences to BoneJ1</span></h4>
<ul><li> Supports hyperstacks</li>
<li> The old version reported <i>Corrected Euler (χ + Δχ)</i> incorrectly as <i>Δχ</i></li></ul>
<h4><span class="mw-headline" id="Related_publications_2">Related publications</span></h4>
<ul><li> Odgaard A, Gundersen HJG (1993), <i>Quantification of connectivity in cancellous bone, with special emphasis on 3-D reconstructions</i>, Bone 14: 173-182, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016/8756-3282(93)90245-6">doi:10.1016/8756-3282(93)90245-6.</a></li>
<li> Toriwaki J, Yonekura T (2002), <i>Euler number and connectivity indexes of a three dimensional digital picture</i>, <a rel="nofollow" class="external text" href="http://www.scipress.org/journals/forma/abstract/1703/17030183.html">Forma 17: 183-209</a></li></ul>
<h2><span class="mw-headline" id="Delete_slice_range_.28WIP.29">Delete slice range (WIP)</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Stacks &gt; Delete slice range</i>
</p><p>Removes a range of slices from a stack, so that cropping in the Z direction is practical.
</p>
<h4><span class="mw-headline" id="Suitable_images_7">Suitable images</span></h4>
<p>A 3D image.
</p>
<h2><span class="mw-headline" id="Ellipsoid_factor">Ellipsoid factor</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Ellipsoid factor</i>.
</p><p>Ellipsoid Factor is a new method for measuring rod/plate geometry. It uses the axis lengths from prolate, oblate and intermediate elipsoids to determine how prolate or oblate the trabecular space is at a particular point. Highly prolate (javelin-shaped, rod-like) ellipsoids have a single long axis (<img class="mwe-math-fallback-image-inline tex" alt="c" src="_images/math/4/a/8/4a8a08f09d37b73795649038408b5f33.png" />) and two short axes (<img class="mwe-math-fallback-image-inline tex" alt="a, b" src="_images/math/b/3/4/b345e1dc09f20fdefdea469f09167892.png" />) such that <img class="mwe-math-fallback-image-inline tex" alt="a &lt; b \ll c" src="_images/math/1/f/e/1feb55f89bc2a13ca3f851d84a8ba65f.png" /> , whereas highly oblate (discus-shaped, plate-like) ellipsoids have two longer axes (<img class="mwe-math-fallback-image-inline tex" alt="b, c" src="_images/math/7/4/a/74aefa13d6ab8e4bfbd241583749dfe8.png" />) and one much shorter axis (<img class="mwe-math-fallback-image-inline tex" alt="a" src="_images/math/0/c/c/0cc175b9c0f1b6a831c399e269772661.png" />) so that <img class="mwe-math-fallback-image-inline tex" alt="a \ll b &lt; c" src="_images/math/d/b/3/db3b74b5ed00d720a8ece426eb8c8729.png" />. Calculating <img class="mwe-math-fallback-image-inline tex" alt="EF" src="_images/math/2/c/9/2c9b682412689d6723e3b31653b5774c.png" /> as the difference in ratios, <img class="mwe-math-fallback-image-inline tex" alt="EF = a/b - b/c" src="_images/math/8/b/f/8bf18c1bbae4cb2789670556c5c172a4.png" /> leads to a useful scale ranging from <img class="mwe-math-fallback-image-inline tex" alt="-1" src="_images/math/6/b/b/6bb61e3b7bce0931da574d19d1d82c88.png" /> (oblate, <img class="mwe-math-fallback-image-inline tex" alt="a/b \approx 0, b/c \approx 1" src="_images/math/5/a/9/5a92b4e66ac372b8e628260d9672f3fe.png" />) to <img class="mwe-math-fallback-image-inline tex" alt="+1" src="_images/math/7/0/1/7015440590904d38e35615ff0672d21c.png" /> (prolate, <img class="mwe-math-fallback-image-inline tex" alt="a/b \approx 1, b/c \approx 0" src="_images/math/3/c/b/3cb71b7a85f9a80d49d8ea50503dda84.png" />). <img class="mwe-math-fallback-image-inline tex" alt="EF" src="_images/math/2/c/9/2c9b682412689d6723e3b31653b5774c.png" /> of <img class="mwe-math-fallback-image-inline tex" alt="0" src="_images/math/c/f/c/cfcd208495d565ef66e7dff9f98764da.png" /> indicates an intermediate ellipsoid where <img class="mwe-math-fallback-image-inline tex" alt="a/b = b/c" src="_images/math/1/d/c/1dcf17ee744e23949b646d1cc93068a2.png" />, which is the case for spheres (<img class="mwe-math-fallback-image-inline tex" alt="a = b = c" src="_images/math/0/6/7/067e8173226f0047ad0e439500259b2f.png" />) and other ellipsoids with axis ratios <img class="mwe-math-fallback-image-inline tex" alt="a:qa:q^{2}a" src="_images/math/1/4/e/14e2a36fe3eabd1c8a14a8835d2256c9.png" />. Ellipsoid Factor runs <a href="Skeletonize3D" title="Skeletonize3D">Skeletonize3D</a> to get the medial axis of the trabeculae, which is used as the seed for sampling. Ellipsoids are seeded from each voxel on the medial axis. A combination of dilation, contraction, rotation and a small amount of translation is run iteratively until the ellipsoid increases no further in volume.
</p><p>The EF at a point in the structure is determined as the EF of the most voluminous ellipsoid which contains that point.
</p><p>If you use Ellipsoid Factor in your work, please cite:
Doube M (2015), <i>The Ellipsoid Factor for quantification of rods, plates and intermediate forms in 3D geometries</i>, Frontiers in Endocrinology, 6:15, <a rel="nofollow" class="external text" href="http://journal.frontiersin.org/Journal/10.3389/fendo.2015.00015/abstract">doi: 10.3389/fendo.2015.00015</a>
</p>
<h4><span class="mw-headline" id="Suitable_images_8">Suitable images</span></h4>
<p>A binary 3D image.
</p>
<h4><span class="mw-headline" id="Parameters_2">Parameters</span></h4>
<ul><li> <b>Sampling increment</b>: distance between sample points on each vector; should be less than the pixel spacing.</li>
<li> <b>Vectors</b>: number of vectors to sample at each seed point.</li>
<li> <b>Skeleton points per ellipsoid</b>: allows dense or sparse sampling, a value of <img class="mwe-math-fallback-image-inline tex" alt="1" src="_images/math/c/4/c/c4ca4238a0b923820dcc509a6f75849b.png" /> means that an ellipsoid is sampled at every seed point.</li>
<li> <b>Contact sensitivity</b>: how many vectors must touch the background before dilation stops.</li>
<li> <b>Maximum iterations</b>: how hard to try to find larger ellipsoids - fitting will stop if no improvement has been made after this number of iterations..</li>
<li> <b>Maximum drift</b>: how far the centroid may be displaced from its seed point.</li>
<li> <b>EF image</b>: stack containing EF values for each point contained by at least one ellipsoid and NaN elsewhere.</li>
<li> <b>Ellipsoid ID image</b>: stack containing the ID of the biggest ellipsoid at each point, ranked in descending order (<img class="mwe-math-fallback-image-inline tex" alt="0" src="_images/math/c/f/c/cfcd208495d565ef66e7dff9f98764da.png" /> is the largest ellipsoid).</li>
<li> <b>Volume image</b>: image showing the volume of the largest ellipsoid containing that point.</li>
<li> <b>Axis ratio images</b>: images showing <img class="mwe-math-fallback-image-inline tex" alt="a/b" src="_images/math/a/7/e/a7e86136543b019d72468ceebf71fb8e.png" /> and <img class="mwe-math-fallback-image-inline tex" alt="b/c" src="_images/math/2/4/e/24e710dda50ffc052a3b8714ea4174bc.png" /> ratios foreach point containing at least one ellipsoid and NaN elsewhere.</li>
<li> <b>Flinn peak plot</b>: plot of <img class="mwe-math-fallback-image-inline tex" alt="a/b" src="_images/math/a/7/e/a7e86136543b019d72468ceebf71fb8e.png" /> vs <img class="mwe-math-fallback-image-inline tex" alt="b/c" src="_images/math/2/4/e/24e710dda50ffc052a3b8714ea4174bc.png" /> weighted by volume, so bright pixels indicate relatively more of the structure has that axis ratio.</li>
<li> <b>Gaussian sigma:</b> amount to blur the Flinn peak plot - set to <img class="mwe-math-fallback-image-inline tex" alt="0" src="_images/math/c/f/c/cfcd208495d565ef66e7dff9f98764da.png" /> for a precise but less 'beautiful' result.</li>
<li> <b>Flinn plot</b>: unweighted Flinn plot - every ellipsoid is represented by the same sized point regardless of ellipsoid size.</li></ul>
<h4><span class="mw-headline" id="Results_4">Results</span></h4>
<ul><li> <b>EF image</b>: stack containing EF values. NaN (not a number) values are used in the background. Summary statistics can be obtained by running Analyze &gt; Histogram</li>
<li> <b>Short-Mid image</b>: stack containing the <img class="mwe-math-fallback-image-inline tex" alt="a/b" src="_images/math/a/7/e/a7e86136543b019d72468ceebf71fb8e.png" /> ratios</li>
<li> <b>Mid-Long image</b>: stack contining the <img class="mwe-math-fallback-image-inline tex" alt="b/c" src="_images/math/2/4/e/24e710dda50ffc052a3b8714ea4174bc.png" /> ratios</li>
<li> <b>Volume image</b>: stack containing ellipsoid volumes</li>
<li> <b>Max id image</b>: stack containing the ID of the largest ellipsoid at each point; IDs relate to the sort order based on volume, so ID = 0 is the largest ellipsoid. <img class="mwe-math-fallback-image-inline tex" alt="-1" src="_images/math/6/b/b/6bb61e3b7bce0931da574d19d1d82c88.png" /> is foreground and background is labelled with a large negative number.</li>
<li> <b>Flinn diagram</b>: plot of <img class="mwe-math-fallback-image-inline tex" alt="a/b" src="_images/math/a/7/e/a7e86136543b019d72468ceebf71fb8e.png" /> versus <img class="mwe-math-fallback-image-inline tex" alt="b/c" src="_images/math/2/4/e/24e710dda50ffc052a3b8714ea4174bc.png" /> values present in the volume</li>
<li> <b>Weighted Flinn plot</b>: Flinn diagram with peaks of intensity proportional to volume occupied by each (<img class="mwe-math-fallback-image-inline tex" alt="a/b" src="_images/math/a/7/e/a7e86136543b019d72468ceebf71fb8e.png" />, <img class="mwe-math-fallback-image-inline tex" alt="b/c" src="_images/math/2/4/e/24e710dda50ffc052a3b8714ea4174bc.png" />) ratio</li></ul>
<h4><span class="mw-headline" id="Related_publications_3">Related publications</span></h4>
<p>Salmon PL, Ohlsson C, Shefelbine SJ, Doube M (2015), <i>Structure model index does not measure rods and plates in trabecular bone</i>, Frontiers in Endocrinology, 6:162, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.3389/fendo.2015.00162">doi:10.3389/fendo.2015.00162</a>.
</p>
<h2><span class="mw-headline" id="Fit_ellipsoid">Fit ellipsoid</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Fit ellipsoid</i>.
</p><p>Finds the ellipsoid that best fits a set of point or multi-point ROIs in the ROI Manager. <i>Fit ellipsoid</i> may fail to fit an ellipsoid. The more points you add, the more likely it is to succeed. Points are scaled to the spatial calibration (voxel widht, height &amp; depth) of the input image.
</p>
<h4><span class="mw-headline" id="Suitable_images_9">Suitable images</span></h4>
<p>A 3D image.
</p>
<h4><span class="mw-headline" id="Parameters_3">Parameters</span></h4>
<ul><li> <b>ROI Manager</b>: ROIs with at least nine points in the ROI Manager.</li></ul>
<h4><span class="mw-headline" id="Results_5">Results</span></h4>
<ul><li> <b>Radii</b>: the radii <i>a</i>, <i>b</i> and <i>c</i> of the fitted ellipsoid. Radius <i>a</i> is the shortest and <i>c</i> the longest.</li>
<li> <b>Centroid</b>: <i>x</i>, <i>y</i> and <i>z</i> coordinates of the ellipsoid centre point.</li></ul>
<h4><span class="mw-headline" id="Differences_from_BoneJ1">Differences from BoneJ1</span></h4>
<ul><li> Supports multi-point ROIs.</li>
<li> Plug-in cancels if an ellipsoid can't be found, instead of reporting invalid results.</li></ul>
<h2><span class="mw-headline" id="Fit_sphere_.28WIP.29">Fit sphere (WIP)</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Fit sphere</i>.
</p><p>Finds the sphere that best fits a set of point ROIs, and optionally displays the image data bounded by the sphere in a new image window. Place a set of point ROIs on structures of interest, hitting [T] to add each point to the ROI Manager. Fit Sphere takes the coordinates of the points from the ROI Manager and applies a least-squares optimisation.
</p>
<h4><span class="mw-headline" id="Suitable_images_10">Suitable images</span></h4>
<p>A 3D image.
</p>
<h4><span class="mw-headline" id="Parameters_4">Parameters</span></h4>
<ul><li> <b>ROI Manager</b>: populated with at least 5 point ROI's</li>
<li> <b>Copy Sphere</b>: Create a new stack containing image data from within the best-fit sphere.</li>
<li> <b>Padding</b>: Number of black pixels to put between the sphere and the sides of the image</li>
<li> <b>Inner Cube</b>: Create a new stack containing image data from the cube that just fits inside the best-fit sphere.</li>
<li> <b>Outer Cube</b>: Create a new stack containing image data from the cube that the best-fit sphere just fits inside.</li>
<li> <b>Crop Factor</b>: Radius used for generating new images is multiplied by crop factor so that a bigger or smaller volume can be produced.</li>
<li> <b>Add to ROI Manager</b>: Add the sphere to the ROI Manager as a set of circular ROIs</li>
<li> <b>Clear ROI Manager</b>: Clear any existing ROIs in the Manager prior to adding circles</li></ul>
<h4><span class="mw-headline" id="Results_6">Results</span></h4>
<ul><li> <b>X Centroid</b>: x-coordinate of sphere's centroid</li>
<li> <b>Y Centroid</b>: y-coordinate of sphere's centroid</li>
<li> <b>Z Centroid</b>: z-coordinate of sphere's centroid</li>
<li> <b>Radius</b>: length of radius</li>
<li> <b>Images (optional)</b>: images containing a copy of the original data within the sphere, or within cubes bounding or bounded by the sphere.</li></ul>
<h2><span class="mw-headline" id="Fractal_dimension">Fractal dimension</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Fractal dimension</i>.
</p><p>This plug-in estimates the fractal dimension of an image by applying the box-counting algorithm. In this algorithm grids of diminishing size are scanned over the image, and the number of boxes containing at least one foreground voxel is counted. As the box size decreases and the grid becomes finer, the proportion of foreground boxes increases in a fractal structure. See <a rel="nofollow" class="external text" href="https://en.wikipedia.org/wiki/Box_counting">Wikipedia</a> for further details. BoneJ uses a fixed-grid scan, with an option to try to find the optimal covering.
</p><p>The box counting algorithm produces a pair of <img class="mwe-math-fallback-image-inline tex" alt="(log(n), -log(m))" src="_images/math/a/f/8/af8ebab307fdb5726339abe19939ad62.png" /> values for each iteration it runs. Here n = number of boxes with foreground, and m = box size. These pairs are passed to a curve-fitting algorithm, which returns the slope of the linear function which describes them (<a rel="nofollow" class="external text" href="https://en.wikipedia.org/wiki/Linear_regression">regression fit</a>). The coefficient of this slope is the fractal dimension.
</p><p>Fractal dimension is markedly influenced by the parameters selected for the box counting algorithm, so it's worth running it several times with different values to find an accurate measure for your image.
</p>
<h4><span class="mw-headline" id="Suitable_images_11">Suitable images</span></h4>
<p>A 2D or 3D binary image.
</p>
<h4><span class="mw-headline" id="Parameters_5">Parameters</span></h4>
<ul><li> <i>Starting box size (px)</i>: the size (sides) in pixels of a box in the sampling grid in the first iteration of the algorithm.</li>
<li> <i>Smallest box size (px)</i>: the minimum size in pixels of a box in the grid. When box size becomes smaller than this limit, the algorithm iterates one more time, and then stops.</li>
<li> <i>Box scaling factor</i>: the term used to divide the box size after iteration. For example, a scaling factor of <img class="mwe-math-fallback-image-inline tex" alt="2.0" src="_images/math/d/1/b/d1bd83a33f1a841ab7fda32449746cc4.png" /> halves the size at each step.</li>
<li> <i>Grid translations</i>: how many times at each iteration the grid is moved to try to the find the optimal covering. The optimal covering covers the objects in the image with the least amount of boxes. The larger the parameter the more likely it is that optimal covering is found. However, this slows down the plug-in considerably.</li>
<li> <i>Automatic parameters</i>: if checked, then the plug-in runs with default parameters.</li>
<li> <i>Show points</i>: if checked, then the plug-in displays the <img class="mwe-math-fallback-image-inline tex" alt="(log(n), -log(m))" src="_images/math/a/f/8/af8ebab307fdb5726339abe19939ad62.png" /> values it calculated.    </li></ul>
<h4><span class="mw-headline" id="Results_7">Results</span></h4>
<ul><li> <i>Fractal dimension</i>: the <a rel="nofollow" class="external text" href="https://en.wikipedia.org/wiki/Fractal_dimension">fractal dimension</a> of the image. For example, the Koch snow flake has a fractal dimension of 1.262.</li>
<li> <i>R²</i>: the <a rel="nofollow" class="external text" href="https://en.wikipedia.org/wiki/Coefficient_of_determination">coefficient of determination</a> of the line fitted to the <img class="mwe-math-fallback-image-inline tex" alt="(log(n), -log(m))" src="_images/math/a/f/8/af8ebab307fdb5726339abe19939ad62.png" /> points.</li></ul>
<p>The measures are reported separately for each 3D subspace in the image, i.e. for each channel and time frame.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_5">Differences to BoneJ1</span></h4>
<ul><li> Supports hyperstacks</li>
<li> Algorithm parameters can be changed by the user.</li></ul>
<h4><span class="mw-headline" id="Related_publications_4">Related publications</span></h4>
<p>Fazzalari NL, Parkinson IH (1996), <i>Fractal dimension and architecture of trabecular bone</i>, J Pathol, 178: 100-5, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1002/(SICI)1096-9896(199601)178:1%3C100::AID-PATH429%3E3.0.CO;2-K">doi:10.1002/(SICI)1096-9896(199601)178:1&lt;100::AID-PATH429&gt;3.0.CO;2-K</a>.
</p>
<h2><span class="mw-headline" id="Inter-trabecular_angles">Inter-trabecular angles</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Inter-trabecular Angles</i>.
</p><p>The plug-in was designed to analyse the angles between trabeculae of cancellous bone. First it calls <a href="Skeletonize3D" title="Skeletonize3D">Skeletonize3D</a> to thin the input image (if necessary). Then it calls <a href="AnalyzeSkeleton" title="AnalyzeSkeleton">AnalyzeSkeleton</a>, which creates a graph of the largest skeleton (by number of nodes) in the thinned image. The graph consists of nodes and the edges that connect them. Nodes are also known as vertices, and edges as branches. Roughly speaking the edges correspond to trabeculae and the nodes to the junction points, where trabeuculae meet.
</p><p>The graph is often not a perfect representation of the trabecular network in the input image. <i>Inter-trabecular angles</i> offers many options to adjust the graph's topology and filter out artefacts that may obfuscate or skew the results. First it allows you to filter out nodes with too many or too few edges. Secondly it can be used to prune very short edges, which often do not represent actual trabeculae. 
</p>
<div class="floatleft"><a href="File:Ita_types.png" class="image" title="Types of edges in pruning"><img alt="Types of edges in pruning" src="_images/c/c2/Ita_types.png" width="150" height="264" /></a></div>
<p>Pruning works differently for different types of edges. There are four kinds: outer, dead-end, inner and short. An outer edge doesn't interconnect different parts of a graph. In other words, one (and only one) of its endpoints connects to only one branch, i.e. the edge itself. In the figure the outer edge is colored black. A dead-end (blue) is an outer edge whose length is less than the minimum set by the user, i.e. it's a short, outer edge. An inner edge (green) connects to end points with more than one branch. A short edge is an inner edge, whose length is less than the set minimum. When a dead-end is pruned, it with its "lonely" end-point are removed from the graph. When a short edge is pruned, it and it's endpoints are removed, and a new node is placed at the midpoint of the former. This new node connects to all the branches the previous nodes connected to.
</p><p><i>Inter-trabecular angles</i> offers two further options to control pruning: <i>iteration</i> and <i>clustering</i>. Iteration repeats pruning until no new short edges are found. Sometimes pruning can create new short edges, and thus the graph may still have them after one iteration. However, iteration can alter the structure of the graph too dramatically. Clustering searches for all nodes connected by short edges, before removing any. In the figure, clustering pruning would remove the four nodes and the red edges between them in one go. It would create a new node at the center of the previous four, and connect it to the blue and green edges. When pruning is not clustering, it removes edges one-by-one. This changes the end result depending on the order the edges are traversed. Clustering creates the same result each time. 
</p><p>Pruning also removes loops and redundant parallel edges. The former connects to the same node on both ends, and the latter connects two nodes that are already connected by another edge. 
</p><p>The pruning and filtering features were added so that we can replicate and continue from the research of <a rel="nofollow" class="external text" href="https://doi.org/10.1016/j.actbio.2016.08.040">Reznikov et al.</a>
</p>
<h4><span class="mw-headline" id="Suitable_images_12">Suitable images</span></h4>
<p>An 8-bit, binary 2D or 3D image. Hyperstacks are not supported.
</p>
<h4><span class="mw-headline" id="Parameters_6">Parameters</span></h4>
<ul><li><b>Minimum valence</b>: minimum number of branches for a node to be included in the analysis.</li>
<li><b>Maximum valence</b>: maximum number of branches for a node to be included in the analysis.</li>
<li><b>Minimum trabecular length (px)</b>: minimum length for a branch to be preserved in the pruning. Length is calculated as the distance between the endpoints of a branch.
<ul><li> This length is also displayed in the units of the image calibration</li></ul></li>
<li><b>Margin (px)</b>: minimum distance of a node from image stack edges to be included in the analysis. Having nodes too close to the edges can make the results less accurate, because you get more branches that do not terminate to a node at the other end.</li>
<li><b>Iterate pruning</b>: repeat pruning until no more new short branches are discovered. When true, the topology of the graph is likely to change more in the pruning.</li>
<li><b>Use clusters</b>: if true then the pruning result is independent of the order in which the graph is traversed. False corresponds with methodology in Reznikov's article. See above for more details.</li>
<li><b>Print centroids</b>: Print the centroids (center coordinates) of the node pairs at the ends of each edge.</li>
<li><b>Print&#160;% culled edges</b>: Print statistics of different types of edges pruned.</li></ul>
<h4><span class="mw-headline" id="Results_8">Results</span></h4>
<ul><li><b>Inter-trabecular angles</b>: angles (in radians) between each branch of each node included in the analysis. Sorted into columns according to the number of branches per node. For example, column <i>"3"</i> shows the angles between branches of nodes with three branches.</li>
<li><b>Centroids</b> (optional): A table of the center coordinates of the node pairs at the ends of each edge.</li>
<li><b>Culled edge percentages</b> (optional): A table showing the percentages of different kinds of edges pruned, compared to the total number of edges in the analysed graph.</li>
<li><b>Skeleton</b> (optional): if the plug-in had to skeletonise the input image, it displays the result of the thinning.</li></ul>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_6">Differences to BoneJ1</span></h4>
<ul><li> Inter-trabecular angles generalizes the idea of <i>Triple point angles</i> for any types of points. It also provides more tools for adjusting the graph for analysis.</li></ul>
<h4><span class="mw-headline" id="Related_publications_5">Related publications</span></h4>
<p>Reznikov, N et al. (2016), <i>Inter-trabecular angle: A parameter of trabecular bone architecture in the human proximal femur that reveals underlying topological motifs</i>, Acta Biomaterialia, 44: 65--72, <a rel="nofollow" class="external text" href="https://doi.org/10.1016/j.actbio.2016.08.040">doi:j.actbio.2016.08.040</a>.
</p>
<h2><span class="mw-headline" id="Thickness">Thickness</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Thickness</i>.
</p><p>This plug-in includes <a href="Local_Thickness" title="Local Thickness">Local_Thickness</a> in BoneJ, and provides some additional options &amp; results. Local thickness measures <i>the diameter of the largest sphere that fits inside the object and contains the point</i> for each point i.e. foreground voxel in an image. The plug-in calculates mean and standard deviation of the trabecular thickness (Tb.Th) or trabecular spacing (Tb.Sp) directly from pixel values in the resulting thickness map. Foreground voxels are considered trabeculae, and background voxels are the spacing. Processing time is heavily dependent on feature size (in pixels); large features can take a very long time to process.
</p>
<h4><span class="mw-headline" id="Suitable_images_13">Suitable images</span></h4>
<p>The input image must be 3D, 8-bit and binary. Hyperstacks are not supported.
</p>
<h4><span class="mw-headline" id="Parameters_7">Parameters</span></h4>
<ul><li> <b>Calculate</b>: chooses which thickness maps to calculate - trabecular thickness, trabecular spacing, or both. In order to calculate trabecular spacing, the image voxels are inverted.</li>
<li> <b>Show thickness maps</b>: display the calculated thickness maps or not.</li>
<li> <b>Mask thickness maps</b>: remove artifacts from the thickness maps. Artifacts are foreground voxels not present in the original image. Resets to true for each run, because unmasked pixels bias the result.</li></ul>
<h4><span class="mw-headline" id="Results_9">Results</span></h4>
<ul><li> The mean and standard deviation for each thickness map calculated.</li>
<li> Displays thickness map images if <i>Show thickness maps</i> was selected. </li></ul>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_7">Differences to BoneJ1</span></h4>
<ul><li> Calls the latest version of <a href="Local_Thickness" title="Local Thickness">Local_Thickness</a>.</li></ul>
<h4><span class="mw-headline" id="Related_publications_6">Related publications</span></h4>
<ul><li> Dougherty R, Kunzelmann K (2007), <i>Computing local thickness of 3D structures with ImageJ</i>, Microsc. Microanal., 13: 1678-1679, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1017/S1431927607074430">doi:10.1017/S1431927607074430</a></li>
<li> Hildebrand T, Rüegsegger P (1997), <i>A new method for the model-independent assessment of thickness in three-dimensional images</i>, J. Microsc., 185: 67-75, <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1046/j.1365-2818.1997.1340694.x">doi:10.1046/j.1365-2818.1997.1340694.x</a></li></ul>
<h2><span class="mw-headline" id="Moments_of_inertia">Moments of inertia</span></h2>
<p>Menu path: <i>Plugins &gt; BoneJ &gt; Moments of Inertia</i>
</p><p>Calculates the three orthogonal principal axes and moments of inertia around those axes. It includes pixels with values between upper and lower limits, which can be defined in terms of unitless grey values or Hounsfield units (HU). It optionally creates a new stack with the image centred and rotated so that the principal axes are parallel to the image stack's x, y and z axes. Calculations are limited to a rectangular ROI if one is drawn. The plugin will guess whether the image is HU calibrated, and if so, apply HU limits of bone (0-4000 HU), otherwise it will calculate auto-thresholds based on the stack's histogram. If a calibration curve is known, the coefficients can be added to get weighted calculations. Aligning a bone with Moments of Inertia may be a useful step prior to Slice Geometry if bones are not aligned with the image z axis.
</p>
<h4><span class="mw-headline" id="Suitable_images_14">Suitable images</span></h4>
<p>An 8-bit or 16-bit image.
</p>
<h4><span class="mw-headline" id="Parameters_8">Parameters</span></h4>
<ul><li> <b>Start slice</b>: First slice to include in calculations</li>
<li> <b>End slice</b>: Last slice to include in calculations</li>
<li> <b>HU Calibrated</b>: Plugin will guess whether the image is HU calibrated. If image is HU calibrated, make sure the box is checked and enter HU calibrated numeric values in the following fields</li>
<li> <b>Bone Min</b>: Lower threshold, in either uncalibrated greys or HU</li>
<li> <b>Bone Max</b>: Upper threshold, in either uncalibrated greys or HU</li>
<li> <b>Slope</b>: m where physical density = m × pixel value + c</li>
<li> <b>Y Intercept</b>: c where physical density = m × pixel value + c</li>
<li> <b>Align result</b>: Draw a new stack with the principal axes parallel to the image axes</li>
<li> <b>Show axes (2D)</b>: Draw the axes in white on the aligned image</li>
<li> <b>Show axes (3D)</b>: Display the stack and its principal axes in a 3D Viewer window</li>
<li> <b>Record unit vectors</b>: Record the elements from the eigenvector matrix in the Result table; this is the orientation of the object's principal axes.</li></ul>
<h4><span class="mw-headline" id="Results_10">Results</span></h4>
<ul><li> <b>Image (optional)</b>: a copy of the input image centered and aligned with the principal axes</li>
<li> <b>Xc</b>: Centroid x-coordinate (mass-weighted by calibrated density)</li>
<li> <b>Yc</b>: Centroid y-coordinate</li>
<li> <b>Zc</b>: Centroid z-coordinate</li>
<li> <b>Vol</b>: Total volume of thresholded voxels</li>
<li> <b>Mass</b>: Mass of bone, from pixel values and density calibration coefficients</li>
<li> <b>Icxx</b>: Moment around x axis passing through centroid</li>
<li> <b>Icyy</b>: Moment around y axis passing through centroid</li>
<li> <b>Iczz</b>: Moment around z axis passing through centroid</li>
<li> <b>Icxy</b>: Off-axis term for constructing inertia tensor</li>
<li> <b>Icxz</b>: Off-axis term for constructing inertia tensor</li>
<li> <b>Icyz</b>: Off-axis term for constructing inertia tensor</li>
<li> <b>I1</b>: Moment around the shortest principal axis</li>
<li> <b>I2</b>: Moment around the middle principal axis</li>
<li> <b>I3</b>: Moment around the longest principal axis</li>
<li> <b>Optional verbose eigenvector elements</b>: the 3 unit vectors' elements sorted column first (i.e. vector0{x, y, z}, vector1{x, y, z}, vector2{x, y, z}) sorted in the <i>reverse</i> order of the eigenvalues I1, I2 and I3.</li>
<li> <b>Verbose output in Log window</b>
<ul><li> <b>Eigenvalues</b>: Result of of eigenvalue decomposition. Moments of inertia</li>
<li> <b>Eigenvectors of principal axes</b>: orientation of input image</li>
<li> <b>Inverse eigenvector matrix</b>: used to map voxel positions in the aligned image back to voxel positions in the original image</li></ul></li>
<li> Optional 3D display of principal axes</li></ul>
<h2><span class="mw-headline" id="Orientation">Orientation</span></h2>
<p>Menu path: <i>Plugins &gt; BoneJ &gt; Orientation</i>
</p><p>Sets the direction in a 2D image, without rotating the image or changing it in any way. <a href="BoneJ_experimental.html#Slice_geometry"><i>Slice Geometry</i></a> optionally uses <i>Orientation</i> to calculate diameter, second moments of area and section moduli around anatomic axes set by the user. To apply the selected orientation to a <i>Slice Geometry</i> analysis, run <i>Orientation</i>, select axes and direction, leave the <i>Orientation</i> window open, run <i>Slice Geometry</i>, and select <i>Use Orientation</i>.
</p><p>The option to use <i>Orientation</i> will be greyed out in <i>Slice Geometry</i> if there is no active <i>Orientation</i> window. Also, in the Results table you will see a list of results that relate to the particular axis pair you have chosen. <i>Orientation</i> can handle multiple open images and will handle different images’ orientation independently.
</p>
<h4><span class="mw-headline" id="Suitable_images_15">Suitable images</span></h4>
<p>A 2D image.
</p>
<h4><span class="mw-headline" id="Parameters_9">Parameters</span></h4>
<ul><li> <b>Orientation</b>: input field displays current orientation (clockwise from 12 o'clock) and takes keyboard input</li>
<li> <b>deg / rad</b>: display and set orientation in degrees or radians</li>
<li> <b>Principal direction</b>: the main direction, the head of which is displayed in red</li>
<li> <b>Secondary direction</b>: the orthogonal direction</li>
<li> <b>Reflect</b>: swap the labels on this axis</li></ul>
<h4><span class="mw-headline" id="Results_11">Results</span></h4>
<ul><li> <b>Orientation axes</b>: draws the axes of the orientation on the image.</li></ul>
<h2><span class="mw-headline" id="Particle_Analyser">Particle Analyser</span></h2>
<p>Menu path: <i>Plugins &gt; BoneJ &gt; Analyze &gt; Particle Analyser</i>
</p><p>Particle Analyser performs connected-components labelling. It started life as an adaptation of the 3D Object Counter, for identifying particles as part of Purify. Once it became obvious that the heavily recursive algorithm was very inefficient, especially on large stacks, particle labelling was rewritten to be multithreaded and much less recursive.
</p><p>A member of our group wished to study osteocyte lacunae in synchrotron microCT images, so the Particle Analyser was born from Purify's particle labelling algorithm. Other groups use Particle Analyser for soil analysis; it could be used for any porous media. Particle Analyser labels unique (unconnected) particles, then analyses each particle separately. Substantial use is made of other plugins in BoneJ, including Connectivity to get the Euler characteristic, Isosurface to get the surface area and Thickness to get the local thickness of individual particles.
</p><p>If you use it in your published work, please cite Doube M. 2020 Multithreaded two-pass connected components labelling and particle analysis in ImageJ. bioRxiv , 2020.02.28.969139. (doi:10.1101/2020.02.28.969139) <a rel="nofollow" class="external autonumber" href="https://doi.org/10.1101/2020.02.28.969139">[1]</a>
</p>
<h4><span class="mw-headline" id="Suitable_images_16">Suitable images</span></h4>
<p>A 2D or 3D binary image.
</p>
<h4><span class="mw-headline" id="Parameters_10">Parameters</span></h4>
<ul><li> <b>Exclude on sides</b>: don't analyse particles touching the stack sides</li>
<li> <b>Feret Diameter</b>: measure particles' Feret diameter (slow, brute force method).</li>
<li> <b>Moments of inertia</b>: measure each particle's moments of inertia</li>
<li> <b>Thickness</b>: Measure each particle's local thickness. Note: does not handle anisotropic voxels due to the underlying implementation of Local Thickness.</li>
<li> <b>Ellipsoids</b>: find the best-fit ellipsoid to the particle's surface mesh</li>
<li> <b>Skeletons</b>: Measure branch length and count branches using <a href="AnalyzeSkeleton" title="AnalyzeSkeleton">AnalyzeSkeleton</a></li>
<li> <b>Surface Area</b>: measure particles' surface area from surface mesh</li>
<li> <b>Enclosed volume</b>: measure volume enclosed by each particle's surface mesh</li>
<li> <b>Euler characteristic</b>: measure each particle's Euler characteristic and connectivity</li>
<li> <b>Mask thickness map</b>: don't count thickness pixels overhanging the foreground</li>
<li> <b>Record unit vectors</b>: log the unit vectors of the inertia tensor and best-fit ellipsoid's axes to the Results table</li>
<li> <b>Min Volume</b>: smallest particle to measure or display, in calibrated units</li>
<li> <b>Max Volume</b>: largest particle to measure or display, in calibrated units</li>
<li> <b>Surface resampling</b>: how much to resample the stack (in pixel units) while creating the surface mesh by marching cubes. Setting a large value smooths out small details and potentially eliminates very small particles. Setting a small value (minimum 1) leads to precise meshing over the pixel grid.</li>
<li> <b>Show particle stack</b>: display an image containing labelled particles</li>
<li> <b>Show thickness stack</b>: display the result of local thickness in a new stack</li>
<li> <b>Show centroids (3D)</b>: display particles' centroids in a 3D Viewer window</li>
<li> <b>Show ellipsoids (3D)</b>: display best-fit ellipsoids in a 3D Viewer window</li>
<li> <b>Show size stack</b>: display an image with particles labelled by their volume</li>
<li> <b>Show surfaces (3D)</b>: display particles' surface meshes in a 3D Viewer window</li>
<li> <b>Show axes (3D)</b>: display principal axes (from moments of inertia) and Feret axes, if they were selected to be calculated, in a 3D Viewer window</li>
<li> <b>Show stack (3D)</b>: display the original binary stack in a 3D Viewer window</li>
<li> <b>Surface colours</b>:
<ul><li> <b>Gradient</b>: Each particle gets a unique colour between red and green</li>
<li> <b>Split</b>: Each particle is coloured depending on whether it is over (red) or under (yellow) the Split value</li>
<li> <b>Orientation</b>: Each particle is coloured by its long-axis deflection from the image <i>z</i> axis</li></ul></li>
<li> <b>Split value</b>: Particles with volume greater than this value are coloured red, otherwise coloured yellow.</li>
<li> <b>Volume resampling</b>: amount of voxel resampling to apply when displaying the original stack</li></ul>
<h4><span class="mw-headline" id="Output">Output</span></h4>
<ul><li> <b>Results</b>
<ul><li> <b>Label</b>: Image file name</li>
<li> <b>ID</b>: unique particle identifier; this number is the label used for the particle in all calculations and output</li>
<li> <b>Vol</b>: particle volume, calculated as pixel count multiplied by pixel grid <i>x</i>, <i>y</i>, and <i>z</i> spacing if the image axes are calibrated</li>
<li> <b>x Cent</b>: x-coordinate of particle centroid</li>
<li> <b>y Cent</b>: y-coordinate of particle centroid</li>
<li> <b>z Cent</b>: z-coordinate of particle centroid</li>
<li> <b>SA</b>: surface area (0 if too small for mesh to be produced; see warning log)</li>
<li> <b>Feret</b>: maximum caliper diameter</li>
<li> <b>FeretAx</b>: x-coordinate of the first point of the maximum caliper diameter</li>
<li> <b>FeretAy</b>: y-coordinate of the first point of the maximum caliper diameter</li>
<li> <b>FeretAz</b>: z-coordinate of the first point of the maximum caliper diameter</li>
<li> <b>FeretBx</b>: x-coordinate of the second point of the maximum caliper diameter</li>
<li> <b>FeretBy</b>: y-coordinate of the second point of the maximum caliper diameter</li>
<li> <b>FeretBz</b>: z-coordinate of the second point of the maximum caliper diameter</li>
<li> <b>Encl. Vol</b>: Volume enclosed by surface mesh (0 if too small for mesh to be produced; see warning log)</li>
<li> <b>I1</b>: moment of inertia around shortest principal axis</li>
<li> <b>I2</b>: moment of inertia around middle principal axis</li>
<li> <b>I3</b>: moment of inertia around longest principal axis</li>
<li> <b>vX</b>: x component of unit vector of longest principal axis (a measure of orientation)</li>
<li> <b>vY</b>: y component of unit vector of longest principal axis</li>
<li> <b>vZ</b>: z component of unit vector of longest principal axis</li>
<li> <b>vX1 - vZ2</b>: xyz components of the unit vectors of the middle and short principal axes</li>
<li> '<i>n Branches</i> number of skeleton branches in the particle's medial axis</li>
<li> <b>Branches length</b> total length of the particle's medial axis</li>
<li> <b>Euler (χ)</b>: Euler characteristic of the particle</li>
<li> <b>Holes (β1)</b>: number of topological holes (handles) in the particle</li>
<li> <b>Cavities (β2)</b>: number of enclosed cavities in the particle</li>
<li> <b>Thickness</b>: mean local thickness of particle</li>
<li> <b>SD Thickness</b>: standard deviation of the mean local thickness of particle</li>
<li> <b>Max Thickness</b>: maximum local thickness of particle</li>
<li> <b>Major radius</b>: length of best-fit ellipsoid's long radius</li>
<li> <b>Int. radius</b>: length of best-fit ellipsoid's intermediate radius</li>
<li> <b>Minor radius</b>: length of best-fit ellipsoid's short radius</li>
<li> <b>V00 - V22</b>: elements of the 3×3 rotation matrix of the best-fit ellipsoid</li></ul></li>
<li> <b>Image Stacks</b>
<ul><li> <b>Thickness image</b> showing local thickness at each point in the input image</li>
<li> <b>Particle image</b> showing the particle ID at each point in the input image (individual particles can be selected by thresholding for that ID value)</li>
<li> <b>Size image</b> showing particles labelled by their volume; particles can be selected based on volume by thresholding this image</li></ul></li>
<li> <b>3D Display</b>
<ul><li> If any of the 3D display options are selected, an ImageJ 3D Viewer window is opened. All features are added individually, so that by hovering the mouse over a particle's surface, you can see its ID in the ImageJ status area. Results pertaining to that particle can then be looked up in the Results table. Particles can be manipulated individually, including transforming and adjusting colour or transparency.</li></ul></li></ul>
<h4><span class="mw-headline" id="Citation">Citation</span></h4>
<p>Doube M. 2020 Multithreaded two-pass connected components labelling and particle analysis in ImageJ. bioRxiv , 2020.02.28.969139. (doi:10.1101/2020.02.28.969139) <a rel="nofollow" class="external autonumber" href="https://doi.org/10.1101/2020.02.28.969139">[2]</a>
</p>
<h2><span class="mw-headline" id="Purify">Purify</span></h2>
<p>Menu path: <i>Plugins &gt; BoneJ &gt; Purify</i>
</p><p>Purify locates all particles in 3D and removes all but the largest foreground and background particles.
</p>
<h4><span class="mw-headline" id="Suitable_images_17">Suitable images</span></h4>
<p>The input image must be 3D, 8-bit and binary.
</p>
<h4><span class="mw-headline" id="Parameters_11">Parameters</span></h4>
<ul><li> <b>Performance Log</b>: Show verbose performance information to help tune your system</li>
<li> <b>Make copy</b>: If checked, shows the result in a new window; if unchecked the result replaces the original image.</li></ul>
<h4><span class="mw-headline" id="Results_12">Results</span></h4>
<ul><li> <b>Performance metrics (optional)</b>
<ul><li> <b>Threads</b>: Number of CPU cores used</li>
<li> <b>Slices</b>: Number of slices in the input image stack</li>
<li> <b>Chunks</b>: Number of chunks of slices, each chunk is processed independently</li>
<li> <b>Chunk size</b>: Number of slices per chunk</li>
<li> <b>Last chunk size</b>: size of the last chunk (the remainder chunk)</li>
<li> <b>Duration</b>: time in seconds to complete purification</li></ul></li>
<li> <b>Purified image</b>: optionally in a new image window.</li></ul>
<h2><span class="mw-headline" id="Skeletonise">Skeletonise</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt;Skeletonise</i>.
</p><p>This plug-in simply includes <a href="Skeletonize3D" title="Skeletonize3D">Skeletonize3D</a> in BoneJ. It adds some additional validation to check that your image suits the tool.
</p>
<h4><span class="mw-headline" id="Suitable_images_18">Suitable images</span></h4>
<p>The input image must be 2D or 3D, 8-bit and binary. Hyperstacks are not supported.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_8">Differences to BoneJ1</span></h4>
<p>Calls the latest version of <a href="Skeletonize3D" title="Skeletonize3D">Skeletonize3D</a>.
</p>
<h2><span class="mw-headline" id="Slice_geometry">Slice geometry</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Slice Geometry</i>
</p><p>Slice Geometry calculates cross-sectional geometric properties of shapes: cross-sectional area, centroid, mean density, second moment of area, section modulus, Feret diameter and local thickness (2D and 3D). Measurements can be limited to a rectangular ROI. If your bone is not well aligned with the image axes, you may find it useful to align the bone to its principal axes using Moments of Inertia or by exporting a transformed volume from the ImageJ 3D Viewer. Importantly, no assumption of geometry is made for any of the measurements.
</p>
<h4><span class="mw-headline" id="Suitable_images_19">Suitable images</span></h4>
<p>A 2D or 3D binary image.
</p>
<h4><span class="mw-headline" id="Parameters_12">Parameters</span></h4>
<ul><li> <b>Bone</b>: Slice Geometry will guess from your image title the bone it is working on. If it is wrong, correct it. If your bone of interest isn't listed, email me.</li>
<li> <b>2D Thickness</b>: Run Thickness on a per-slice basis; this fits circles rather than spheres</li>
<li> <b>3D Thickness</b>: Run Thickness on the whole stack, fitting spheres, then report results per slice</li>
<li> <b>Draw Axes</b>: Draw axes on an annotated copy of the stack</li>
<li> <b>Draw Centroids</b>: Draw centroids on an annotated copy of the stack</li>
<li> <b>Annotated Copy (2D)</b>: Create a new stack showing the centroids and principal axes</li>
<li> <b>3D Annotation</b>: Display the stack, principal axes and centroids in the 3D viewer</li>
<li> <b>Process Stack</b>: Calculate parameters for all slices in the stack</li>
<li> <b>Clear Results</b>: Remove all data in the results table without saving, prior to calculating parameters</li>
<li> <b>Use Orientation</b>: Also calculate parameters based on directions defined in <a href="BoneJ_experimental.html#Orientation">Orientation</a></li>
<li> <b>HU Calibrated</b>: Allows you to enter your thresholds in Hounsfield units (HU) or uncalibrated units</li>
<li> <b>Bone Min</b>: minimum pixel value to use in calculations</li>
<li> <b>Bone Max</b>: maximum pixel value to use in calculations</li>
<li> <b>Slope</b>: <img class="mwe-math-fallback-image-inline tex" alt="m" src="_images/math/6/f/8/6f8f57715090da2632453988d9a1501b.png" /> where physical density <img class="mwe-math-fallback-image-inline tex" alt="= m * pixel value + c" src="_images/math/1/b/c/1bc5c0db80fa5b8b5998318628d827d4.png" /></li>
<li> <b>Y Intercept</b>: <img class="mwe-math-fallback-image-inline tex" alt="c" src="_images/math/4/a/8/4a8a08f09d37b73795649038408b5f33.png" /> where physical density <img class="mwe-math-fallback-image-inline tex" alt="= m * pixel value + c" src="_images/math/1/b/c/1bc5c0db80fa5b8b5998318628d827d4.png" /></li>
<li> <b>Note</b>: density-weighted calculations are only applied to centroid determination at present</li>
<li> <b>Partial volume compensation</b>: Use model that assumes Gaussian blur of imaging modality and linear transform between pixel value and sample 'density' to correct for blurred and pixelated images (e.g. small bone in clinical CT)</li>
<li> <b>Background</b>: pixel value that corresponds to zero bone density (could be the 'fat', 'air' or 'muscle' pixel value)</li>
<li> <b>Foreground</b>: pixel value that corresponds to 100% bone density</li>
<li> <b>A rectangular ROI (optional)</b>: if there's a ROI, calculations are limited to its area.</li></ul>
<h4><span class="mw-headline" id="Results_13">Results</span></h4>
<ul><li> <b>Images (optional)</b>: displays the result images selected in the parameters</li>
<li> <b>Bone code</b>: Unique numeric identifier for the anatomic name of each bone. Further bone codes can be added to BoneJ on request</li>
<li> <b>Slice</b>: slice number indicating which slice contributed image data for this row of the results</li>
<li> <b>CSA</b>: cross-sectional area</li>
<li> <b>X cent.</b>: Centroid x-coordinate</li>
<li> <b>Y cent.</b>: Centroid y-coordinate</li>
<li> <b>Density</b>: mean physical density, calculated from pixel values and calibration coefficients</li>
<li> <b>wX cent</b>: Density-weighted x-coordinate of centroid</li>
<li> <b>wY cent</b>: Density-weighted y-coordinate of centroid</li>
<li> <b>Theta</b>: angle of minor axis (axis for Imin, the long axis of your specimen's cross section) from the horizontal, ranging from <img class="mwe-math-fallback-image-inline tex" alt="-\pi/2" src="_images/math/7/2/e/72e954820cf96e8e14449e653db9862a.png" /> to <img class="mwe-math-fallback-image-inline tex" alt="\pi/2" src="_images/math/1/f/8/1f8b57e932cf2d585ed71af904be8638.png" />, with positive as clockwise</li>
<li> <b>R1</b>: maximum chord length from minor axis</li>
<li> <b>R2</b>: maximum chord length from major axis</li>
<li> <b>I<sub>min</sub></b>: Second moment of area around major axis</li>
<li> <b>I<sub>max</sub></b>: Second moment of area around minor axis</li>
<li> <b>I<sub>pm</sub></b>: Product moment of area (= 0 if no errors are present, e.g. due to pixelation)</li>
<li> <b>Z<sub>max</sub></b>: Section modulus around major axis (<i>I</i><sub>max</sub> / R2)</li>
<li> <b>Z<sub>min</sub></b>: Section modulus around minor axis (<i>I</i><sub>min</sub> / R1)</li>
<li> <b>Z<sub>pol</sub></b>: Polar section modulus</li>
<li> <b>Feret Min</b>: Minimum caliper width</li>
<li> <b>Feret Max</b>: Maximum caliper width</li>
<li> <b>Feret Angle</b>: Orientation of maximum caliper width</li>
<li> <b>Perimeter</b>: Distance around external surface</li>
<li> <b>Max Thick 2D</b>: Maximum thickness determined by local thickness in 2D</li>
<li> <b>Mean Thick 2D</b>: Mean thickness determined by local thickness in 2D</li>
<li> <b>SD Thick 2D</b>: Standard deviation of the mean thickness determined by local thickness in 2D</li>
<li> <b>Max Thick 3D</b>: Maximum thickness in this slice determined by local thickness in 3D</li>
<li> <b>Mean Thick 3D</b>: Mean thickness in this slice determined by local thickness in 3D</li>
<li> <b>SD Thick 3D</b>: Standard deviation of the mean thickness in this slice determined by local thickness in 3D Directional measurements, using Orientation directions, and the specimen's slice centroid</li>
<li> <b>A (rad)</b>: Principal orientation (direction A)</li>
<li> <b>B (rad)</b>: Secondary orientation (direction B)</li>
<li> <b>IAa</b>: Second moment of area around Aa axis</li>
<li> <b>IBb</b>: Second moment of area around Bb axis</li>
<li> <b>ZAa</b>: Section modulus around Aa axis</li>
<li> <b>ZBb</b>: Section modulus around Bb axis</li>
<li> <b>RAa</b>: maximum chord length from Aa axis</li>
<li> <b>RBb</b>: maximum chord length from Bb axis</li>
<li> <b>DAa</b>: Calliper diameter in direction of Aa</li>
<li> <b>DBb</b>: Calliper diameter in direction of Bb</li></ul>
<h2><span class="mw-headline" id="Surface_area">Surface area</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Surface area</i>
</p><p><i>Surface area</i> creates a mesh from the bone in the image, and then calculates the area of the surface of the mesh. A mesh is a collection of triangular faces that defines the shape of an object in 3D graphics. The plug-in assumes that all foreground voxels represent bone.
</p>
<h4><span class="mw-headline" id="Suitable_images_20">Suitable images</span></h4>
<p>A 3D binary image.
</p>
<h4><span class="mw-headline" id="Parameters_13">Parameters</span></h4>
<ul><li> <b>Export STL file(s)</b>: if checked, then the meshes created from the image are saved as .stl-files. A dialog opens for you to select a folder for the files.</li></ul>
<h4><span class="mw-headline" id="Results_14">Results</span></h4>
<ul><li> <b>Surface area</b>: the surface area of the bone mesh</li>
<li> <b>STL-file</b>: the mesh created from the bone image</li></ul>
<p>The surface area is reported and an .stl-file saved separately for each 3D subspace in the image. 
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_9">Differences to BoneJ1</span></h4>
<ul><li> Supports hyperstacks.</li>
<li> Results differ, because the marching cubes and mesh volume implementations are different.</li>
<li> In BoneJ1 this plug-in was called <i>Isosurface</i></li></ul>
<h2><span class="mw-headline" id="Surface_fraction">Surface fraction</span></h2>
<p>Menu path <i>Plugins &gt; BoneJ &gt; Fraction &gt; Surface fraction</i>
</p><p><i>Surface fraction</i> calculates the fraction of bone volume in an image by comparing meshes created from bone particles and the whole image. A mesh is a collection of triangular faces that defines the shape of an object in 3D graphics. The plug-in assumes that all foreground voxels represent bone. 
</p><p>More formally defined, <i>Surface fraction</i> calculates the fraction <i>BV/TV</i>, which is the volume of mineralised bone <i>BV</i> per unit volume of the sample <i>TV</i>.
</p>
<h4><span class="mw-headline" id="Suitable_images_21">Suitable images</span></h4>
<p>The input image must be 3D and binary.
</p>
<h4><span class="mw-headline" id="Results_15">Results</span></h4>
<ul><li> <b>Bone volume</b>: volume of the mesh created from bone voxels.</li>
<li> <b>Total volume</b>: volume of the mesh created from the whole image.</li>
<li> <b>Volume ratio</b>: ratio of bone to total volume.</li></ul>
<p>The measures are reported separately for each 3D subspace in the image, i.e. for each channel and time frame.
</p>
<h4><span class="mw-headline" id="Differences_to_BoneJ1_10">Differences to BoneJ1</span></h4>
<ul><li> Supports hyperstacks.</li>
<li> Results differ, because the marching cubes and mesh volume implementations are different.</li>
<li> In BoneJ1 this plug-in was called <i>Volume fraction</i></li></ul>
<h2><span class="mw-headline" id="Results_table">Results table</span></h2>
<p>The BoneJ plug-ins print their results into a shared result table. This is because we often need to calculate several measures for the same image,  so it's handy to have them on one row. Repeated measures for the same image are reported on different rows. The results persist even if the table is closed. To clear the table run <i>Plugins &gt; BoneJ &gt; Table &gt; Clear BoneJ results</i>.
</p><p>Note that some of the plug-ins (marked with <i>WIP</i>) still use a ImageJ1 style results table that works slightly differently. As they are modernized they'll move to use the same new table as the others.
</p>
<h2><span class="mw-headline" id="Usage_reporting">Usage reporting</span></h2>
<p>Menu path <i> Edit &gt; Options &gt; BoneJ Usage (Modern | Legacy)</i>
</p>
<h4><span class="mw-headline" id="What_is_collected.3F">What is collected?</span></h4>
<p>BoneJ uses Google Analytics to report when a plugin's run() method completes successfully.
</p>
<pre class="brush:plain">
https://www.google-analytics.com/__utm.gif?utmwv=5.2.5&amp;utms=0&amp;utmn=1074354874&amp;utmhn=bonej.org&amp;utmt=event&amp;utme=5(Plugin%20Usage*org.bonej.wrapperPlugins.wrapperUtils.UsageReporterOptions*0.5.1)&amp;utmcs=UTF-8&amp;utmsr=3840x1080&amp;utmvp=3840x1080&amp;utmsc=24-bit&amp;utmul=en-gb&amp;utmje=0&amp;utmcn=1&amp;utmdt=bonej.org%20Usage%20Statistics&amp;utmhid=512699200&amp;utmr=-&amp;utmp=%2Fstats&amp;utmac=UA-366405-8&amp;utmcc=__utma%3D1589599318.1327557233.1538550102.1538550102.1538550102.2%3B%2B__utmz%3D1589599318.1538550102.79.42.utmcsr%3Dgoogle%7Cutmccn%3D(organic)%7Cutmcmd%3Dorganic%7Cutmctr%3DBoneJ%20Usage%20Reporter%3B
</pre>
<p>A one-pixel GIF image is requested from Google, with a rather long set of parameters. Reported details are:
</p>
<ul><li> A unique ID which is stored in your ImageJ preferences between sessions</li>
<li> Your screen resolution</li>
<li> The name of the plugin's Java class</li>
<li> The version of BoneJ</li>
<li> The first time, last time and current time you ran a BoneJ plugin</li>
<li> Your system language and character map</li>
<li> Google also records your IP address as all webservers do in their logs</li>
<li> Some of these values are saved in your preferences</li>
<li> Your operating system and Java version are reported in the user agent ID</li></ul>
<h4><span class="mw-headline" id="What_does_the_report_look_like.3F">What does the report look like?</span></h4>
<p>Data are collated with the Analytics report already used to track hits on bonej.org.
</p>
<h4><span class="mw-headline" id="For_what_is_the_data_used.3F">For what is the data used?</span></h4>
<p>So far, BoneJ has been supported mainly by grants to scientists. We wish to create quantitative reports that detail how much BoneJ is being used in the community, so that decisions to fund further development can be made rationally.
</p><p>In addition, development resources are limited so it is useful to see which features are used heavily and can be consolidated, while little-used features can be abandoned or made more useful.
</p>
<h4><span class="mw-headline" id="That_is_noble_but_my_work_is_top_secret">That is noble but my work is top secret</span></h4>
<p>If you are uncomfortable with this level of data collection then please opt out, at the dialog which is prompted on first run, or by running <i> Edit &gt; Options &gt; BoneJ Usage (Modern | Legacy) </i>. Hitting Cancel (Legacy) or unchecking the box (Modern) will prevent any report from being gathered or sent and will also clear any values that were previously saved.
</p><p>You can opt back in by running <i> Edit &gt; Options  &gt; BoneJ Usage (Modern | Legacy)</i>, checking the box (Modern only) and hitting OK.
</p>
<h4><span class="mw-headline" id="Can_I_see_the_report.3F">Can I see the report?</span></h4>
<p>At the moment the report is kept private. If there is a strong will from the user community, who created the data in the first place, the report could be made more open. Please feel free to discuss on the user forum.
</p>
<h4><span class="mw-headline" id="How_is_it_implemented.3F">How is it implemented?</span></h4>
<p>You can read the code on Github and in every copy of BoneJ. If you turn on ImageJ's debug mode (in <i> Edit &gt; Options &gt; Misc. </i> for Legacy and <i>Edit &gt; Options &gt; ImageJ2... &gt; Scijava log level DEBUG</i> for Modern) the URL and the GIF response will be logged to ImageJ's log window.
</p>
<h2><span class="mw-headline" id="Where_is_my_favourite_plug-in.3F">Where is my favourite plug-in?</span></h2>
<p>We have removed some plug-ins from BoneJ. <i>Neck shaft angle</i>, <i>Plateness</i> and <i>Structure model index</i> have been discontinued. <i>Interpolate ROIs</i>, <a rel="nofollow" class="external text" href="3D_Binary_Filters"><i>Dilate 3D</i> and <i>Erode 3D</i></a> come pre-packaged with ImageJ, so they are no longer included in BoneJ2.
</p><p>Support for <i>Kontron IMG</i>, <i>Scanco ISQ</i> and <i>Stratec pQCT</i> file formats has been moved to <a href="SCIFIO" title="SCIFIO">SCIFIO</a>. Just run <i>Edit  &gt; Options  &gt; ImageJ2</i>, and check <i>Use SCIFIO when opening files</i>. When the option is enabled, these kinds of files can be opened from <i>File &gt; Open</i> or dragging &amp; dropping them like any other format. 
</p><p><i>Distribution analysis</i> and other pQCT related tools can now be downloaded independently from the <a href="PQCT" title="PQCT">PQCT</a> update site.
</p>
<h2><span class="mw-headline" id="Licence">Licence</span></h2>
<p>BoneJ2 is free, open-source software. You can redistribute it and/or modify it under the terms of the <a rel="nofollow" class="external text" href="https://github.com/bonej-org/BoneJ2/blob/master/LICENCE.md">BSD 2-clause licence</a>. The software is provided "as is" and any warranties are disclaimed. In no event shall the copyright holder or contributors be liable.
</p>
<h2><span class="mw-headline" id="Citation_2">Citation</span></h2>
<p>If you'd like to cite the software, we will soon publish a paper about migrating BoneJ from ImageJ1 to ImageJ2. You can read the draft (and send us your feedback) as we're writing it <a rel="nofollow" class="external text" href="https://v1.overleaf.com/read/yxdkwpcdkjzj">for Wellcome Open Research on Overleaf</a>. We recommend you cite the specific release used in your research.
</p>
<h2><span class="mw-headline" id="Funding">Funding</span></h2>
<p>The redesign and porting effort to create BoneJ2 was supported by a <a rel="nofollow" class="external text" href="https://wellcome.ac.uk/grant-funding/people-and-projects/grants-awarded/bonej2-fullfeatured-nextgeneration-bonej-standardised">Wellcome Trust Biomedical Resource and Technology Development Grant</a>. Interaction with <a href="SciView" title="SciView">SciView</a> was assisted by a <a rel="nofollow" class="external text" href="https://royalsociety.org/grants-schemes-awards/grants/international-exchanges/">Royal Society International Exchange grant</a>. <a href="BoneJ_experimental.html#Ellipsoid_factor"><i>Ellipsoid Factor</i></a> validation and improvements were funded by the UK <a rel="nofollow" class="external text" href="https://bbsrc.ukri.org/research/grants-search/AwardDetails/?FundingReference=BB%2fP006167%2f1">BBSRC</a>.
</p>
<!-- 
NewPP limit report
Cached time: 20200713064654
Cache expiry: 86400
Dynamic content: false
CPU time usage: 0.120 seconds
Real time usage: 0.123 seconds
Preprocessor visited node count: 1584/1000000
Preprocessor generated node count: 5960/1000000
Post‐expand include size: 1746/2097152 bytes
Template argument size: 914/2097152 bytes
Highest expansion depth: 10/40
Expensive parser function count: 0/3
-->

<!-- 
Transclusion expansion time report (%,ms,calls,template)
100.00%   24.617      1 - Template:Infobox
100.00%   24.617      1 - -total
 73.85%   18.179      5 - Template:Person
  8.03%    1.977      1 - Template:GitHub
-->
</div><div class="printfooter">
Retrieved from "<a dir="ltr" href="index.php?title=BoneJ&amp;oldid=45837">http://imagej.net/index.php?title=BoneJ&amp;oldid=45837</a>"</div>
							</div>

			<div id="footer">
				<p> This page was last modified on 27 June 2020, at 07:56.</p><ul><li><a href="ImageJ:Privacy_policy" title="ImageJ:Privacy policy">Privacy policy</a></li><li><a href="ImageJ:About" class="mw-redirect" title="ImageJ:About">About ImageJ</a></li><li><a href="Imprint" title="Imprint">Imprint</a></li><li><a href="index.php?title=BoneJ&amp;mobileaction=toggle_view_mobile" class="noprint stopMobileRedirectToggle">Mobile view</a></li></ul>			</div>

			<div id="catlinks" class="catlinks catlinks-allhidden" data-mw="interface"></div>		</div>

		<div id="bottom-wrap">
		<div id="footer-wrap-inner">

		<div id="primary" class="footer">
			<ul>

			<li id="search" class="widget">
				<h3>Search</h3>
				<form action="http://imagej.net/index.php" id="searchform">
					<input type='hidden' name="title" value="Special:Search" />
					<div>
						<input name="search" placeholder="Search ImageJ" title="Search ImageJ [f]" accesskey="f" id="s"/>						<input type="submit" name="go" value="Search" title="Go to a page with this exact name if it exists" class="searchButton" id="searchsubmit"/>					</div>
				</form>
			</li>

			
						<li class="widget">
				<h3>Personal tools</h3>
				<div>
					<ul>
					<li id="pt-createaccount"><a href="index.php?title=Special:CreateAccount&amp;returnto=BoneJ" title="You are encouraged to create an account and log in; however, it is not mandatory">Create account</a></li><li id="pt-login"><a href="index.php?title=Special:UserLogin&amp;returnto=BoneJ" title="You are encouraged to log in; however, it is not mandatory [o]" accesskey="o">Log in</a></li>					</ul>
				</div>
			</li>
			
			<li class="widget">
							</li>

			</ul>
		</div>

		<div id="secondary" class="footer">
			<ul>

			<li id="toolbox" class="widget">
				<h3>Tools</h3>
				<ul>
				<li id="t-whatlinkshere"><a href="Special:WhatLinksHere/BoneJ" title="A list of all wiki pages that link here [j]" accesskey="j">What links here</a></li><li id="t-recentchangeslinked"><a href="Special:RecentChangesLinked/BoneJ" rel="nofollow" title="Recent changes in pages linked from this page [k]" accesskey="k">Related changes</a></li><li id="t-specialpages"><a href="Special:SpecialPages" title="A list of all special pages [q]" accesskey="q">Special pages</a></li><li id="t-print"><a href="index.php?title=BoneJ&amp;printable=yes" rel="alternate" title="Printable version of this page [p]" accesskey="p">Printable version</a></li><li id="t-permalink"><a href="index.php?title=BoneJ&amp;oldid=45837" title="Permanent link to this revision of the page">Permanent link</a></li><li id="t-info"><a href="index.php?title=BoneJ&amp;action=info" title="More information about this page">Page information</a></li>				</ul>
			</li>

			
			<li class="widget">
							</li>

			</ul>
		</div>

		<div id="ternary" class="footer">
			<ul>

			<li class="widget">
				<img id="logo" src="skins/imagej-128.png" alt=""/>			</li>

			<li class="widget">
				<h3><span class="mw-headline" id="MORE_TOOLS">MORE TOOLS</span></h3>
<ul><li> <a href="Special:ChangeUploadPassword" title="Special:ChangeUploadPassword">Set/change upload password</a></li>
<li> <a href="Special:RecentChanges" title="Special:RecentChanges">Recent changes</a></li>
<li> <a href="Special:LonelyPages" title="Special:LonelyPages">Orphaned pages</a></li></ul>
			</li>

			</ul>
		</div>
		<div style="clear: both"></div>

		</div>
		</div>

		<script>(window.RLQ=window.RLQ||[]).push(function(){mw.loader.load(["ext.fancytree","ext.suckerfish","mediawiki.toc","mediawiki.action.view.postEdit","site","mediawiki.user","mediawiki.hidpi","mediawiki.page.ready","mediawiki.searchSuggest","ext.SimpleTooltip"]);});</script><script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgBackendResponseTime":325});});</script>
		</body>
		</html>
		