<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">
<head>
<meta charset="UTF-8"/>
<title>Conference 2015 Program - ImageJ</title>
<script>document.documentElement.className = document.documentElement.className.replace( /(^|\s)client-nojs(\s|$)/, "$1client-js$2" );</script>
<script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":0,"wgPageName":"Conference_2015_Program","wgTitle":"Conference 2015 Program","wgCurRevisionId":21068,"wgRevisionId":21068,"wgArticleId":2972,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Conference"],"wgBreakFrames":false,"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRelevantPageName":"Conference_2015_Program","wgRelevantArticleId":2972,"wgRequestId":"314fcbeda9eac79b2d0c0117","wgIsProbablyEditable":false,"wgRestrictionEdit":[],"wgRestrictionMove":[],"wgPreferredVariant":"en","fancytree_path":"/extensions/TreeAndMenu/fancytree"});mw.loader.state({"site.styles":"ready","noscript":"ready","user.styles":"ready","user.cssprefs":"ready","user":"ready","user.options":"loading","user.tokens":"loading","mediawiki.legacy.shared":"ready","mediawiki.legacy.commonPrint":"ready","mediawiki.sectionAnchor":"ready","skins.erudite":"ready"});mw.loader.implement("user.options@0j3lz3q",function($,jQuery,require,module){mw.user.options.set({"variant":"en"});});mw.loader.implement("user.tokens@1ku9xth",function ( $, jQuery, require, module ) {
mw.user.tokens.set({"editToken":"+\\","patrolToken":"+\\","watchToken":"+\\","csrfToken":"+\\"});/*@nomin*/;

});mw.loader.load(["mediawiki.page.startup"]);});</script>
<link rel="stylesheet" href="load.php%3Fdebug=false&amp;lang=en&amp;modules=mediawiki.legacy.commonPrint%252Cshared%257Cmediawiki.sectionAnchor%257Cskins.erudite&amp;only=styles&amp;skin=erudite.css"/>
<script async="" src="load.php%3Fdebug=false&amp;lang=en&amp;modules=startup&amp;only=scripts&amp;skin=erudite"></script>
<link rel="stylesheet" href="extensions/TreeAndMenu/fancytree/fancytree.css"/><link rel="stylesheet" href="extensions/TreeAndMenu/suckerfish/suckerfish.css"/>
<meta name="ResourceLoaderDynamicStyles" content=""/>
<link rel="stylesheet" href="load.php%3Fdebug=false&amp;lang=en&amp;modules=site.styles&amp;only=styles&amp;skin=erudite.css"/>
<meta name="generator" content="MediaWiki 1.28.0"/>
<meta name="description" content="This page describes the full program for the 2015 ImageJ User &amp;amp; Developer Conference."/>
<link rel="shortcut icon" href="skins/ij2.ico"/>
	<meta property="og:type" content="article"/>

	<meta property="og:site_name" content="ImageJ"/>

	<meta property="og:title" content="Conference 2015 Program"/>

	<meta property="og:description" content="This page describes the full program for the 2015 ImageJ User &amp;amp; Developer Conference."/>


<meta name="viewport" content="width=device-width, initial-scale=1" />
</head>
<body class="mediawiki ltr sitedir-ltr mw-hide-empty-elt ns-0 ns-subject page-Conference_2015_Program rootpage-Conference_2015_Program skin-erudite action-view">
		<div class="mw-jump">
			<a href="Conference_2015_Program.html#bodyContent">Skip to content</a>, 			<a href="Conference_2015_Program.html#search">Skip to search</a>
		</div>

		<div id="top-wrap" role="banner">
			<h1><a href="Welcome" title="ImageJ" rel="home">ImageJ</a></h1>
			<div id="tagline">From ImageJ</div>

			<a id="menubutton" href="Conference_2015_Program.html#menu">Menu</a>
			<div id="nav" role="navigation">
			<ul id='menu'>
<li id="menu-item-n-About"><a href="ImageJ">About</a></li>
<li id="menu-item-n-Downloads"><a href="Downloads">Downloads</a></li>
<li id="menu-item-n-Learn"><a href="Learn">Learn</a></li>
<li id="menu-item-n-Develop"><a href="Development">Develop</a></li>
<li id="menu-item-n-News"><a href="News">News</a></li>
<li id="menu-item-n-Events"><a href="Events">Events</a></li>
<li id="menu-item-n-Help"><a href="Help">Help</a></li>
</ul>
			</div>
		</div>

		<div id="mw-js-message"></div>
		
		<div id="main" role="main">
			<div id="nav-meta">
			<span id="ca-nstab-main" class="selected"><a href="Conference_2015_Program" title="View the content page [c]" accesskey="c">Page</a></span><span class="meta-sep">|</span><span id="ca-talk" class="new"><a href="index.php?title=Talk:Conference_2015_Program&amp;action=edit&amp;redlink=1" rel="discussion" title="Discussion about the content page [t]" accesskey="t">Discussion</a></span><span class="meta-sep">|</span><span id="ca-viewsource"><a href="index.php?title=Conference_2015_Program&amp;action=edit" title="This page is protected.&#10;You can view its source [e]" accesskey="e">View source</a></span><span class="meta-sep">|</span><span id="ca-history"><a href="index.php?title=Conference_2015_Program&amp;action=history" title="Past revisions of this page [h]" accesskey="h">History</a></span><span class="meta-sep">|</span>			</div>

			<div id="bodyContent">
				<h1>Conference 2015 Program</h1>
				
				<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><p>This page describes the full program for the <a href="Conference_2015" title="Conference 2015">2015 ImageJ User &amp; Developer Conference</a>.
</p>
<div id="toc" class="toc"><div id="toctitle"><h2>Contents</h2></div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="Conference_2015_Program.html#Program"><span class="tocnumber">1</span> <span class="toctext">Program</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="Conference_2015_Program.html#Keynote_presentations"><span class="tocnumber">2</span> <span class="toctext">Keynote presentations</span></a>
<ul>
<li class="toclevel-2 tocsection-3"><a href="Conference_2015_Program.html#Wayne_Rasband_-_Introduction_to_ImageJ"><span class="tocnumber">2.1</span> <span class="toctext">Wayne Rasband - Introduction to ImageJ</span></a></li>
<li class="toclevel-2 tocsection-4"><a href="Conference_2015_Program.html#Curtis_Rueden_-_ImageJ2_and_Fiji:_the_ImageJ2_platform.2C_and_the_Fiji_distribution_of_ImageJ"><span class="tocnumber">2.2</span> <span class="toctext">Curtis Rueden - ImageJ2 and Fiji: the ImageJ2 platform, and the Fiji distribution of ImageJ</span></a></li>
<li class="toclevel-2 tocsection-5"><a href="Conference_2015_Program.html#Mark_Tsuchida_-_Microscope_control_and_image_acquisition_with_Micro-Manager"><span class="tocnumber">2.3</span> <span class="toctext">Mark Tsuchida - Microscope control and image acquisition with Micro-Manager</span></a></li>
<li class="toclevel-2 tocsection-6"><a href="Conference_2015_Program.html#Florian_Jug_-_Getting_stuff_done_.E2.80.93_Software_synergy_with_ImageJ_.2B_KNIME"><span class="tocnumber">2.4</span> <span class="toctext">Florian Jug - Getting stuff done – Software synergy with ImageJ + KNIME</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-7"><a href="Conference_2015_Program.html#Featured_presentations"><span class="tocnumber">3</span> <span class="toctext">Featured presentations</span></a>
<ul>
<li class="toclevel-2 tocsection-8"><a href="Conference_2015_Program.html#Pariksheet_Nanda_-_Sharing_the_value_of_software_freedom_with_non-programmers"><span class="tocnumber">3.1</span> <span class="toctext">Pariksheet Nanda - Sharing the value of software freedom with non-programmers</span></a></li>
<li class="toclevel-2 tocsection-9"><a href="Conference_2015_Program.html#Tiago_Ferreira_-_Alternative_approaches_for_neuronal_morphometry"><span class="tocnumber">3.2</span> <span class="toctext">Tiago Ferreira - Alternative approaches for neuronal morphometry</span></a></li>
<li class="toclevel-2 tocsection-10"><a href="Conference_2015_Program.html#John_Bogovic_-_The_Thin_Plate_Spline_method_for_smooth_non-rigid_transformations"><span class="tocnumber">3.3</span> <span class="toctext">John Bogovic - The Thin Plate Spline method for smooth non-rigid transformations</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-11"><a href="Conference_2015_Program.html#Lightning_talks"><span class="tocnumber">4</span> <span class="toctext">Lightning talks</span></a>
<ul>
<li class="toclevel-2 tocsection-12"><a href="Conference_2015_Program.html#Anne_Alerding_-_Quantifying_soybean_stem_tissues_using_common_tools_in_ImageJ_.26_Fiji"><span class="tocnumber">4.1</span> <span class="toctext">Anne Alerding - Quantifying soybean stem tissues using common tools in ImageJ &amp; Fiji</span></a></li>
<li class="toclevel-2 tocsection-13"><a href="Conference_2015_Program.html#Fabrice_de_Chaumont_-_Yet_another_ImageJ_friend:_Icy"><span class="tocnumber">4.2</span> <span class="toctext">Fabrice de Chaumont - Yet another ImageJ friend: Icy</span></a></li>
<li class="toclevel-2 tocsection-14"><a href="Conference_2015_Program.html#Jesus_Manuel_Falagan_-_Plugins_to_import.2C_visualize.2C_process_.26_write_astronomical_FITS_files"><span class="tocnumber">4.3</span> <span class="toctext">Jesus Manuel Falagan - Plugins to import, visualize, process &amp; write astronomical FITS files</span></a></li>
<li class="toclevel-2 tocsection-15"><a href="Conference_2015_Program.html#Bill_Heeschen_-_Quantitative_characterization_of_cellular_irregularities_in_extruded_polystyrene_foam_using_digital_image_processing_and_analysis"><span class="tocnumber">4.4</span> <span class="toctext">Bill Heeschen - Quantitative characterization of cellular irregularities in extruded polystyrene foam using digital image processing and analysis</span></a></li>
<li class="toclevel-2 tocsection-16"><a href="Conference_2015_Program.html#Mark_Hiner_-_ImageJ.2BMATLAB:_Reunited"><span class="tocnumber">4.5</span> <span class="toctext">Mark Hiner - ImageJ+MATLAB: Reunited</span></a></li>
<li class="toclevel-2 tocsection-17"><a href="Conference_2015_Program.html#Irene_Landrum_-_SWAMP_.E2.80.93_a_no-cost.2C_open.2C_high-performance_computing_platform_for_continuous_software_assurance"><span class="tocnumber">4.6</span> <span class="toctext">Irene Landrum - SWAMP – a no-cost, open, high-performance computing platform for continuous software assurance</span></a></li>
<li class="toclevel-2 tocsection-18"><a href="Conference_2015_Program.html#Melissa_Linkert_-_OME_Bio-Formats:_10_years_of_proprietary_image_data"><span class="tocnumber">4.7</span> <span class="toctext">Melissa Linkert - OME Bio-Formats: 10 years of proprietary image data</span></a></li>
<li class="toclevel-2 tocsection-19"><a href="Conference_2015_Program.html#Michael_Majurski_-_Accurate_and_scalable_Microscopy_Image_Stitching_Tool_.28MIST.29"><span class="tocnumber">4.8</span> <span class="toctext">Michael Majurski - Accurate and scalable Microscopy Image Stitching Tool (MIST)</span></a></li>
<li class="toclevel-2 tocsection-20"><a href="Conference_2015_Program.html#Benjamin_Nanes_-_Slide_Set:_Simplified_batch_processing_with_ImageJ2"><span class="tocnumber">4.9</span> <span class="toctext">Benjamin Nanes - Slide Set: Simplified batch processing with ImageJ2</span></a></li>
<li class="toclevel-2 tocsection-21"><a href="Conference_2015_Program.html#Brian_Northan_-_Flexible_deconvolution_using_ImageJ_Ops"><span class="tocnumber">4.10</span> <span class="toctext">Brian Northan - Flexible deconvolution using ImageJ Ops</span></a></li>
<li class="toclevel-2 tocsection-22"><a href="Conference_2015_Program.html#Stephan_Saalfeld_-_Seamless_stitching_without_blending"><span class="tocnumber">4.11</span> <span class="toctext">Stephan Saalfeld - Seamless stitching without blending</span></a></li>
<li class="toclevel-2 tocsection-23"><a href="Conference_2015_Program.html#Raghavender_Sahdev_-_TimeLapseReg_.E2.80.93_An_ImageJ_plugin_for_drift_correction_of_video_sequence_in_time-lapse_microscopy"><span class="tocnumber">4.12</span> <span class="toctext">Raghavender Sahdev - TimeLapseReg – An ImageJ plugin for drift correction of video sequence in time-lapse microscopy</span></a></li>
<li class="toclevel-2 tocsection-24"><a href="Conference_2015_Program.html#Benjamin_Schmid_-_Real-time_multi-view_deconvolution_of_time-lapse_data_on_the_GPU"><span class="tocnumber">4.13</span> <span class="toctext">Benjamin Schmid - Real-time multi-view deconvolution of time-lapse data on the GPU</span></a></li>
<li class="toclevel-2 tocsection-25"><a href="Conference_2015_Program.html#Amitabh_Verma_-_Acquisition.2C_processing_.26_re-processing_using_OpenPolScope_plugin_for_Micro-Manager"><span class="tocnumber">4.14</span> <span class="toctext">Amitabh Verma - Acquisition, processing &amp; re-processing using OpenPolScope plugin for Micro-Manager</span></a></li>
<li class="toclevel-2 tocsection-26"><a href="Conference_2015_Program.html#Thorsten_Wagner_-_NanoTrackJ_.E2.80.93_An_open-source_multi-modal_size_characterization_tool_for_nanoparticle_tracking_analysis_based_on_ImageJ"><span class="tocnumber">4.15</span> <span class="toctext">Thorsten Wagner - NanoTrackJ – An open-source multi-modal size characterization tool for nanoparticle tracking analysis based on ImageJ</span></a></li>
<li class="toclevel-2 tocsection-27"><a href="Conference_2015_Program.html#Petr_Walczysko_-_The_Open_Microscopy_Environment_.28OME.29"><span class="tocnumber">4.16</span> <span class="toctext">Petr Walczysko - The Open Microscopy Environment (OME)</span></a></li>
<li class="toclevel-2 tocsection-28"><a href="Conference_2015_Program.html#Jay_Warrick_-_JEX_.E2.80.93_a_platform_for_data_management_and_batch_processing_with_standard_tools"><span class="tocnumber">4.17</span> <span class="toctext">Jay Warrick - JEX – a platform for data management and batch processing with standard tools</span></a></li>
<li class="toclevel-2 tocsection-29"><a href="Conference_2015_Program.html#Aryeh_Weiss_-_Automated_processing_of_telomeres_with_ImageJ_and_R"><span class="tocnumber">4.18</span> <span class="toctext">Aryeh Weiss - Automated processing of telomeres with ImageJ and R</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-30"><a href="Conference_2015_Program.html#Workshops"><span class="tocnumber">5</span> <span class="toctext">Workshops</span></a>
<ul>
<li class="toclevel-2 tocsection-31"><a href="Conference_2015_Program.html#Marcel_Austenfeld_-_Bio7_.E2.80.93_Statistical_image_analysis_and_reproducible_reports_with_ImageJ_and_R"><span class="tocnumber">5.1</span> <span class="toctext">Marcel Austenfeld - Bio7 – Statistical image analysis and reproducible reports with ImageJ and R</span></a></li>
<li class="toclevel-2 tocsection-32"><a href="Conference_2015_Program.html#Jan_Brocher_-_Image_processing.2C_feature_extraction_and_analysis_using_the_BioVoxxel_Toolbox"><span class="tocnumber">5.2</span> <span class="toctext">Jan Brocher - Image processing, feature extraction and analysis using the BioVoxxel Toolbox</span></a></li>
<li class="toclevel-2 tocsection-33"><a href="Conference_2015_Program.html#Fabrice_de_Chaumont_and_Stephane_Dallongeville_-_Hands_on_Icy_.21"><span class="tocnumber">5.3</span> <span class="toctext">Fabrice de Chaumont and Stephane Dallongeville - Hands on Icy&#160;!</span></a></li>
<li class="toclevel-2 tocsection-34"><a href="Conference_2015_Program.html#Tiago_Ferreira_-_Scripting_with_BAR"><span class="tocnumber">5.4</span> <span class="toctext">Tiago Ferreira - Scripting with BAR</span></a></li>
<li class="toclevel-2 tocsection-35"><a href="Conference_2015_Program.html#Dominik_Lindner_and_Petr_Walczysko_-_The_OMERO_platform_for_management_of_microscopy_data_and_beyond"><span class="tocnumber">5.5</span> <span class="toctext">Dominik Lindner and Petr Walczysko - The OMERO platform for management of microscopy data and beyond</span></a></li>
<li class="toclevel-2 tocsection-36"><a href="Conference_2015_Program.html#Matt_McCormick_-_SimpleITK"><span class="tocnumber">5.6</span> <span class="toctext">Matt McCormick - SimpleITK</span></a></li>
<li class="toclevel-2 tocsection-37"><a href="Conference_2015_Program.html#Curtis_Rueden_and_Brian_Northan_-_ImageJ2_scripts:_parameters_.2B_ImageJ_Ops"><span class="tocnumber">5.7</span> <span class="toctext">Curtis Rueden and Brian Northan - ImageJ2 scripts: parameters + ImageJ Ops</span></a></li>
<li class="toclevel-2 tocsection-38"><a href="Conference_2015_Program.html#Aryeh_Weiss1.2C2_-_An_introduction_to_digital_image_processing"><span class="tocnumber">5.8</span> <span class="toctext">Aryeh Weiss<sup>1,2</sup> - An introduction to digital image processing</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-39"><a href="Conference_2015_Program.html#Posters"><span class="tocnumber">6</span> <span class="toctext">Posters</span></a>
<ul>
<li class="toclevel-2 tocsection-40"><a href="Conference_2015_Program.html#Ignacio_Arganda-Carreras_-_Trainable_Weka_Segmentation:_a_tool_for_machine-learning-based_image_segmentation"><span class="tocnumber">6.1</span> <span class="toctext">Ignacio Arganda-Carreras - Trainable Weka Segmentation: a tool for machine-learning-based image segmentation</span></a></li>
<li class="toclevel-2 tocsection-41"><a href="Conference_2015_Program.html#Karl_Bellve_-_Design_and_implementation_of_3D_focus_stabilization_for_fluorescence_microscopy"><span class="tocnumber">6.2</span> <span class="toctext">Karl Bellve - Design and implementation of 3D focus stabilization for fluorescence microscopy</span></a></li>
<li class="toclevel-2 tocsection-42"><a href="Conference_2015_Program.html#Jan_Brocher_-_The_BioVoxxel_Toolbox"><span class="tocnumber">6.3</span> <span class="toctext">Jan Brocher - The BioVoxxel Toolbox</span></a></li>
<li class="toclevel-2 tocsection-43"><a href="Conference_2015_Program.html#Matthew_Caputo_-_Size_distribution_of_metal_particles_for_additive_manufacturing_using_image_analysis_techniques_by_ImageJ"><span class="tocnumber">6.4</span> <span class="toctext">Matthew Caputo - Size distribution of metal particles for additive manufacturing using image analysis techniques by ImageJ</span></a></li>
<li class="toclevel-2 tocsection-44"><a href="Conference_2015_Program.html#Sherry_Clendenon_-_Image_Acquisition_and_Processing_for_Cell_Based_Model_Instantiation_and_Validation"><span class="tocnumber">6.5</span> <span class="toctext">Sherry Clendenon - Image Acquisition and Processing for Cell Based Model Instantiation and Validation</span></a></li>
<li class="toclevel-2 tocsection-45"><a href="Conference_2015_Program.html#Fabrice_de_Chaumont_-_Icy:_an_open_source_software_for_image_analysis"><span class="tocnumber">6.6</span> <span class="toctext">Fabrice de Chaumont - Icy: an open source software for image analysis</span></a></li>
<li class="toclevel-2 tocsection-46"><a href="Conference_2015_Program.html#Olaya_Fernandez-Gayol_-_Using_the_wrMTrck_plugin_to_analyze_the_Morris_Water_maze_in_mice"><span class="tocnumber">6.7</span> <span class="toctext">Olaya Fernandez-Gayol - Using the wrMTrck plugin to analyze the Morris Water maze in mice</span></a></li>
<li class="toclevel-2 tocsection-47"><a href="Conference_2015_Program.html#Andreas_Jahnen_-_Segmentation_of_intracranial_aneurysms_in_2D_Digital_Subtraction_Angiography_.28DSA.29_images_using_ImageJ_Tools_and_Plugins"><span class="tocnumber">6.8</span> <span class="toctext">Andreas Jahnen - Segmentation of intracranial aneurysms in 2D Digital Subtraction Angiography (DSA) images using ImageJ Tools and Plugins</span></a></li>
<li class="toclevel-2 tocsection-48"><a href="Conference_2015_Program.html#David_Legland_-_MorphoLibJ:_an_integrated_library_for_morphological_filtering.2C_segmentation_and_analysis_of_plant_tissues"><span class="tocnumber">6.9</span> <span class="toctext">David Legland - MorphoLibJ: an integrated library for morphological filtering, segmentation and analysis of plant tissues</span></a></li>
<li class="toclevel-2 tocsection-49"><a href="Conference_2015_Program.html#David_Logan_-_The_ImageJ-CellProfiler_Interface"><span class="tocnumber">6.10</span> <span class="toctext">David Logan - The ImageJ-CellProfiler Interface</span></a></li>
<li class="toclevel-2 tocsection-50"><a href="Conference_2015_Program.html#Michael_Majurski_-_Accurate_and_scalable_Microscopy_Image_Stitching_Tool_.28MIST.29_2"><span class="tocnumber">6.11</span> <span class="toctext">Michael Majurski - Accurate and scalable Microscopy Image Stitching Tool (MIST)</span></a></li>
<li class="toclevel-2 tocsection-51"><a href="Conference_2015_Program.html#Jeffrey_Murphy_-_Untitled"><span class="tocnumber">6.12</span> <span class="toctext">Jeffrey Murphy - Untitled</span></a></li>
<li class="toclevel-2 tocsection-52"><a href="Conference_2015_Program.html#Kelsey_Nemec_-_Untitled"><span class="tocnumber">6.13</span> <span class="toctext">Kelsey Nemec - Untitled</span></a></li>
<li class="toclevel-2 tocsection-53"><a href="Conference_2015_Program.html#John_Parrish_-_Sperm_nuclear_shape_and_texture_analysis"><span class="tocnumber">6.14</span> <span class="toctext">John Parrish - Sperm nuclear shape and texture analysis</span></a></li>
<li class="toclevel-2 tocsection-54"><a href="Conference_2015_Program.html#Tobias_Pietzsch_-_BigDataViewer.E2.80.94Visualization_and_Image_Processing_for_Terabyte_Data_Sets"><span class="tocnumber">6.15</span> <span class="toctext">Tobias Pietzsch - BigDataViewer&#8212;Visualization and Image Processing for Terabyte Data Sets</span></a></li>
<li class="toclevel-2 tocsection-55"><a href="Conference_2015_Program.html#Stephan_Saalfeld_-_ImgLib2_for_visualization_and_generic_image_processing"><span class="tocnumber">6.16</span> <span class="toctext">Stephan Saalfeld - ImgLib2 for visualization and generic image processing</span></a></li>
<li class="toclevel-2 tocsection-56"><a href="Conference_2015_Program.html#Md_Abdul_Kader_Sagar_-_The_SLIM_Curve_plugin_for_lifetime_image_analysis"><span class="tocnumber">6.17</span> <span class="toctext">Md Abdul Kader Sagar - The SLIM Curve plugin for lifetime image analysis</span></a></li>
<li class="toclevel-2 tocsection-57"><a href="Conference_2015_Program.html#Raghavender_Sahdev_-_TimeLapseReg_.E2.80.93_An_ImageJ_plugin_for_drift_correction_of_video_sequence_in_time-lapse_microscopy_2"><span class="tocnumber">6.18</span> <span class="toctext">Raghavender Sahdev - TimeLapseReg – An ImageJ plugin for drift correction of video sequence in time-lapse microscopy</span></a></li>
<li class="toclevel-2 tocsection-58"><a href="Conference_2015_Program.html#Joey_Tidei_-_Untitled"><span class="tocnumber">6.19</span> <span class="toctext">Joey Tidei - Untitled</span></a></li>
<li class="toclevel-2 tocsection-59"><a href="Conference_2015_Program.html#Jean-Yves_Tinevez_-_TrackMate"><span class="tocnumber">6.20</span> <span class="toctext">Jean-Yves Tinevez - TrackMate</span></a></li>
<li class="toclevel-2 tocsection-60"><a href="Conference_2015_Program.html#Amitabh_Verma_-_Acquisition.2C_Processing_.26_Re-Processing_using_OpenPolScope_plugin_for_Micro-Manager_2"><span class="tocnumber">6.21</span> <span class="toctext">Amitabh Verma - Acquisition, Processing &amp; Re-Processing using OpenPolScope plugin for Micro-Manager</span></a></li>
<li class="toclevel-2 tocsection-61"><a href="Conference_2015_Program.html#Thorsten_Wagner_-_Tools_and_methods_of_the_Biomedical_Imaging_Group_for_ImageJ:_An_Overview"><span class="tocnumber">6.22</span> <span class="toctext">Thorsten Wagner - Tools and methods of the Biomedical Imaging Group for ImageJ: An Overview</span></a></li>
</ul>
</li>
</ul>
</div>

<h2><span class="mw-headline" id="Program">Program</span></h2>
<p><iframe width="100%" height="700" frameborder="0" src="https://docs.google.com/spreadsheets/d/1xl0owZ1kAJDLrA2XtNjlvKwhO5ZsS5_CMmw-dU64sFk/pubhtml?gid=;single=true&amp;widget=true&amp;headers=false"></iframe>
</p>
<div style="clear:both;"></div>
<p><br />
</p>
<h2><span class="mw-headline" id="Keynote_presentations">Keynote presentations</span></h2>
<p>There were four 45-minute keynote presentations.
</p>
<h3><span class="mw-headline" id="Wayne_Rasband_-_Introduction_to_ImageJ">Wayne Rasband - Introduction to ImageJ</span></h3>
<p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929691/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Curtis_Rueden_-_ImageJ2_and_Fiji:_the_ImageJ2_platform.2C_and_the_Fiji_distribution_of_ImageJ">Curtis Rueden - ImageJ2 and Fiji: the ImageJ2 platform, and the Fiji distribution of ImageJ</span></h3>
<p>[ <a rel="nofollow" class="external text" href="http://imagej.github.io/presentations/2015-09-03-imagej2-and-fiji/#/">Presentation slides</a> ] [ <a rel="nofollow" class="external text" href="https://vimeo.com/140929687/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Mark_Tsuchida_-_Microscope_control_and_image_acquisition_with_Micro-Manager">Mark Tsuchida - Microscope control and image acquisition with Micro-Manager</span></h3>
<p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098816/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Florian_Jug_-_Getting_stuff_done_.E2.80.93_Software_synergy_with_ImageJ_.2B_KNIME">Florian Jug - Getting stuff done – Software synergy with ImageJ + KNIME</span></h3>
<p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098839/">Presentation video</a> ]
</p>
<h2><span class="mw-headline" id="Featured_presentations">Featured presentations</span></h2>
<p>Three lightning talks were selected as featured presentations, with elongated presentation times.
</p>
<h3><span class="mw-headline" id="Pariksheet_Nanda_-_Sharing_the_value_of_software_freedom_with_non-programmers">Pariksheet Nanda - Sharing the value of software freedom with non-programmers</span></h3>
<p>The liberating benefits of software like <a href="ImageJ" title="ImageJ">ImageJ</a> are <a href="Philosophy" title="Philosophy">self-evident</a> to free and <a href="Open_source" class="mw-redirect" title="Open source">open source</a> software developers. Yet many scientists opt for proprietary software for their data analysis needs. Why is this? Proprietary software companies create their own gated communities, offering convenience, and guarantees of support and expertise. In recent years, major hardware companies no longer treat imaging software as a commodity, and have instead recreated it as an all-in-one-experiment solution. The growth of open scientific computing would benefit from a stronger local presence. Proprietary software companies rely on personal interaction, including face-to-face lab presence, to build and maintain their community. The null hypothesis is for free and open software to continue to rely heavily on online communication, which leaves neophytes vulnerable to supplier lock-in, and isolation from the free software community. This lightning talk will discuss specific approaches of empowering and culturalizing scientists into the open scientific computing community.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/pariksheet-nanda/transcript.pdf">Transcript</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098825/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Tiago_Ferreira_-_Alternative_approaches_for_neuronal_morphometry">Tiago Ferreira - Alternative approaches for neuronal morphometry</span></h3>
<p>Fully automated systems capable of comprehensive characterization of neuronal arbors remain unavailable. As a consequence, quantitative neuroanatonomy relies on onerous reconstruction of individual cells. Here, I introduce alternative approaches to cellular morphometry that bypass time-consuming manual reconstructions. I will demonstrate the utility of such approaches by providing several case studies: First, I show how the <a href="Sholl" class="mw-redirect" title="Sholl">Sholl</a> technique can be performed directly on segmented images to classify highly-related cortical interneurons and to quantify complex arbors from Brainbow-expressing cerebellum. Second, I show how <a href="Strahler" class="mw-redirect" title="Strahler">skeletonization</a> can be used to unravel the genetics of dendritic arborization in Drosophila sensory neurons. Third, I discuss preliminary data on how the three-dimensional anatomy of ensembles of networked glia can be quantified in the human brain. To conclude, I will discuss how these approaches could be generalized in the neurosciences, and how these efforts could pave the way to an ImageJ toolbox dedicated to neuronal morphometry.
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098832/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="John_Bogovic_-_The_Thin_Plate_Spline_method_for_smooth_non-rigid_transformations">John Bogovic - The Thin Plate Spline method for smooth non-rigid transformations</span></h3>
<p>The Thin Plate Spline is an established method to yield a smooth non-rigid transformation from a sparse set of control points. It is attractive compared to other geometric interpolation schemes because it is independent of the number of dimensions, parameterless, yields a deformation with minimal bending energy, and has a closed form solution. We implemented the n-dimensional Thin Plate Spline in pure Java adapting and extending the open source C++ implementation in <a href="ITK" title="ITK">ITK</a>. In addition to a standalone <a href="Open_source" class="mw-redirect" title="Open source">open source</a> library, we made the implementation accessible through the transformation interfaces of the mpicbg library, <a href="TrakEM2" title="TrakEM2">TrakEM2</a>, and <a href="ImgLib2" title="ImgLib2">ImgLib2</a> such that it can be easily integrated into existing and new <a href="ImageJ" title="ImageJ">ImageJ</a>-based software. Finally, we developed end-user plugins that use the <a href="BigDataViewer" title="BigDataViewer">BigDataViewer</a> and the Thin Plate Spline for manual interactive landmark-based registration of 2- and 3- dimensional images. We made our tools and library available through the ImageJ distribution <a href="Fiji" title="Fiji">Fiji</a>.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/john-bogovic/2015_imagej.pptx">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929688/">Presentation video</a> ]
</p>
<h2><span class="mw-headline" id="Lightning_talks">Lightning talks</span></h2>
<p>The lightning talks will be 10-minute presentations including time for questions.
</p>
<h3><span class="mw-headline" id="Anne_Alerding_-_Quantifying_soybean_stem_tissues_using_common_tools_in_ImageJ_.26_Fiji">Anne Alerding - Quantifying soybean stem tissues using common tools in ImageJ &amp; Fiji</span></h3>
<p>Plant organs such as stems are composed of tissues containing thick and thin-walled cells that are easily viewed using histological microscopy. Two dyes commonly used to differentiate between cell types are safranin, which stains the lignin in secondary walls of thick-walled cells red and fastgreen, which turns cellulose in thin, primary cell walls blue. Cells with walls containing both lignin and cellulose appear purple. Complicating successful image analysis and quantification of different cell and tissue types is that each plant differentially absorbs dye, yielding dynamic RGB values with overlapping values for each color group. Thus, quantifying plant tissues via image analysis requires outlining tissues one at a time and saving overlays using the ROI manager. We have perfected this technique to quantify soybean stem tissues using common tools in <a href="ImageJ" title="ImageJ">ImageJ</a> and <a href="Fiji" title="Fiji">Fiji</a> and with zoom-in mouse technology.
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098824/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Fabrice_de_Chaumont_-_Yet_another_ImageJ_friend:_Icy">Fabrice de Chaumont - Yet another ImageJ friend: Icy</span></h3>
<p><a href="Icy" title="Icy">Icy</a> is a free open-source bio-image analysis software. During its 3 years of life it has been continuously improved and continuously brings new features to users and developers. Today Icy is used by more than one thousands of regular users which appreciate its intuitive GUI, its ray-traced 3D visualization and its cutting-edge analysis methods. Users can also adapt and create new algorithms with script and the graphical programming protocol designer and store them on the Icy website which centralize all resources (<a rel="nofollow" class="external free" href="http://icy.bioimageanalysis.org">http://icy.bioimageanalysis.org</a>). Thus it makes available plug-ins, protocols and scripts to everybody. Centralization also allows searching directly from within the application for specific features and enables it in a one-click install. During the last months, Icy has received news exciting features: revisited 3D rendering based on latest VTK version and a new powerful Undo framework. Finally, Icy is always evolving, thanks to the feedback of all its users on <a rel="nofollow" class="external free" href="http://icy.bioimageanalysis.org/support">http://icy.bioimageanalysis.org/support</a>&#160;!
</p><p><span class="Z3988" title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rft.genre=book&amp;rft.title=Icy%3A+an+open+bioimage+informatics+platform+for+extended+reproducible+research&amp;rft.aulast=de+Chaumont&amp;rft.aufirst=Fabrice&amp;rft.date=2012&amp;rft.volume=9%287%29&amp;rft.pages=690-696&amp;rft_id=info:pmid/22743774&amp;rft_id=http%3A%2F%2Fwww.nature.com%2Fnmeth%2Fjournal%2Fv9%2Fn7%2Ffull%2Fnmeth.2075.html"><cite id="CITEREFde_ChaumontDallongevilleChenouardHerv.C3.A92012">de Chaumont, Fabrice; Dallongeville, Stéphane&#32;&amp; Chenouard, Nicolas&#32;et al.&#32;(2012),&#32;"<a rel="nofollow" class="external text" href="http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html">Icy: an open bioimage informatics platform for extended reproducible research</a>",&#32;<i>Nature methods</i>&#32;<b>9(7)</b>:  690-696, PMID 22743774<span class="printonly">, &lt;<a rel="nofollow" class="external free" href="http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html">http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html</a>&gt;</span></cite></span> (<a rel="nofollow" class="external text" href="http://scholar.google.com/scholar?cluster=8170444028478407032">on Google Scholar</a>).
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929690/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Jesus_Manuel_Falagan_-_Plugins_to_import.2C_visualize.2C_process_.26_write_astronomical_FITS_files">Jesus Manuel Falagan - Plugins to import, visualize, process &amp; write astronomical FITS files</span></h3>
<p>This abstract summarizes the progress on the development of tools for astronomical data cube (stack) visualization and spectral analysis (<a rel="nofollow" class="external text" href="http://cab.inta-csic.es/madcuba/Portada.html#MADCUBA_.2B_IMAGEJ">MADCUBA_IJ</a>) using the ImageJ infrastructure we are carrying out at the Centro de Astrobiologia (INTA-CSIC) in Spain. We will present the development of plugins to import, visualize, process and write multi-axis astronomical FITS files. We have developed a new class, the AstronomicalImagePlus, which extends ImagePlus with the astronomical "calibration" (WCSGlobal). This allows applying the full potential of ImageJ data processing and the managing of the FITS header to properly account for the processing applied to the stack. All axes, spatial, spectral, polarization and intensity are fully "calibrated" allowing visualization and processing with their associated astronomical units and their conversion to other related units. Implementation of the new classes/plugins will extend the use of ImageJ to astronomical FITS files with properly updated header.
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929695/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Bill_Heeschen_-_Quantitative_characterization_of_cellular_irregularities_in_extruded_polystyrene_foam_using_digital_image_processing_and_analysis">Bill Heeschen - Quantitative characterization of cellular irregularities in extruded polystyrene foam using digital image processing and analysis</span></h3>
<p>Flow-induced banding patterns in extrusion-based foams can lead to aesthetic issues, or, in severe cases, to compromised product performance. Image collection and analysis procedures are described that allow the pattern to be captured with good contrast and quantitatively analyzed in a manner that is consistent with human perception. The procedure for pattern characterization and classification will be discussed. The key component is isolating the bright/dark banding in the context of the cellular structure where the size of the individual cells can be on the order of the width of the bands. Cell-by-cell contrast is often greater than the humanly-perceived contrast from bands of correlated cell size within an area. Removing this ambiguity in the assignment of bright/dark patterns to cells versus bands allows the foam pattern to be analyzed directly. Image analysis results are shown to be consistent with human panel ratings for a series of several foam samples.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/bill-heeschen/Heeschen_2015ImageJConf_TalkSlides.pdf">Presentation slides (PDF)</a>, <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/bill-heeschen/Heeschen_2015ImageJConf_TalkSlides.pptx">Presentation slides (PPTX)</a>, <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/bill-heeschen/Heeschen_2015ImageJConf_FoamPatternMacros.ijm">Macro</a> 
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098843/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Mark_Hiner_-_ImageJ.2BMATLAB:_Reunited">Mark Hiner - ImageJ+MATLAB: Reunited</span></h3>
<p>ImageJ and MATLAB both provide powerful scientific development paradigms. To lower the barrier between these tools, ImageJ now offers a MATLAB script language plugin, making it possible to execute parameterized MATLAB scripts from within ImageJ's script editor. In the other direction, the ability to call ImageJ from MATLAB has been improved with an enhanced MIJ.m script, which supports the full range of installed plugins, and supports the use of both ImageJ1 and ImageJ2 data structures from the MATLAB command prompt. This bidirectional support facilitates the sharing and reuse of efforts from both communities.
</p><p>For further details and code links, see <a href="MATLAB_Scripting" title="MATLAB Scripting">MATLAB Scripting</a>.
</p><p>[ <a rel="nofollow" class="external text" href="http://imagej.github.io/presentations/2015-09-03-imagej-matlab/">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929686/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Irene_Landrum_-_SWAMP_.E2.80.93_a_no-cost.2C_open.2C_high-performance_computing_platform_for_continuous_software_assurance">Irene Landrum - SWAMP – a no-cost, open, high-performance computing platform for continuous software assurance</span></h3>
<p>The <a rel="nofollow" class="external text" href="https://www.mir-swamp.org/">Software Assurance Marketplace</a> (SWAMP) is a no-cost, open, high-performance computing platform for <a rel="nofollow" class="external text" href="https://continuousassurance.org/">continuous software assurance</a>. It includes an array of open-source and commercial software security testing tools, a library of applications with known vulnerabilities, and a comprehensive results viewer (CodeDx) for vulnerability remediation throughout the software development lifecycle.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/irene-landrum/SWAMP%20%20Presentation%202015Aug25.pptx">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098818/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Melissa_Linkert_-_OME_Bio-Formats:_10_years_of_proprietary_image_data">Melissa Linkert - OME Bio-Formats: 10 years of proprietary image data</span></h3>
<p>Despite significant advances in biological imaging and analysis, major informatics challenges remain unsolved: file formats are proprietary, storage and analysis facilities are lacking, as are standards for sharing image data and results. The <a href="Open_Microscopy_Environment" class="mw-redirect" title="Open Microscopy Environment">Open Microscopy Environment</a> (OME) is an <a href="Open-source" class="mw-redirect" title="Open-source">open-source</a> software framework developed to address these challenges. OME has three components: 1) an open data model for biological imaging, the OME data model; 2) standardized file formats (OME-TIFF) and software libraries for file conversion (<a href="Bio-Formats" title="Bio-Formats">Bio-Formats</a>); and 3) a software platform for image data management and analysis (<a href="OMERO" title="OMERO">OMERO</a>). OME has now been developing the Bio-Formats library for 10 years, during which time it has seen rapid advances in the number of file formats and imaging modalities that are supported. In recognition of this milestone, we will discuss the history of Bio-Formats, notable improvements in the recent 5.1.x releases, and plans for supporting the next generation of imaging technology.
</p><p>[ <a rel="nofollow" class="external text" href="http://ome.github.io/presentations/2015/bioformats-overview-imagej-conf/#/">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929689/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Michael_Majurski_-_Accurate_and_scalable_Microscopy_Image_Stitching_Tool_.28MIST.29">Michael Majurski - Accurate and scalable Microscopy Image Stitching Tool (MIST)</span></h3>
<p>With new microscope technologies scientists are acquiring terabyte-sized datasets to cover large areas. An automated optical microscope images a large sample by generating a grid of partially overlapping images. This is required because its field of view is much smaller than the dimensions of the specimen being imaged. This process generates a great number of images that need to be stitched into a large mosaic in order to derive meaningful information. A <a href="Stitching" class="mw-redirect" title="Stitching">stitching</a> tool is required to generate these large mosaics. The resulting image can have tens of thousands of pixels per side. Stitching one large image mosaic is computationally taxing and the computation becomes overwhelming when stitching repeated frames of live cell experiments.
</p><p>Optical microscopy often has to process feature-poor images (e.g., sparsely populated cell cultures) and may be used to derive measurements (quantitative information such as counts, densities, intensities, etc.) from such images. We developed a new stitching technique that minimizes the uncertainty due to the lack of features between overlapping areas while being faster than the leading alternative tool in the field. This method optimizes the translations computed by a pairwise image registration method, like the Fourier-based phase correlation image alignment method [1], using the mechanical stage system model. The computed translations are optimized using the Hill Climbing algorithm constrained to a square area of 4x the stage repeatability per side. The optimization function is normalized cross correlation between images.
</p><p>Our algorithm is implemented in Java as an <a href="ImageJ" title="ImageJ">ImageJ</a> plugin which runs on both CPUs and/or GPUs taking advantage of compiled native libraries from FFTW [2], the <a rel="nofollow" class="external text" href="https://developer.nvidia.com/cuda-toolkit">NVIDIA CUDA toolkit</a> using <a rel="nofollow" class="external text" href="http://www.jcuda.org/jcuda/JCuda.html">JCuda</a>, and <a rel="nofollow" class="external text" href="https://developer.nvidia.com/cuFFT">CuFFT</a> using <a rel="nofollow" class="external text" href="http://www.jcuda.org/jcuda/jcufft/JCufft.html">JCufft</a>. The implementation is cognizant of the available computer memory and scales the execution based on the number of CPUs and GPUs on a system. The algorithm can run using as little as 200 MB (on 1040x1392 pixel image tile), though using less memory incurs longer execution times. A representative test case consisting of 16x22 image tiles (1040x1392 pixels) stitches in 24 s on the CPU and 12 s on the GPU using an Intel Xeon CPU E5-2620 @ 2.0 GHz with NVIDIA Quadro 4000 GPU (2048 MB GPU memory). This test case produces a stitched image roughly 20 000 x 21 000 pixels as shown in Figure 1. Figure 2 shows the stitching tool’s ImageJ plugin GUI.
</p><p><b>Acknowledgments:</b> This work has been supported by NIST. We would like to acknowledge the team members of the computational science in biological metrology project at NIST for providing invaluable inputs to our work.
</p>
<ul><li> [1] C. Kuglin and D. Hines, "The Phase Correlation Image Alignement Method," in Proceedings of the 1975 IEEE International Conference on Cybernetics and Society, 1975, pp. 163–165.</li>
<li> [2] M. Frigo and S. Johnson, "The Design and Implementation of {FFTW3}," Proc. IEEE, vol. 93, no. 2, pp. 216–231, 2005.</li></ul>
<p>[ <a rel="nofollow" class="external text" href="https://github.com/NIST-ISG/MIST">MIST source code</a>, <a rel="nofollow" class="external text" href="https://isg.nist.gov/deepzoomweb/resources/csmet/pages/image_stitching/image_stitching.html">MIST project page</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929697/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Benjamin_Nanes_-_Slide_Set:_Simplified_batch_processing_with_ImageJ2">Benjamin Nanes - Slide Set: Simplified batch processing with ImageJ2</span></h3>
<p>Most imaging studies in the biological sciences rely on analyses that are relatively simple. However, manual repetition of analysis tasks across even a modestly sized data set can be tedious, error prone, and difficult to reproduce. While fully automated image analysis solutions are useful for very large data sets, they are sometimes impractical for the small- and medium-sized data sets common in biology. A significant gap exists between fully interactive image analysis workflows, which are easy to establish, but difficult to reproduce, and fully automated analysis workflows, which are easy to reproduce, but difficult to establish. <a rel="nofollow" class="external text" href="http://cellbio.emory.edu/bnanes/slideset/">Slide Set</a>, a framework for reproducible image analysis and batch processing with ImageJ, bridges this gap by organizing images and related metadata, recording region of interest selections and analysis parameters, and automating analysis tasks across multiple images. Slide Set includes many built-in analysis tasks and is easily extended to automate other ImageJ plugins.
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098828/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Brian_Northan_-_Flexible_deconvolution_using_ImageJ_Ops">Brian Northan - Flexible deconvolution using ImageJ Ops</span></h3>
<p>This lightning talk will describe the implementation of deconvolution into the <a href="ImageJ2" title="ImageJ2">ImageJ2</a> platform. We will go over the basics of <a href="Deconvolution" title="Deconvolution">deconvolution</a> and introduce concepts such as noise handling, edge handling, algorithm acceleration and point spread functions. We will review strengths and weaknesses of existing ImageJ deconvolution plugins. We will then describe the design of reusable and extensible deconvolution components, that have been implemented within the <a href="ImageJ_Ops" title="ImageJ Ops">ImageJ Ops</a> framework. Where possible previous open source components have been reused and/or reimplemented. We consider the potential need for a native library bridge in order to reuse existing C and GPU based libraries. Preliminary results will be presented that have been generated using the Richardson-Lucy Algorithm with total variation regularization, non-circulant edge handling, and vector extrapolation acceleration. All code is <a href="Open_source" class="mw-redirect" title="Open source">open source</a> and has been placed in publicly available git repositories.
</p><p>[ <a rel="nofollow" class="external text" href="http://imagej.github.io/presentations/2015-09-04-imagej2-deconvolution/">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098821/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Stephan_Saalfeld_-_Seamless_stitching_without_blending">Stephan Saalfeld - Seamless stitching without blending</span></h3>
<p>In modern science, images of large samples are often acquired as mosaics of overlapping tiles that are then stitched to yield a single high resolution image. Photomoetric differences across tiles can arise from both the imaging system and varying properties of the sample leading to visible seams and abrupt intensity variation in the stitched image. We have developed a method that calculates a smooth intensity correction field for each individual tile that minimizes the differences of corresponding intensities in overlapping image areas and by that leads to seamless montages. Our method performs well in setups that include uncalibrated acquisition devices and sample-dependent intensity variances and does not require blending. Our method guarantees minimal deviation from original acquisitions and thus enables quantitative comparison of intensities across stitched images. We have implemented the method in Java using the <a href="ImgLib2" title="ImgLib2">ImgLib2</a> and mpicbg libraries and published the open source code on <a href="GitHub" title="GitHub">GitHub</a>. For end-users, the method is available as an extension of the <a href="TrakEM2" title="TrakEM2">TrakEM2</a> software and included in the ImageJ distribution <a href="Fiji" title="Fiji">Fiji</a>.
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098840/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Raghavender_Sahdev_-_TimeLapseReg_.E2.80.93_An_ImageJ_plugin_for_drift_correction_of_video_sequence_in_time-lapse_microscopy">Raghavender Sahdev - TimeLapseReg – An ImageJ plugin for drift correction of video sequence in time-lapse microscopy</span></h3>
<p>The main functionality of the <a href="TurboReg" title="TurboReg">TurboReg</a> plugin allows for aligning or matching two images. The plugin is widely used in Neuroscience for pre-processing of both static and time-lapse imaging data. In this project we develop an ImageJ plugin to be used by an end-user to correct for the drift in very long sequences of noisy multi-channel fluorescence images. TimeLapseReg uses the well-known TurboReg/<a href="StackReg" title="StackReg">StackReg</a> engine method to register frames to a reference image. It computes and applies rigid transformations of the user-selected channel and stores them for subsequent use. The geometrical transformations are further applied to all channels. We present the user with a drift trajectory over the entire image sequence. The user has the option of choosing the reference image and discarding corrupted images. We have applied our software to calcium imaging data for helping neuroscientists to better infer neuronal network spiking activity.
</p><p>[ <a rel="nofollow" class="external text" href="http://neuroinformatics.be/projects/stackreg-plus-improved-multiple-image-alignment-with-imagej/">TimeLapseReg web page</a>, <a rel="nofollow" class="external text" href="https://github.com/incfbelgiannode/StackReg-Plus_Multiple-Image-Alignment">TimeLapseReg source code</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098822/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Benjamin_Schmid_-_Real-time_multi-view_deconvolution_of_time-lapse_data_on_the_GPU">Benjamin Schmid - Real-time multi-view deconvolution of time-lapse data on the GPU</span></h3>
<p>State-of-the-art multi-view (MV) fusion algorithms simultaneously fuse and deconvolve data acquired by MV imaging techniques such as Selective Plane Illumination Microscopy (SPIM). MV deconvolution, however, has so far been too computationally demanding to be carried out in real-time. Here, we approximated the full 3D transfer function with a 2D PSF by neglecting one lateral component. This enabled us to process planes orthogonal to the rotation axis independently. Memory consumption was thereby reduced by 2-3 orders of magnitude, so that the entire deconvolution could be performed on the GPU. While achieving similar results as established 3D MV deconvo-lution, we thereby reduced processing times by a factor of up to 75, allowing us finally to apply MV deconvolution in real-time during time-lapse acquisition.
</p><p>[ <a rel="nofollow" class="external text" href="https://github.com/bene51/gpu_deconvolution">gpu_deconvolution source code</a>, <a rel="nofollow" class="external text" href="https://github.com/bene51/SPIM_Reconstruction_Cuda">SPIM_Reconstruction_Cuda source code</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098826/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Amitabh_Verma_-_Acquisition.2C_processing_.26_re-processing_using_OpenPolScope_plugin_for_Micro-Manager">Amitabh Verma - Acquisition, processing &amp; re-processing using OpenPolScope plugin for Micro-Manager</span></h3>
<p><a rel="nofollow" class="external text" href="http://www.openpolscope.org/">OpenPolScope</a> software provides a suite of plugins for <a href="Micro-Manager" title="Micro-Manager">Micro-Manager</a> and extends its capabilities of acquisition and adds features of processing and reprocessing. The plugin also provides a wrapper to the <a href="Macros" class="mw-redirect" title="Macros">ImageJ macro language</a> which can then be applied to a Micro-Manager acquisition stream. OpenPolScope architecture is based on plugins called Mode Plugins and is thus extensible and can be applied to a number of imaging techniques that require processing like Ratiometric imaging. The reprocessing plugin is ImageJ based using Micro-Manager libraries to access datasets. The OpenPolScope API provides a common processing pipeline for both the Micro-Manager acquisition and ImageJ re-processing plugin. OpenPolScope plugin was designed to support Lc-PolScope based Birefringence imaging where consecutive images are acquired (as channels) modulating a liquid crystal between the microscopes light-path. A resultant Retardance image is then computed on-the-fly and inserted in the acquisition stream as an additional channel.
</p><p>[ <a rel="nofollow" class="external text" href="http://openpolscope.org/pages/OpenPolScopeInstaller.htm">OpenPolScope downloads</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098844/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Thorsten_Wagner_-_NanoTrackJ_.E2.80.93_An_open-source_multi-modal_size_characterization_tool_for_nanoparticle_tracking_analysis_based_on_ImageJ">Thorsten Wagner - NanoTrackJ – An open-source multi-modal size characterization tool for nanoparticle tracking analysis based on ImageJ</span></h3>
<p>Nanoparticle tracking analysis is a widely known method to determine the hydrodynamic diameter of nanoparticles (NP) in liquids. Commercial systems have been built for convenient measurements and ease of use. However, they rely on a multitude of analysis algorithms and statistical calculations which are not disclosed to the user. To give a better understanding of what is calculated, making the method applicable for other experimental setups and additionally extending the method for exploiting color information we developed a free plugin for <a href="ImageJ" title="ImageJ">ImageJ</a> called <a rel="nofollow" class="external text" href="http://sourceforge.net/projects/nanotrackj/">NanoTrackJ</a>. It is an <a href="Open-source" class="mw-redirect" title="Open-source">open-source</a> software where all algorithms are documented and freely accessible. The talk will give an overview of the functionalities of NanoTrackJ by presenting analysis of different video sequences from various devices.
</p><p>[ <a rel="nofollow" class="external text" href="http://sourceforge.net/projects/nanotrackj/">NanoTrackJ</a>, <a rel="nofollow" class="external text" href="https://github.com/thorstenwagner/TrackMate">source code</a>, <a rel="nofollow" class="external text" href="https://dl.dropboxusercontent.com/u/560426/imagej/conference2015/slides.pdf">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098848/">Presentation Video</a> ]
</p>
<h3><span class="mw-headline" id="Petr_Walczysko_-_The_Open_Microscopy_Environment_.28OME.29">Petr Walczysko - The Open Microscopy Environment (OME)</span></h3>
<p>Despite significant advances in biological imaging and analysis, major informatics challenges remain unsolved: file formats are proprietary, storage and analysis facilities are lacking, as are standards for sharing image data and results. The <a href="Open_Microscopy_Environment" class="mw-redirect" title="Open Microscopy Environment">Open Microscopy Environment</a> (OME) is an open-source software framework developed to address these challenges. OME has three components—an open data model for biological imaging: OME data model; standardized file formats (OME-TIFF) and software libraries for file conversion (<a href="Bio-Formats" title="Bio-Formats">Bio-Formats</a>); and a software platform for image data management and analysis (<a href="OMERO" title="OMERO">OMERO</a>). The Java-based OMERO client-server platform comprises an image metadata store, an image repository, visualization and analysis by remote access, enabling sharing and publishing of image data. OMERO’s model-based architecture has enabled its extension into a range of imaging domains, including light and electron microscopy, high content screening and recently into applications using non-image data from clinical and genomic studies [<a rel="nofollow" class="external text" href="http://www.openmicroscopy.org/site/products/partner">1</a>]. Our current version, OMERO-5, improves support for large datasets and reads images directly from their original file format, allowing access by third party software. OMERO and Bio-Formats run the <a rel="nofollow" class="external text" href="http://jcb-dataviewer.rupress.org/">JCB DataViewer</a>, the world’s first on-line scientific image publishing system and several other institutional image data repositories (e.g. [<a rel="nofollow" class="external text" href="http://odr.stowers.org">2</a>, <a rel="nofollow" class="external text" href="http://emdatabank.org/">3</a>]).
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098837/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Jay_Warrick_-_JEX_.E2.80.93_a_platform_for_data_management_and_batch_processing_with_standard_tools">Jay Warrick - JEX – a platform for data management and batch processing with standard tools</span></h3>
<p><a href="ImageJ" title="ImageJ">ImageJ</a>, <a href="Fiji" title="Fiji">Fiji</a>, <a href="SciJava" title="SciJava">SciJava</a>, <a href="SCIFIO" title="SCIFIO">SCIFIO</a>, and <a href="ImgLib2" title="ImgLib2">ImgLib2</a> provide a host of capabilities for enabling advanced image analysis and have been integrated into many notable softwares including <a href="CellProfiler" title="CellProfiler">CellProfiler</a>, <a href="KNIME" title="KNIME">KNIME</a> Image Processing, and <a href="OMERO" title="OMERO">OMERO</a> to name just a few. This talk provides a vignette of another option in that growing list, <a href="JEX" title="JEX">JEX</a>. Although "reinventing the wheel" should be avoided when possible, the base java / image analysis libraries actually allow each of these separate but related softwares to enhance each other and further strengthen the common bases. In this vignette, a brief overview of JEX will be provided with the goal of highlighting one of the primary focuses of JEX, keeping your experimental data organized in a simple fashion in a context that enables batch processing with established ImageJ functionalities, java libraries, and statistical analysis / modeling softwares such as R and Octave for more streamlined workflows and objective analysis across experimental conditions.
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098819/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Aryeh_Weiss_-_Automated_processing_of_telomeres_with_ImageJ_and_R">Aryeh Weiss - Automated processing of telomeres with ImageJ and R</span></h3>
<p>Telomeres are regions of repetitive nucleotide sequences located at the ends of chromosomes, which protect them from fusion and degradation. In every cell division, the telomeres are shortened until they reach a critical length that signals the cell to go into senescence. Abnormally short or long telomeres are common phenomenon in cancer cells, as compared to healthy controls. Manual characterization of telomere length distribution is at best difficult. Here we present an image processing workflow which was developed in ImageJ and R to enable automated processing of telomere count and length distribution in thousands of cells. Over 3000 cells were imaged, each containing 30-150 visible telomeres. This could not have been done manually. The ImageJ environment provided us with the ability to acquire and analyze large quantities of data in a reproducible manner, and demonstrated the spread in data that resulted from the fundamental biology and physiology that were being studied.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/aryeh-weiss/ij2015telomere-A.ppt">Presentation slides</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929694/">Presentation video</a> ]
</p>
<h2><span class="mw-headline" id="Workshops">Workshops</span></h2>
<p>The workshops are one-to-two hour blocks which help users to explore topics interactively and in depth.
</p>
<h3><span class="mw-headline" id="Marcel_Austenfeld_-_Bio7_.E2.80.93_Statistical_image_analysis_and_reproducible_reports_with_ImageJ_and_R">Marcel Austenfeld - Bio7 – Statistical image analysis and reproducible reports with ImageJ and R</span></h3>
<p>In this workshop an overview is given about <a href="Bio7" title="Bio7">Bio7</a> which is an <a href="IDE" class="mw-redirect" title="IDE">Integrated Development Environment</a> for Ecological Modeling, Scientific Image Analysis and Statistical Analysis based on the <a href="Eclipse" class="mw-redirect" title="Eclipse">Eclipse</a> Rich Client Platform. The dynamic compilation and execution of <a href="Java" class="mw-redirect" title="Java">Java</a> code, integration of the Eclipse Java Development tools and the embedded JavaFX SceneBuilder to create and run JavaFX interfaces at runtime are some highlights of the latest development efforts which will be presented in this session. The main part of the workshop with practical exercises will be concentrated on the recently improved embedded Graphical User Interface to the statistical software R and the powerful 'spatstat' package for spatial point pattern analysis. Concomitantly a package for reproducible R reports will be used to document the different analysis steps of ImageJ data in R.
</p><p><b>Workshop requirements:</b> See <a rel="nofollow" class="external text" href="https://github.com/Bio7/Bio7_Workshop">Bio7 workshop materials</a> page
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098838/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Jan_Brocher_-_Image_processing.2C_feature_extraction_and_analysis_using_the_BioVoxxel_Toolbox">Jan Brocher - Image processing, feature extraction and analysis using the BioVoxxel Toolbox</span></h3>
<p>The workshop will give a short introduction on the influence of contrast adjustments, shading correction, different background subtraction methods and a few basic convolution filters. This includes some considerations about user bias and good scientific practice. Furthermore, it builds the basis for the main part which will focus on object extraction from grayscale images. The latter includes the usage of automatic thresholds and emphasizes the "Threshold Check" as possibility for an easier determination of suitable threshold algorithms. In the final part, the workshop will give an overview over several analysis possibilities on extracted binary objects including functions from the <a href="BioVoxxel_Toolbox" title="BioVoxxel Toolbox">BioVoxxel Toolbox</a>. Examples will include absolute counting, size and shape measurements, relative comparison of object features, object-based object extraction and differential analysis according to shape descriptors. One additional aim is to learn methods which enable the final automation of processing and analyses in a script e.g. the <a href="Macros" class="mw-redirect" title="Macros">ImageJ macro language</a>.
</p><p><b>Workshop requirements:</b> <a href="ImageJ" title="ImageJ">ImageJ</a> or <a href="Fiji" title="Fiji">Fiji</a> with the <a rel="nofollow" class="external text" href="http://www.biovoxxel.de/development/">BioVoxxel Toolbox</a> installed
</p><p><a rel="nofollow" class="external text" href="https://www.dropbox.com/s/iurb5he4qpsqfrn/BioVoxxel%20Workshop.pdf?dl=0/">Workshop_Content</a>
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929696/">Presentation Video</a> ]
</p>
<h3><span class="mw-headline" id="Fabrice_de_Chaumont_and_Stephane_Dallongeville_-_Hands_on_Icy_.21">Fabrice de Chaumont and Stephane Dallongeville - Hands on Icy&#160;!</span></h3>
<p>In this practical workshop, discover how to use <a href="Icy" title="Icy">Icy</a> to watch and analyse your images&#160;! During this session, we will see the different functionalities of Icy in action, from visualization (2D/3D) to analysis based on a real biologic case with a lot of analysis traps&#160;! Let's investigate&#160;!
</p><p><span class="Z3988" title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rft.genre=book&amp;rft.title=Icy%3A+an+open+bioimage+informatics+platform+for+extended+reproducible+research&amp;rft.aulast=de+Chaumont&amp;rft.aufirst=Fabrice&amp;rft.date=2012&amp;rft.volume=9%287%29&amp;rft.pages=690-696&amp;rft_id=info:pmid/22743774&amp;rft_id=http%3A%2F%2Fwww.nature.com%2Fnmeth%2Fjournal%2Fv9%2Fn7%2Ffull%2Fnmeth.2075.html"><cite id="CITEREFde_ChaumontDallongevilleChenouardHerv.C3.A92012">de Chaumont, Fabrice; Dallongeville, Stéphane&#32;&amp; Chenouard, Nicolas&#32;et al.&#32;(2012),&#32;"<a rel="nofollow" class="external text" href="http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html">Icy: an open bioimage informatics platform for extended reproducible research</a>",&#32;<i>Nature methods</i>&#32;<b>9(7)</b>:  690-696, PMID 22743774<span class="printonly">, &lt;<a rel="nofollow" class="external free" href="http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html">http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html</a>&gt;</span></cite></span> (<a rel="nofollow" class="external text" href="http://scholar.google.com/scholar?cluster=8170444028478407032">on Google Scholar</a>).
</p><p><b>Workshop requirements:</b> An installation of <a rel="nofollow" class="external text" href="http://icy.bioimageanalysis.org/download">Icy</a>
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929698/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Tiago_Ferreira_-_Scripting_with_BAR">Tiago Ferreira - Scripting with BAR</span></h3>
<p>The <a href="Scripting" title="Scripting">scripting</a> capabilities of <a href="ImageJ" title="ImageJ">ImageJ</a> play a pivotal role in its success because they empower users with the ability to perform efficient and reproducible scientific image processing. 
For this reason, we created Broadly Applicable Routines (<a href="BAR" title="BAR">BAR</a>): a curated collection of plugins and multi-language scripts that streamline ImageJ usage by 1) providing tools that help users create, access and organize their own macros and scripts; 2) by offering source code templates and customizable scripting libraries; and 3) by bridging the gap between IJ1 and IJ2 through the modernization of IJ1 built-in commands. BAR is <a href="Open_source" class="mw-redirect" title="Open source">open source</a>, encourages community contributions and is available through an <a href="Update_site" class="mw-redirect" title="Update site">update site</a>.
The workshop will be divided in three parts:
First, we will explore the advantages of multi-language scripting and skim through the basics of popular <a href="Scripting" title="Scripting">scripting</a> languages such as <a href="BeanShell" class="mw-redirect" title="BeanShell">BeanShell</a>, <a href="Python" title="Python">Python</a>, and (time allowing) the <a href="Introduction_into_Macro_Programming" title="Introduction into Macro Programming">ImageJ macro language</a>.
Secondly, we will explore how BAR can be used to streamline ImageJ workflows. 
Thirdly, we will explore the advantages of personalized scripting libraries that are shared across routines.
</p><p><b>Workshop requirements:</b>
</p>
<ol><li> An updated installation of Fiji</li>
<li> An internet connection, required for download of sample images and browsing of <a rel="nofollow" class="external free" href="http://javadoc.imagej.net">http://javadoc.imagej.net</a></li></ol>
<dl><dt>NB</dt>
<dd> If you subscribe to the ImageJ-dev update site, ensure that your Image1 version is at least 1.50a (the  ImageJ-dev update site may bundle an outdated version of ImageJ1). To do so, use the ImageJ1 command <span><em><span style="border-bottom:1px dotted #ccc;"> Help </span>&#160;&#8250; <span style="border-bottom:1px dotted #ccc;">Update ImageJ...</span></em></span></dd></dl>
<p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140099767/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Dominik_Lindner_and_Petr_Walczysko_-_The_OMERO_platform_for_management_of_microscopy_data_and_beyond">Dominik Lindner and Petr Walczysko - The OMERO platform for management of microscopy data and beyond</span></h3>
<p>In this workshop, we will outline and demonstrate the <a href="OMERO" title="OMERO">OMERO</a> platform, and show how you can use it to work with your microscopy and/or HCS data. In addition we will demonstrate some of the applications that have been released by OME and some of the integration with 3rd party tools, including:
</p>
<ul><li> OMERO.figure - fast figures from your OMERO images</li>
<li> <a href="ImageJ" title="ImageJ">ImageJ</a> - improved interaction with OMERO</li>
<li> <a href="Micro-Manager" title="Micro-Manager">Micro-Manager</a> - how to import directly images to OMERO</li></ul>
<p>We've designed OMERO to be as flexible as possible, and this has enabled its use in a range of imaging domains, including light and electron microscopy, high content screening. Come along to the workshop and bring your favourite data.
</p><p><b>Workshop requirements:</b>
</p>
<ol><li> <a rel="nofollow" class="external text" href="http://downloads.openmicroscopy.org/omero/5.1.3/#clients">OMERO.insight 5.1.3</a> standalone client.</li>
<li> <a rel="nofollow" class="external text" href="http://downloads.openmicroscopy.org/omero/5.1.3/#plugins">OMERO.insight-ij 5.1.3</a> plugin for ImageJ.</li></ol>
<p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140099766/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Matt_McCormick_-_SimpleITK">Matt McCormick - SimpleITK</span></h3>
<p>To quantitatively compare the contents of two images, they must be spatially aligned. Image registration finds the spatial transformation that aligns two images. This workshop presents SimpleITK, the simplified interface to the <a href="Insight_Toolkit" class="mw-redirect" title="Insight Toolkit">Insight Toolkit</a>, to perform 2D and 3D image registration. With this programming library, which interoperates with <a href="ImageJ" title="ImageJ">ImageJ</a>/<a href="Fiji" title="Fiji">Fiji</a>, a spatial transformation between two images can be estimated. This spatial transformation can be quantified to learn about movement or deformation in the images, or it can be used to resample one of the images so that pixel-to-pixel content can be compared. Through interactive examples in IPython notebooks, we will explore and learn about the components of the registration framework and how they work together: transforms, interpolators, similarity metrics, and optimizers.
</p><p><b>Workshop requirements:</b> A web browser and access to the Internet.
</p><p><a rel="nofollow" class="external text" href="http://bit.ly/1hD5FZF">An interactive version of the presentation available at the conference.</a>
</p><p><a rel="nofollow" class="external text" href="https://github.com/InsightSoftwareConsortium/SimpleITKWorkshopImageJ2015">Persistent sources.</a>
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140099763/">Presentation video</a> ]
</p>
<h3><span class="mw-headline" id="Curtis_Rueden_and_Brian_Northan_-_ImageJ2_scripts:_parameters_.2B_ImageJ_Ops">Curtis Rueden and Brian Northan - ImageJ2 scripts: parameters + ImageJ Ops</span></h3>
<p>The <a href="ImageJ2" title="ImageJ2">ImageJ2</a> project updates ImageJ with powerful new features. Two especially useful improvements are: 1) the <a href="ImageJ_Ops" title="ImageJ Ops">ImageJ Ops</a> library, which provides a powerful, unified, extensible and performant collection of image processing routines; and 2) <a href="index.php?title=Scripting_parameters&amp;action=edit&amp;redlink=1" class="new" title="Scripting parameters (page does not exist)">parameterized scripts</a> and macros, which may now declare typed input and output parameters. This mechanism allows scripts to be prototyped more quickly and succinctly, and makes them compatible with other tools beyond ImageJ, including the <a href="KNIME" title="KNIME">KNIME</a> data mining platform, <a href="CellProfiler" title="CellProfiler">CellProfiler</a> image analysis tool, and <a href="OMERO" title="OMERO">OMERO</a> image database. This workshop will walk users through how to write parameterized macros and scripts using ImageJ's <a href="Script_Editor" class="mw-redirect" title="Script Editor">Script Editor</a>, which leverage image processing routines from ImageJ Ops. It will demonstrate how to execute them <a href="Headless" title="Headless">headlessly</a> from the command line, and briefly illustrate how these scripts can be used from other tools in the <a href="SciJava" title="SciJava">SciJava</a> software ecosystem.
</p><p>[ <a rel="nofollow" class="external text" href="http://imagej.github.io/presentations/2015-09-04-imagej2-scripting/">Workshop slides</a> ]
</p><p><b>Workshop requirements:</b> An up-to-date installation of <a href="Fiji" title="Fiji">Fiji</a>
</p><p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098817/">Presentation video</a> ]
[ <a rel="nofollow" class="external text" href="https://vimeo.com/140098835/">Presentation video continued</a> ]
</p>
<h3><span class="mw-headline" id="Aryeh_Weiss1.2C2_-_An_introduction_to_digital_image_processing">Aryeh Weiss<sup>1,2</sup> - An introduction to digital image processing</span></h3>
<p><sup>1</sup>Faculty of Engineering, Bar Ilan University, Ramat Gan 52900 Israel
<br /><sup>2</sup>Bio-imaging Unit, Institute for Life Sciences, Hebrew University, Israel
</p><p>Digital image processing is the tool that allows quantitative and reproducible data to be extracted from the images such as those that are produced in microscopy, forensics, and astronomy. A proper understanding of image processing is required in order to avoid introduction of bias or other artifacts, and also to process the very large datasets that are often produced.
</p><p>In this workshop, the basics of digital images and the stages of the image processing workflow that follow acquisition will be presented and explained. The workshop will cover the following topics:
</p>
<ol><li> Digital image formats and representations</li>
<li> Basic tools for digital image characterization (for example, histogram and intensity line profile).</li>
<li> Image enhancement and noise reduction.</li>
<li> Identification of objects of interest (segmentation).</li>
<li> Analysis and data reduction.</li></ol>
<p>This workshop assumes no prior knowledge of image processing, and is designed to enable the participants to understand the subsequent presentations and talks.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/aryeh-weiss/IntroImageProcessingB.ppt">Workshop slides</a> ]
</p><p><b>Workshop requirements:</b>
</p>
<ul><li> An installation of <a href="ImageJ" title="ImageJ">ImageJ</a> or <a href="Fiji" title="Fiji">Fiji</a></li>
<li> The <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/aryeh-weiss/stk_0001_Result%20of%20cftr-wt1.avi.zip">sample data</a> downloaded to your local computer</li></ul>
<p>[ <a rel="nofollow" class="external text" href="https://vimeo.com/140929692/">Presentation video</a> ]
</p>
<h2><span class="mw-headline" id="Posters">Posters</span></h2>
<p>Posters will be available to browse at any time throughout the conference. There will also be a dedicated poster session on Thursday evening during the reception.
</p>
<h3><span class="mw-headline" id="Ignacio_Arganda-Carreras_-_Trainable_Weka_Segmentation:_a_tool_for_machine-learning-based_image_segmentation">Ignacio Arganda-Carreras - Trainable Weka Segmentation: a tool for machine-learning-based image segmentation</span></h3>
<p>The problem of image segmentation, i.e. partitioning an image into multiple segments, remains unsolved. In recent years, methods incorporating machine learning techniques into the process have emerged as powerful tools, improving the accuracy of detected boundaries or labeled areas. Many different approaches have been proposed based on predefined image features, filters and training sets with "gold standard" provided by humans. However, very often those tools and methods do not have their code open, need to be separately compiled and combined or are not freely available. In this paper we propose a bridge between the machine learning and the image processing worlds. We benefit from combining two of the most popular and powerful platforms of each respective field: the Fiji toolkit, frequently used for biomedical image processing but with a wider spectrum; and the Waikato Environment for Knowledge Analysis (WEKA) suite, which supports several standard data mining tasks, more specifically, data preprocessing, clustering, classification, regression, visualization, and feature selection. Integrated in the same library and graphical interface, they provide a novel and completely open-source framework to combine, evaluate and compare any available learning algorithm to perform general-purpose image segmentation. The source and binary code is completely available and runs on any modern computing platform.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/ignacio-arganda-carreras/poster-TWS-ImageJ-Conf-2015.pdf">Poster</a> ]
</p>
<h3><span class="mw-headline" id="Karl_Bellve_-_Design_and_implementation_of_3D_focus_stabilization_for_fluorescence_microscopy">Karl Bellve - Design and implementation of 3D focus stabilization for fluorescence microscopy</span></h3>
<p>Focus stabilization is critical for many imaging modalities like TIRF, PALM and STORM. The focus stabilization device presented here, named pgFocus, is an open source and open hardware solution that can be integrated into microscopes with an existing objective positioner. pgFocus is a programmable and inexpensive circuit board consisting of a micro-controller, linear sensor array, DAC and an ADC. While pgFocus can stabilize on a single focal plane within ±3nM at 30Hz, it can also follow and correct 3D focus changes when imaging multiple Z positions. pgFocus works by monitoring the reflection of an 808nm laser beam that is internally reflected at a glass/water interface. The translation of the reflected laser beam is converted into Δ distance change between the objective and the glass/water interface. The relationship between movement of the objective and the translation of the return laser beam is determined through an calibration procedure. This Δ distance change measurement is used to modify and adjust a pass-through voltage signal that is directed to a piezo objective positioner. The pass-through voltage is continually adjusted to move the reflected laser beam back to the original focus position.
</p>
<h3><span class="mw-headline" id="Jan_Brocher_-_The_BioVoxxel_Toolbox">Jan Brocher - The BioVoxxel Toolbox</span></h3>
<p>The <a href="BioVoxxel_Toolbox" title="BioVoxxel Toolbox">BioVoxxel Toolbox</a> is a collection of plugins and macros for Fiji and ImageJ which is designed to assist in various image processing and analysis tasks such as lighting and background correction, image filtering, high quality feature extraction as well as binary object analysis. An adaptive filter enables noise removal and flexibility during image filtering due to access to the filter kernel usage. The "Threshold Check" facilitates semi-quantitative assessment of 25 automatic binarization methods to achieve a high quality feature extraction. The "Speckle Inspector" allows an object-wise comparison of small features contained in bigger objects such as biomarker spots in cells or nuclei. Object extraction by secondary features is facilitated by the "Binary Feature Extractor". Analysis and classification of objects by size and shape descriptors is achieved by the "Extended Particle Analyzer" and the color-coded "Shape Descriptor Maps".
</p><p>[ <a rel="nofollow" class="external text" href="https://www.dropbox.com/s/7e6bjh3iqnxv014/IJcon2015-Poster%20Brocher%20%28150dpi%29.png?dl=0">Poster</a> ]
</p>
<h3><span class="mw-headline" id="Matthew_Caputo_-_Size_distribution_of_metal_particles_for_additive_manufacturing_using_image_analysis_techniques_by_ImageJ">Matthew Caputo - Size distribution of metal particles for additive manufacturing using image analysis techniques by ImageJ</span></h3>
<p>Matthew Caputo<sup>1</sup> and C. Virgil Solomon<sup>1,2</sup>
<br /><sup>1</sup> Materials Science &amp; Engineering Program
<br /><sup>2</sup> Department of Mechanical &amp; Industrial Engineering, Youngstown State University, Youngstown OH 44555
</p><p>Additive manufacturing that employs the binder jet method is a process that utilizes micro-meter sized particles to construct parts, layer by layer, until a completed near net shape component is obtained. Three dimensional printed (3DP) parts show better mechanical strength when the material exploits uniform morphology, such as spherical particles. These conditions improve density through packing factors and also reduced binder consumption. Ni-Mn-Ga metal powders produced by the spark erosion process show both dual morphological (spherical and irregular shaped particles) and wide size distribution.  In order to improve spherical particle gain, spark eroded powders were mechanically separated and classified by size (sieving). Spherical particle gain and size distribution was quantified by image analysis of scanning electron micro-graphs of classified spark eroded powders. This poster elaborates on the methods used in the Image J software to complete this procedure. 
</p>
<h3><span class="mw-headline" id="Sherry_Clendenon_-_Image_Acquisition_and_Processing_for_Cell_Based_Model_Instantiation_and_Validation">Sherry Clendenon - Image Acquisition and Processing for Cell Based Model Instantiation and Validation</span></h3>
<p>Sherry G. Clendenon, Maciek Swat, Xiao Fu, Jeffrey L. Clendenon, James Sluka, James Glazier
</p><p>We have developed a standardized method for morphological staining of liver sections that enables identification and segmentation of hepatocytes, leukocytes, endothelial cells, blood vessels and zonation within a lobule. Using this morphological staining, we performed proof of concept segmentation and analysis using ImageJ. Segmented morphological images are 1) the starting point for all subsequent image analysis, 2) essential for direct instantiation of cell based computational models, and then 3) validation of simulation output using the same image analysis tools as used for the biological data. To this end, we are now focused on automation of these proof of concept analyses and development of a workflow pipeline incorporating image storage in OMERO, image analysis in ImageJ, and cell based computational modeling of virtual tissues in Compucell-3D.
</p>
<h3><span class="mw-headline" id="Fabrice_de_Chaumont_-_Icy:_an_open_source_software_for_image_analysis">Fabrice de Chaumont - Icy: an open source software for image analysis</span></h3>
<p>In this poster we present a panel of Icy's features. Live demo based on your images, and we will try to process them thanks to one of the 300 image analysis plugins available in Icy&#160;!
</p><p><span class="Z3988" title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rft.genre=book&amp;rft.title=Icy%3A+an+open+bioimage+informatics+platform+for+extended+reproducible+research&amp;rft.aulast=de+Chaumont&amp;rft.aufirst=Fabrice&amp;rft.date=2012&amp;rft.volume=9%287%29&amp;rft.pages=690-696&amp;rft_id=info:pmid/22743774&amp;rft_id=http%3A%2F%2Fwww.nature.com%2Fnmeth%2Fjournal%2Fv9%2Fn7%2Ffull%2Fnmeth.2075.html"><cite id="CITEREFde_ChaumontDallongevilleChenouardHerv.C3.A92012">de Chaumont, Fabrice; Dallongeville, Stéphane&#32;&amp; Chenouard, Nicolas&#32;et al.&#32;(2012),&#32;"<a rel="nofollow" class="external text" href="http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html">Icy: an open bioimage informatics platform for extended reproducible research</a>",&#32;<i>Nature methods</i>&#32;<b>9(7)</b>:  690-696, PMID 22743774<span class="printonly">, &lt;<a rel="nofollow" class="external free" href="http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html">http://www.nature.com/nmeth/journal/v9/n7/full/nmeth.2075.html</a>&gt;</span></cite></span> (<a rel="nofollow" class="external text" href="http://scholar.google.com/scholar?cluster=8170444028478407032">on Google Scholar</a>).
</p>
<h3><span class="mw-headline" id="Olaya_Fernandez-Gayol_-_Using_the_wrMTrck_plugin_to_analyze_the_Morris_Water_maze_in_mice">Olaya Fernandez-Gayol - Using the wrMTrck plugin to analyze the Morris Water maze in mice</span></h3>
<p>The Morris Water Maze is a standard behavioral test used for assessing spatial memory in mice and rats. The animal must learn the location of a hidden platform in a round pool using external cues. Removing the platform allows for assessment of learning by measuring the time the animal spends in the vicinity of where the platform used to be. A manual quantification can be very time consuming (many trials per animal), less accurate and more time consuming than an automated one. However, imaging software can be very expensive and not readily available to researchers. Here I present a macro that profits from the wrMTrck plugin (developed for C. elegans by JS Pedersen), but which is more focused on the parameters relevant to the analysis of the Morris Water Maze.
</p><p><b>Resources:</b> <a rel="nofollow" class="external text" href="https://dl.dropboxusercontent.com/u/21343536/Fernandez%20Gayol%20O.pptx">Poster</a>, <a rel="nofollow" class="external text" href="https://dl.dropboxusercontent.com/u/21343536/MWM_wrMTrck%20v1.ijm">Macro</a>
</p>
<h3><span class="mw-headline" id="Andreas_Jahnen_-_Segmentation_of_intracranial_aneurysms_in_2D_Digital_Subtraction_Angiography_.28DSA.29_images_using_ImageJ_Tools_and_Plugins">Andreas Jahnen - Segmentation of intracranial aneurysms in 2D Digital Subtraction Angiography (DSA) images using ImageJ Tools and Plugins</span></h3>
<p>Vera Brockmeyer, Christian Moll, Eva Fiorinelli, Damien Bresson, Cécile Legallais, Charbel Mounayer and Andreas Jahnen
</p><p>Some complex intracranial aneurysms are treated using flow diverter stents. Evaluation of blood flow modification inside the aneurysm before and after treatment is mandatory to assess treatment efficiency. We compare 2D DSA pre- and post-stent sequences. The present work describes the interactive aneurysm segmentation algorithm. A combination of several algorithms and ImageJ plugins are used to segment the aneurysm. First the time series is Z projected. The actual segmentation is done using a graph cut algorithm. The Lazy Snapping variant of Yuri Boykov was implemented. An automatic segmentation is implemented with the Statistical Region Merging plugin. Results have been evaluated using the Hausdorff Distance and the Jaccard Coefficient. The combination of the selected Z-projection and application of the graph cut algorithm provides excellent segmentation results. If the user is not satisfied by the result, he can may interactively improve the segmentation. Limitations are on overlapping hyperdense structures (bones, implants, arteries or veins) and artery.
</p>
<h3><span class="mw-headline" id="David_Legland_-_MorphoLibJ:_an_integrated_library_for_morphological_filtering.2C_segmentation_and_analysis_of_plant_tissues">David Legland - MorphoLibJ: an integrated library for morphological filtering, segmentation and analysis of plant tissues</span></h3>
<p>David Legland, Ignacio Arganda-Carreras, Philippe Andrey
</p><p>Understanding the cellular bases of complex phenomena such as plant development and morphogenesis requires the precise determination and quantification of cellular morphology and of tissular organization. Imaging techniques such as 3D confocal microscopy constantly gain in spatial resolution, signal quality and throughput, thus allowing 3D imaging of several dozens or hundreds of cells and of their spatial organization. However, the automated analysis of plant tissues still remains challenging. 
</p><p>The MorphoLibJ library therefore proposes a collection of tools that facilitate the exploitation of 3D images of plant cells. These tools are mostly based on mathematical morphology, including powerful algorithms such as 3D watershed and 3D geodesic reconstruction. Moreover, these algorithms were integrated into user-friendly interfaces to facilitate their appropriation by biologists.
</p><p>[ <a rel="nofollow" class="external text" href="http://conference.imagej.net/2015/ignacio-arganda-carreras/poster_MorphoLibJ.pdf">Poster</a> ]
</p>
<h3><span class="mw-headline" id="David_Logan_-_The_ImageJ-CellProfiler_Interface">David Logan - The ImageJ-CellProfiler Interface</span></h3>
<p>CellProfiler is open-source cell image analysis software designed for biologists without training in computer vision or programming to quantitatively measure phenotypes from thousands of images automatically. Via its RunImageJ module, CellProfiler can load images, run an ImageJ macro or plugin on them, and retrieve the resulting images for downstream analysis. We have now created the converse: an ImageJ plugin that allows CellProfiler pipelines to be called from within ImageJ. This would allow established CellProfiler analysis pipelines to be utilized within an ImageJ workflow. This new functionality will be in our upcoming fall release, is available for preview in our trunk build, and will eventually be deployed with Fiji.
</p><p>[ <a rel="nofollow" class="external text" href="http://www.broadinstitute.org/~dlogan/posters/2015_09_03_ImageJ_Conf_DLogan_FINAL.pptx">Poster slides</a> ]
</p>
<h3><span class="mw-headline" id="Michael_Majurski_-_Accurate_and_scalable_Microscopy_Image_Stitching_Tool_.28MIST.29_2">Michael Majurski - Accurate and scalable Microscopy Image Stitching Tool (MIST)</span></h3>
<p>With new microscope technologies scientists are acquiring terabyte-sized datasets to cover large areas. An automated optical microscope images a large sample by generating a grid of partially overlapping images. This is required because its field of view is much smaller than the dimensions of the specimen being imaged. This process generates a great number of images that need to be stitched into a large mosaic in order to derive meaningful information. A <a href="Stitching" class="mw-redirect" title="Stitching">stitching</a> tool is required to generate these large mosaics. The resulting image can have tens of thousands of pixels per side. Stitching one large image mosaic is computationally taxing and the computation becomes overwhelming when stitching repeated frames of live cell experiments.
</p><p>Optical microscopy often has to process feature-poor images (e.g., sparsely populated cell cultures) and may be used to derive measurements (quantitative information such as counts, densities, intensities, etc.) from such images. We developed a new stitching technique that minimizes the uncertainty due to the lack of features between overlapping areas while being faster than the leading alternative tool in the field. This method optimizes the translations computed by a pairwise image registration method, like the Fourier-based phase correlation image alignment method [1], using the mechanical stage system model. The computed translations are optimized using the Hill Climbing algorithm constrained to a square area of 4x the stage repeatability per side. The optimization function is normalized cross correlation between images.
</p><p>Our algorithm is implemented in Java as an <a href="ImageJ" title="ImageJ">ImageJ</a> plugin which runs on both CPUs and/or GPUs taking advantage of compiled native libraries from FFTW [2], the <a rel="nofollow" class="external text" href="https://developer.nvidia.com/cuda-toolkit">NVIDIA CUDA toolkit</a> using <a rel="nofollow" class="external text" href="http://www.jcuda.org/jcuda/JCuda.html">JCuda</a>, and <a rel="nofollow" class="external text" href="https://developer.nvidia.com/cuFFT">CuFFT</a> using <a rel="nofollow" class="external text" href="http://www.jcuda.org/jcuda/jcufft/JCufft.html">JCufft</a>. The implementation is cognizant of the available computer memory and scales the execution based on the number of CPUs and GPUs on a system. The algorithm can run using as little as 200 MB (on 1040x1392 pixel image tile), though using less memory incurs longer execution times. A representative test case consisting of 16x22 image tiles (1040x1392 pixels) stitches in 24 s on the CPU and 12 s on the GPU using an Intel Xeon CPU E5-2620 @ 2.0 GHz with NVIDIA Quadro 4000 GPU (2048 MB GPU memory). This test case produces a stitched image roughly 20 000 x 21 000 pixels as shown in Figure 1. Figure 2 shows the stitching tool’s ImageJ plugin GUI.
</p><p><b>Acknowledgments:</b> This work has been supported by NIST. We would like to acknowledge the team members of the computational science in biological metrology project at NIST for providing invaluable inputs to our work.
</p>
<ul><li> [1] C. Kuglin and D. Hines, "The Phase Correlation Image Alignement Method," in Proceedings of the 1975 IEEE International Conference on Cybernetics and Society, 1975, pp. 163–165.</li>
<li> [2] M. Frigo and S. Johnson, "The Design and Implementation of {FFTW3}," Proc. IEEE, vol. 93, no. 2, pp. 216–231, 2005.</li></ul>
<h3><span class="mw-headline" id="Jeffrey_Murphy_-_Untitled">Jeffrey Murphy - Untitled</span></h3>
<p>Block copolymer (BCP) thin films can form widely varied patterns based on the self-assembly of nanoscale phase-separated polymer domains. These patterns are reaching industrial application as soft lithographic etch masks for fabrication semiconductor integrated circuits and bit-patterned media. In order for these applications to find success, the patterns require extremely low levels of defectivity and line-edge roughness, plus good uniformity and registration. This is achieved using a combination of ordering techniques: graphoepitaxy, chemoepitaxy, and annealing. Quantification of defects (line edge roughness, orientation, and defectivity) is essential for screening newly developed BCP architectures, annealing methods, and substrate designs. To facilitate this, we have developed an ImageJ tool for the automated analysis of line patterns templated from BCP thin films and have used it to characterize solvent, thermal, and microwave annealing techniques using scanning electron microscope images of BCP patterns on silicon substrates.
</p>
<h3><span class="mw-headline" id="Kelsey_Nemec_-_Untitled">Kelsey Nemec - Untitled</span></h3>
<p>Neurogenesis plays a significant role in learning and memory, and its deficiencies are associated with many neurological diseases. Fragile X syndrome, the most common single gene contributor to mental retardation and autism spectrum disorders, is caused by the absence of the fragile X mental retardation protein (FMRP). Two homologs of FMRP have been identified, FXR1 and FXR2. This project investigates the role of FXR1 on neurogenesis in vivo and tests the hypothesis that Fxr1 deletion leads to a decrease in proliferation of adult neural stem cells (aNSCs), leading to overall decreases in cell populations over the course of neurogenesis. Using an inducible knockout mouse line with a tdTomato reporter (cKO/Nes-CreERT2/tdTom), we were able to delete Fxr1 and track the fate of aNSCs over time, quantifying cell populations using ImageJ. Preliminary data suggest that Fxr1 deletion in aNSCs leads to reduced cell genesis in the adult hippocampus.
</p>
<h3><span class="mw-headline" id="John_Parrish_-_Sperm_nuclear_shape_and_texture_analysis">John Parrish - Sperm nuclear shape and texture analysis</span></h3>
<p>The evaluation of sperm morphology has proven problematic due to variation in what observers term normal or abnormal. We developed methodology within ImageJ to threshold sperm nuclei labeled with a fluorescent DNA stain, output the perimeter coordinates and convert these to a Fourier Series from which harmonic amplitudes quantified components of sperm shape. The harmonic amplitudes were found related to bull fertility and environmental impacts of heat on males. It was also possible to create an average sperm nuclear shape for a male. We have further created a point-spread function from fluorescent microspheres. This was then used to deconvolve the fluorescent image of sperm nuclei to remove out-of-focus light. The resulting images were then subjected to nuclear texture analysis with the ImageJ Texture Analyzer (J. Cabrera). Sperm nuclear texture was related to bull fertility. ImageJ proved beneficial in the evaluation of sperm nuclear morphometry.
</p><p>[ <a rel="nofollow" class="external text" href="http://www.ansci.wisc.edu/jjp1/imageJ/index.html">Software</a> ]
</p>
<h3><span class="mw-headline" id="Tobias_Pietzsch_-_BigDataViewer.E2.80.94Visualization_and_Image_Processing_for_Terabyte_Data_Sets">Tobias Pietzsch - BigDataViewer&#8212;Visualization and Image Processing for Terabyte Data Sets</span></h3>
<p>The necessity to make large volumetric datasets available for interactive visualization and analysis has been widely recognized.  However, existing solutions build upon proprietary file formats requiring that data are copy-converted before visualization, or use dedicated servers to generate virtual slices that are transferred to client applications, practically leading to insufficient frame rates for truly interactive experience. With <a href="BigDataViewer" title="BigDataViewer">BigDataViewer</a> we provide an easily accessible and extensible open source solution for interactive arbitrary virtual re-slicing of very large volumes and time series of volumes.
</p><p><a href="BigDataViewer" title="BigDataViewer">BigDataViewer</a> is a Java library and <a href="Fiji" title="Fiji">Fiji</a> plugin to interactively navigate and visualize large image sequences from both local and remote data sources.  The client software renders an arbitrarily oriented virtual slice through global 3D coordinate space.  Individual image stacks are arbitrarily arranged in the global space, and can be displayed independently or as a color-coded composite.  Brightness and color can be adjusted for each view separately.  The viewer allows free translation, rotation, and zoom for image stacks and moving between timepoints. Thus, multi-terabyte image datasets can be navigated smoothly.
</p><p>[ <a rel="nofollow" class="external text" href="https://www.dropbox.com/s/3st8ezbj9jmjzd8/bdv-poster-imagej2015.pdf?dl=0">Poster</a> ]
</p>
<h3><span class="mw-headline" id="Stephan_Saalfeld_-_ImgLib2_for_visualization_and_generic_image_processing">Stephan Saalfeld - ImgLib2 for visualization and generic image processing</span></h3>
<p><a href="ImgLib2" title="ImgLib2">ImgLib2</a> is an open-source Java library for <i>n</i>-dimensional data representation and manipulation with focus on image processing.  It aims at minimizing code duplication by cleanly separating pixel-algebra, data access and data representation in memory.  Algorithms can be implemented for classes of pixel types and generic access patterns by which they become independent of the specific dimensionality, pixel type and data representation.  <a href="ImgLib2" title="ImgLib2">ImgLib2</a> illustrates that an elegant high-level programming interface can be achieved without sacrificing performance.
</p><p>It provides efficient implementations of common data types, storage layouts, and algorithms.  <a href="ImgLib2" title="ImgLib2">ImgLib2</a> is the data model underlying <a href="ImageJ2" title="ImageJ2">ImageJ2</a>, the <a rel="nofollow" class="external text" href="https://tech.knime.org/community/image-processing">KNIME Image Processing toolbox</a>, the <a href="BigDataViewer" title="BigDataViewer">BigDataViewer</a>, and an increasing number of Fiji-Plugins.
</p><p><a href="ImgLib2" title="ImgLib2">ImgLib2</a> is licensed under <a rel="nofollow" class="external text" href="https://github.com/imglib/imglib2/blob/master/LICENSE.txt">BSD</a>.  Documentation and source code are available at <a rel="nofollow" class="external free" href="http://imglib2.net">http://imglib2.net</a> and in a public source code repository at <a rel="nofollow" class="external text" href="https://github.com/imglib/imglib2">GitHub</a>.
</p><p>[ <a rel="nofollow" class="external text" href="https://www.dropbox.com/s/f23tr3q607dxvlh/imglib2-imagej2015.pdf?dl=0">Poster</a> ]
</p>
<h3><span class="mw-headline" id="Md_Abdul_Kader_Sagar_-_The_SLIM_Curve_plugin_for_lifetime_image_analysis">Md Abdul Kader Sagar - The SLIM Curve plugin for lifetime image analysis</span></h3>
<p>The <a href="SLIM_Curve" title="SLIM Curve">SLIM Curve</a> plugin for ImageJ is an ImageJ plugin for lifetime image analysis which uses the SLIM Curve library. SLIM Curve is an exponential curve fitting library used for Fluorescent Lifetime Imaging (FLIM) and Spectral Lifetime Imaging (SLIM). This time in excited state is affected by its chemical microenvironment such as proximity to other fluorophores, pH and hydrophobic regions. This ability has made FLIM a popular and unique tool for studies ranging from metabolic measurement to measuring distances between proteins. FLIM allows us to calculate how long a fluorophore stays in excited state. The need for analyzing different type of FLIM images from different system either independently or as part of an overall image analysis workflow was the motivation for development of open source SLIM Curve library.SLIM Curve Plugin uses rapid lifetime determination (RLD), Levenberg-Marquardt (LM) or hybrid (RLD+LM) fitting algorithms and can perform Single, double and triple exponential fits, as well as stretched exponential fits.
</p>
<h3><span class="mw-headline" id="Raghavender_Sahdev_-_TimeLapseReg_.E2.80.93_An_ImageJ_plugin_for_drift_correction_of_video_sequence_in_time-lapse_microscopy_2">Raghavender Sahdev - TimeLapseReg – An ImageJ plugin for drift correction of video sequence in time-lapse microscopy</span></h3>
<p>Raghavender Sahdev 1, Tomasz Konopczynski 2, Dimiter Prodanov 2,3 and Daniel Sage 4
</p><p>1) BITS Pilani Hyderabad Campus, India, 2) Belgian National INCF Node, 3) EHS/NERF, IMEC, Belgium 4) Biomedical imaging group, EPFL Switzerland
</p><p>The main functionality of the Turboreg plugin allows for aligning or matching two images. The plugin is widely used in Neuroscience for pre-processing of both static and time-lapse imaging data. In this project we develop an ImageJ's plugin to be used by an end-user to correct for the drift in very long sequences of noisy multi-channel fluorescence images. TimeLapseReg uses the well-known TurboReg/StackReg engine method to register frames to a reference image. It computes and applies rigid transformations of the user-selected channel and stores them for subsequent use. The geometrical transformations are further applied to all channels. We present the user with a drift trajectory over the entire image sequence. The user has the option of choosing the reference image and discarding corrupted images. We have applied our software to calcium imaging data for helping neuro-scientists to better infer neuronal network spiking activity.
</p><p>[ <a rel="nofollow" class="external text" href="http://www.raghavendersahdev.com/uploads/3/9/6/2/39623741/timelapsereg.pdf">Poster slides</a>, <a rel="nofollow" class="external text" href="http://neuroinformatics.be/projects/stackreg-plus-improved-multiple-image-alignment-with-imagej/">TimeLapseReg web page</a>, <a rel="nofollow" class="external text" href="https://github.com/incfbelgiannode/StackReg-Plus_Multiple-Image-Alignment">TimeLapseReg source code</a> ]
</p>
<h3><span class="mw-headline" id="Joey_Tidei_-_Untitled">Joey Tidei - Untitled</span></h3>
<p>The mammalian ELAV-like protein HuD is a neuronal RNA-binding protein implicated in neuronal development, plasticity, and diseases. Although HuD has long been associated with neuronal development, the function of HuD in neural stem cell differentiation and the underlying mechanisms have gone largely unexplored. Here we show that HuD promotes neuronal differentiation of neural stem/progenitor cells (NSCs) in the adult subventricular zone by stabilizing the mRNA of special AT-rich DNA-binding protein 1 (SATB1), a critical transcriptional regulator in neurodevelopment. We find that SATB1 deficiency impairs the neuronal differentiation of NSCs, whereas SATB1 overexpression rescues the neuronal differentiation phenotypes resulting from HuD deficiency. Interestingly, we also discover that SATB1 is a transcriptional activator of HuD during NSC neuronal differentiation. In addition, we demonstrate that NeuroD1, a neuronal master regulator, is a direct downstream target of SATB1. Therefore, HuD and SATB1 form a positive regulatory loop that enhances NeuroD1 transcription and subsequent neuronal differentiation. Our results here reveal a novel positive feedback network between an RNA-binding protein and a transcription factor, which plays critical regulatory roles in neurogenesis.
</p>
<h3><span class="mw-headline" id="Jean-Yves_Tinevez_-_TrackMate">Jean-Yves Tinevez - TrackMate</span></h3>
<p>We present TrackMate, a Fiji plugin for the automated, semi-automated and manual tracking of single-particles. Its development focuses on achieving two concomitant goals. First it aims at offering a generic solution that works out of the box for end users, through a simple and sensible user interface. It operates indifferently for 1D, 2D or 3D, multi-channel images and provides several visualization and analysis tools that help assessing the relevance of results. Second, because there is not a universal tracking algorithm that can be optimal for all the tracking challenges met in Life Science, it is a platform where developers can easily write their own detection, particle-linking, visualization or analysis module for TrackMate. The goal is to provide a framework for researchers that wants to develop a new algorithm without the painful burden to also write a GUI, several visualization tools, analysis tools and exporting facilities, by reusing what is already there in TrackMate.
</p>
<h3><span class="mw-headline" id="Amitabh_Verma_-_Acquisition.2C_Processing_.26_Re-Processing_using_OpenPolScope_plugin_for_Micro-Manager_2">Amitabh Verma - Acquisition, Processing &amp; Re-Processing using OpenPolScope plugin for Micro-Manager</span></h3>
<p><a rel="nofollow" class="external text" href="http://www.openpolscope.org/">OpenPolScope</a> software provides a suite of plugins for <a href="Micro-Manager" title="Micro-Manager">Micro-Manager</a> and extends its capabilities of acquisition and adds features of processing and reprocessing. The plugin also provides a wrapper to the ImageJ <a href="Macro" class="mw-redirect" title="Macro">macro</a> language which can then be applied to a Micro-Manager acquisition stream. OpenPolScope architecture is based on plugins called Mode Plugins and is thus extensible and can be applied to a number of imaging techniques that require on-the-fly processing. The reprocessing plugin is ImageJ based using Micro-Manager libraries to access datasets. The OpenPolScope API provides a common processing pipeline for both the Micro-Manager acquisition and ImageJ re-processing plugin. Here we demonstrate the OpenPolScope plugin and API using Lc-PolScope based Birefringence imaging where consecutive images are acquired (as channels) modulating a liquid crystal between the microscopes light-path. A resultant Retardance image is then computed on-the-fly and inserted in the acquisition stream as an additional channel.
</p><p>[ <a rel="nofollow" class="external text" href="http://openpolscope.org/pages/OpenPolScopeInstaller.htm">OpenPolScope downloads</a> ]
</p>
<h3><span class="mw-headline" id="Thorsten_Wagner_-_Tools_and_methods_of_the_Biomedical_Imaging_Group_for_ImageJ:_An_Overview">Thorsten Wagner - Tools and methods of the Biomedical Imaging Group for ImageJ: An Overview</span></h3>
<p>The Biomedical Imaging Group (BIG) is part of the Dept. of Computer Sciences at University of Applied Sciences Dortmund (Germany) and cooperates with several national and international institutions. Within the context of a number of national and international funded projects BIG develops tools and methods for imagej to find solutions for various scientific questions. This work gives an overview of the available (or soon available tools) of the Biomedical Imaging Group. This includes tools for nanoparticle tracking analysis, shape filtering, connected component analysis, line detection and noise filtering.
</p><p>[ <a rel="nofollow" class="external text" href="https://dl.dropboxusercontent.com/u/560426/imagej/conference2015/poster.pdf">Poster</a> ]
</p>
<!-- 
NewPP limit report
Cached time: 20200713064911
Cache expiry: 86400
Dynamic content: false
CPU time usage: 0.104 seconds
Real time usage: 0.106 seconds
Preprocessor visited node count: 1663/1000000
Preprocessor generated node count: 13373/1000000
Post‐expand include size: 22617/2097152 bytes
Template argument size: 4642/2097152 bytes
Highest expansion depth: 13/40
Expensive parser function count: 0/3
-->

<!-- 
Transclusion expansion time report (%,ms,calls,template)
100.00%   66.963      1 - -total
 90.37%   60.512      3 - Template:Publication
 68.81%   46.080      3 - Template:Citation
 59.09%   39.571      3 - Template:Citation/core
  5.97%    3.997      1 - Template:Bc
  3.82%    2.556      3 - Template:Link
  2.07%    1.387      3 - Template:Scholar
  1.88%    1.256      1 - Template:Clear
  1.78%    1.190      1 - Template:Arrow
-->
</div><div class="printfooter">
Retrieved from "<a dir="ltr" href="index.php?title=Conference_2015_Program&amp;oldid=21068">http://imagej.net/index.php?title=Conference_2015_Program&amp;oldid=21068</a>"</div>
							</div>

			<div id="footer">
				<p> This page was last modified on 9 July 2016, at 10:25.</p><ul><li><a href="ImageJ:Privacy_policy" title="ImageJ:Privacy policy">Privacy policy</a></li><li><a href="ImageJ:About" class="mw-redirect" title="ImageJ:About">About ImageJ</a></li><li><a href="Imprint" title="Imprint">Imprint</a></li><li><a href="index.php?title=Conference_2015_Program&amp;mobileaction=toggle_view_mobile" class="noprint stopMobileRedirectToggle">Mobile view</a></li></ul>			</div>

			<div id="catlinks" class="catlinks" data-mw="interface"><div id="mw-normal-catlinks" class="mw-normal-catlinks"><a href="Special:Categories" title="Special:Categories">Category</a>: <ul><li><a href="Category:Conference" title="Category:Conference">Conference</a></li></ul></div></div>		</div>

		<div id="bottom-wrap">
		</div>
		<script>(window.RLQ=window.RLQ||[]).push(function(){mw.loader.load(["ext.fancytree","ext.suckerfish","mediawiki.toc","mediawiki.action.view.postEdit","site","mediawiki.user","mediawiki.hidpi","mediawiki.page.ready","mediawiki.searchSuggest","ext.SimpleTooltip"]);});</script><script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgBackendResponseTime":323});});</script>
		</body>
		</html>
		