<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">
<head>
<meta charset="UTF-8"/>
<title>Elastic Alignment and Montage - ImageJ</title>
<script>document.documentElement.className = document.documentElement.className.replace( /(^|\s)client-nojs(\s|$)/, "$1client-js$2" );</script>
<script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":0,"wgPageName":"Elastic_Alignment_and_Montage","wgTitle":"Elastic Alignment and Montage","wgCurRevisionId":41233,"wgRevisionId":41233,"wgArticleId":1242,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Plugins","Registration","TrakEM2","Citable"],"wgBreakFrames":false,"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRelevantPageName":"Elastic_Alignment_and_Montage","wgRelevantArticleId":1242,"wgRequestId":"1910bf3063028bcfc6625549","wgIsProbablyEditable":false,"wgRestrictionEdit":[],"wgRestrictionMove":[],"wgPreferredVariant":"en","fancytree_path":"/extensions/TreeAndMenu/fancytree"});mw.loader.state({"site.styles":"ready","noscript":"ready","user.styles":"ready","user.cssprefs":"ready","user":"ready","user.options":"loading","user.tokens":"loading","ext.cite.styles":"ready","mediawiki.legacy.shared":"ready","mediawiki.legacy.commonPrint":"ready","mediawiki.sectionAnchor":"ready","skins.erudite":"ready"});mw.loader.implement("user.options@0j3lz3q",function($,jQuery,require,module){mw.user.options.set({"variant":"en"});});mw.loader.implement("user.tokens@1ku9xth",function ( $, jQuery, require, module ) {
mw.user.tokens.set({"editToken":"+\\","patrolToken":"+\\","watchToken":"+\\","csrfToken":"+\\"});/*@nomin*/;

});mw.loader.load(["mediawiki.page.startup"]);});</script>
<link rel="stylesheet" href="load.php%3Fdebug=false&amp;lang=en&amp;modules=ext.cite.styles%257Cmediawiki.legacy.commonPrint%252Cshared%257Cmediawiki.sectionAnchor%257Cskins.erudite&amp;only=styles&amp;skin=erudite.css"/>
<link rel="stylesheet" href="extensions/TreeAndMenu/fancytree/fancytree.css"/><link rel="stylesheet" href="extensions/TreeAndMenu/suckerfish/suckerfish.css"/>
<meta name="ResourceLoaderDynamicStyles" content=""/>
<link rel="stylesheet" href="load.php%3Fdebug=false&amp;lang=en&amp;modules=site.styles&amp;only=styles&amp;skin=erudite.css"/>
<meta name="generator" content="MediaWiki 1.28.0"/>
<meta name="description" content="Please note that the elastic alignment and montage plugin available through Fiji, is based on a publication. If you use it successfully for your research please cite our work:"/>
<link rel="shortcut icon" href="skins/ij2.ico"/>
	<meta property="og:type" content="article"/>

	<meta property="og:site_name" content="ImageJ"/>

	<meta property="og:title" content="Elastic Alignment and Montage"/>

	<meta property="og:description" content="Please note that the elastic alignment and montage plugin available through Fiji, is based on a publication. If you use it successfully for your research please cite our work:"/>


<meta name="viewport" content="width=device-width, initial-scale=1" />
</head>
<body class="mediawiki ltr sitedir-ltr mw-hide-empty-elt ns-0 ns-subject page-Elastic_Alignment_and_Montage rootpage-Elastic_Alignment_and_Montage skin-erudite action-view">
		<div class="mw-jump">
			<a href="Elastic_Alignment_and_Montage.html#bodyContent">Skip to content</a>, 			<a href="Elastic_Alignment_and_Montage.html#search">Skip to search</a>
		</div>

		<div id="top-wrap" role="banner">
			<h1><a href="Welcome" title="ImageJ" rel="home">ImageJ</a></h1>
			<div id="tagline">From ImageJ</div>

			<a id="menubutton" href="Elastic_Alignment_and_Montage.html#menu">Menu</a>
			<div id="nav" role="navigation">
			<ul id='menu'>
<li id="menu-item-n-About"><a href="ImageJ">About</a></li>
<li id="menu-item-n-Downloads"><a href="Downloads">Downloads</a></li>
<li id="menu-item-n-Learn"><a href="Learn">Learn</a></li>
<li id="menu-item-n-Develop"><a href="Development">Develop</a></li>
<li id="menu-item-n-News"><a href="News">News</a></li>
<li id="menu-item-n-Events"><a href="Events">Events</a></li>
<li id="menu-item-n-Help"><a href="Help">Help</a></li>
</ul>
			</div>
		</div>

		<div id="mw-js-message"></div>
		
		<div id="main" role="main">

			<div id="bodyContent">
				<h1>Elastic Alignment and Montage</h1>
				
				<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><table class="infobox" cellspacing="5" style="max-width: 31em; font-size: 90%; text-align: left; float: right; border: 1px solid #a0a0a0;">
<tr>
<th colspan="2" style="text-align: center; font-size: 130%;"> <div style="float: left"></div>Elastic Align and Montage (Fiji)<div style="clear: left;"></div>
</th></tr>
<tr>
<th> Author
</th>
<td> <a href="./User:Saalfeld" class="mw-redirect" title="User:Saalfeld">Stephan Saalfeld</a> (<a rel="nofollow" class="external autonumber" href="mailto:saalfeld@mpi-cbg.de">[1]</a>)
</td></tr>




<tr>
<th> Maintainer
</th>
<td> <a href="./User:Saalfeld" class="mw-redirect" title="User:Saalfeld">Stephan Saalfeld</a>
</td></tr>


<tr>
<th> Source
</th>
<td> <a rel="nofollow" class="external autonumber" href="https://fiji.sc/cgi-bin/gitweb.cgi?p=mpicbg.git;a=tree;f=mpicbg/ij/plugin">[2]</a>
</td></tr>
<tr>
<th> Initial release
</th>
<td> March 11<sup>th</sup>, 2011
</td></tr>
<tr>
<th> Latest version
</th>
<td> October 22<sup>nd</sup>, 2011
</td></tr>
<tr>
<th> Development status
</th>
<td> experimental, active
</td></tr>
<tr>
<th> Category
</th>
<td> <a href="./Category:Plugins" title="Category:Plugins">Plugins</a>, <a href="./Category:Registration" title="Category:Registration">Registration</a>
</td></tr>

</table>
<div id="toc" class="toc"><div id="toctitle"><h2>Contents</h2></div>
<ul>
<li class="toclevel-1"><a href="Elastic_Alignment_and_Montage.html#Citation"><span class="tocnumber">1</span> <span class="toctext">Citation</span></a></li>
<li class="toclevel-1 tocsection-1"><a href="Elastic_Alignment_and_Montage.html#Introduction"><span class="tocnumber">2</span> <span class="toctext">Introduction</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="Elastic_Alignment_and_Montage.html#Elastic_Deformation_with_Spring_Meshes"><span class="tocnumber">3</span> <span class="toctext">Elastic Deformation with Spring Meshes</span></a></li>
<li class="toclevel-1 tocsection-3"><a href="Elastic_Alignment_and_Montage.html#Linear_Initialization"><span class="tocnumber">4</span> <span class="toctext">Linear Initialization</span></a></li>
<li class="toclevel-1 tocsection-4"><a href="Elastic_Alignment_and_Montage.html#Block_Matching"><span class="tocnumber">5</span> <span class="toctext">Block Matching</span></a></li>
<li class="toclevel-1 tocsection-5"><a href="Elastic_Alignment_and_Montage.html#Parameter_Discussion"><span class="tocnumber">6</span> <span class="toctext">Parameter Discussion</span></a></li>
<li class="toclevel-1 tocsection-6"><a href="Elastic_Alignment_and_Montage.html#Future_Work"><span class="tocnumber">7</span> <span class="toctext">Future Work</span></a></li>
<li class="toclevel-1 tocsection-7"><a href="Elastic_Alignment_and_Montage.html#References"><span class="tocnumber">8</span> <span class="toctext">References</span></a></li>
</ul>
</div>

<h2><span class="mw-headline" id="Citation">Citation</span></h2>
<p>Please note that the elastic alignment and montage plugin available through Fiji, is based on a publication. If you use it successfully for your research please cite our work:
</p><p>S. Saalfeld, R. Fetter, A. Cardona and P. Tomancak (2012) "Elastic volume reconstruction from series of ultra-thin microscopy sections, <i>Nature Methods</i>, 9(7), 717-720 <a rel="nofollow" class="external text" href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.2072.htmll">Webpage</a> <a href="_images/1/14/Saalfeld_nmeth.2072.pdf" class="internal" title="Saalfeld nmeth.2072.pdf">PDF</a> <a href="_images/f/f3/Saalfeld_nmeth.2072-S1.pdf" class="internal" title="Saalfeld nmeth.2072-S1.pdf">Supplement</a>
</p><p>Supplementary videos demonstrating the performance of the method are available <a rel="nofollow" class="external text" href="http://fly.mpi-cbg.de/elastic">here</a>.
</p>
<h2><span class="mw-headline" id="Introduction">Introduction</span></h2>
<div class="thumb tright"><div class="thumbinner" style="width:168px;"><a href="./File:Aligned-series-crop-512.gif" class="image"><img alt="" src="_images/b/ba/Aligned-series-crop-512.gif" width="166" height="166" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:Aligned-series-crop-512.gif" class="internal" title="Enlarge"></a></div>Example 2: Example for elastic alignment and montaging.  7 serial TEM sections of the neuropil of a <i>Drosophila melanogaster</i> first instar larva, detail at 100% pixel resolution. Image courtesy of <a href="index.php?title=Template:Person:Albertcardona&amp;action=edit&amp;redlink=1" class="new" title="Template:Person:Albertcardona (page does not exist)">Template:Person:Albertcardona</a>.</div></div></div><div class="thumb tright"><div class="thumbinner" style="width:168px;"><a href="./File:Aligned-series-512.gif" class="image"><img alt="" src="_images/d/d9/Aligned-series-512.gif" width="166" height="166" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:Aligned-series-512.gif" class="internal" title="Enlarge"></a></div>Example 1: Example for elastic alignment and montaging.  7 serial TEM sections of the neuropil of a <i>Drosophila melanogaster</i> first instar larva, downscaled by a factor of 12.  Image courtesy of <a href="index.php?title=Template:Person:Albertcardona&amp;action=edit&amp;redlink=1" class="new" title="Template:Person:Albertcardona (page does not exist)">Template:Person:Albertcardona</a>.</div></div></div>
<p>We describe here our elastic alignment method for series or groups of overlapping 2d-images.  The method is accessible through the plugins <b>Elastic Stack Alignment</b> and <b>Elastic Montage</b> and incorporated in the <b><a href="TrakEM2" title="TrakEM2">TrakEM2</a></b> software.  Applications are:
</p>
<dl><dt>Elastic Montage</dt>
<dd>montaging mosaics from overlapping tiles where the tiles have non-linear relative deformation</dd>
<dt>Elastic Stack Alignment</dt>
<dd>alignment of deformed section series from serially sectioned volumes</dd>
<dt>TrakEM2</dt>
<dd>incorporates both elastic montaging and elastic series alignment through the alignment menu for series alignment and montaging of large multi-tile section series</dd></dl>
<p>Images are warped such that corresponding regions overlap optimally.  The warp for each single image is calculated such that each image in the whole global montage or series will be deformed minimally.  That way, arbitrarily large series of images can be aligned without accumulating artificial warps.
</p>
<h2><span class="mw-headline" id="Elastic_Deformation_with_Spring_Meshes">Elastic Deformation with Spring Meshes</span></h2>
<div class="thumb tright"><div class="thumbinner" style="width:168px;"><a href="./File:Mesh.png" class="image"><img alt="" src="_images/8/8b/Mesh.png" width="166" height="157" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:Mesh.png" class="internal" title="Enlarge"></a></div>Fig. 1: Triangular section mesh with a resolution of 5 vertices per each long row.</div></div></div>
<p>We achieve this globally minimized deformation by simulating the alignment as an elastic system of spring connected vertices.  Zero-length springs connect corresponding locations between two overlapping images and warp the images towards perfect overlap.  Non-zero length springs within the image preserve each images shape at locally rigid transformation.  That way, the system penalizes arbitrary warp and distributes the deformation evenly among all images.
</p><p>Each image is tessellated into a mesh of regular triangles with each vertex being connected to its neighboring vertices by a spring (see Fig. 1).  A triangle of springs has two families of cost minima in the plane: 1) at rigid transformations and 2) at rigid transformations flipped.  That is, for all local deformations smaller than the size of a triangle, the mesh will drag towards a rigid transformation.  For larger deformation, it may fold.  The vertices of a triangle define an affine transformation for all pixels in the triangle.
</p><p>For those vertices of a source image overlapping a target image, we identify its corresponding location in the target image.  The vertex is then connected into the target mesh by a zero-length spring with its target end being located at an arbitrary place in a triangle of the target mesh.  This `passive' end does not contribute to the deformation of the target mesh.  During simulation, it moves according to the affine transformation defined by the target triangle.  Vice versa, vertices of the target image are connected to their corresponding location in the source image with their `passive' ends being moved by the respective affine transformation of a source triangle.
</p>
<h2><span class="mw-headline" id="Linear_Initialization">Linear Initialization</span></h2>
<p>Both relaxing the system of meshes and identifying corresponding locations between images require good initialization.  The former because the meshes will fold otherwise, the latter to narrow the matching space with the benefit of both, increased speed and better reliability.  We initialize the system with a linear-per-image optimal pre-alignment based on <a href="Feature_Extraction" title="Feature Extraction">local image features</a>.<sup id="cite_ref-SaalfeldAl2010_1-0" class="reference"><a href="Elastic_Alignment_and_Montage.html#cite_note-SaalfeldAl2010-1">&#91;1&#93;</a></sup>
</p>
<h2><span class="mw-headline" id="Block_Matching">Block Matching</span></h2>
<div class="thumb tright"><div class="thumbinner" style="width:363px;"><a href="./File:Correlation.png" class="image"><img alt="" src="_images/3/35/Correlation.png" width="361" height="211" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:Correlation.png" class="internal" title="Enlarge"></a></div>Fig. 2: Match filter based on the correlation surface. Starting from an approximate alignment (e.g. affine), for each vertex of the spring mesh, an offset is searched calculating the PMCC coefficient r of a block at all possible x,y translations in a given local vicinity over the overlapping image. The above windows display six examples of such correlation surfaces for translations in a square region with the origin in the middle. The <a href="http://en.wikipedia.org/wiki/Pearson_product-moment_correlation_coefficient" class="extiw" title="wikipedia:Pearson product-moment correlation coefficient">PMCC coefficent</a> r is grey-coded in the range from -0.7 to 0.7. The candidate for the translational offset is the translation with maximal r. Candidates are rejected if either r was too low (not similar), there was more than one maximum with very similar r (ambiguous), the maximum is not well localized in both dimensions (an edge pattern that fits everywhere alongside the edge).</div></div></div>
<p>Corresponding locations are searched through block matching.  Initialized from an approximate linear pairwise alignment that is estimated using <a href="Feature_Extraction" title="Feature Extraction">local image features</a>, the local vicinity around each vertex is inspected for an optimal match.  We use the The <a href="http://en.wikipedia.org/wiki/Pearson_product-moment_correlation_coefficient" class="extiw" title="wikipedia:Pearson product-moment correlation coefficient">PMCC coefficent</a> <i>r</i> of a patch around the vertex and the overlapping patch in the other image as the quality measure for a match.  The location with maximal <i>r</i> specifies the offset of the vertex relative to the initial linear alignment.
</p><p>We perform block matching at a reasonably down-scaled version of the images.  The ideal scaling factor depends on the application and quality of the signal.  To overcome the reduced accuracy of the estimated offset, we use Brown's method<sup id="cite_ref-BrownL2002_2-0" class="reference"><a href="Elastic_Alignment_and_Montage.html#cite_note-BrownL2002-2">&#91;2&#93;</a></sup> to estimate an approximate sub-pixel offset.  Furthermore, in order to reject wrong matches, three local filters based on the correlation surface are in place:
</p>
<dl><dt>minimal threshold for <i>r</i></dt>
<dd>if the match has an <i>r</i> lower than a given threshold it is rejected</dd>
<dt>edge response filter</dt>
<dd>if the principal curvatures of <i>r(x,y)</i> at matches location are related by a factor larger than a given threshold, the match is rejected for being an edge response.<sup id="cite_ref-Lowe2004_3-0" class="reference"><a href="Elastic_Alignment_and_Montage.html#cite_note-Lowe2004-3">&#91;3&#93;</a></sup></dd>
<dt>ambiguity filter</dt>
<dd>if the second best <i>r(x,y)</i> is very similar to the best, the match is rejected for being ambiguous.  Similar means related by a factor larger than a given threshold.<sup id="cite_ref-Lowe2004_3-1" class="reference"><a href="Elastic_Alignment_and_Montage.html#cite_note-Lowe2004-3">&#91;3&#93;</a></sup></dd></dl>
<p>See also <a href="Test_Block_Matching_Parameters" title="Test Block Matching Parameters">Test Block Matching Parameters</a> plugin
</p>
<h2><span class="mw-headline" id="Parameter_Discussion">Parameter Discussion</span></h2>
<p>All parameters that specify a distance in pixels refer to the original scale of the image.
</p>
<div class="thumb tright"><div class="thumbinner" style="width:249px;"><a href="./File:Edge-filter.png" class="image"><img alt="" src="_images/7/77/Edge-filter.png" width="247" height="53" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:Edge-filter.png" class="internal" title="Enlarge"></a></div>Fig. 3: Edge response filter. The ratio of the two principal curvatures (Hessian eigenvalues) of at a detection determines how well it is defined in both dimensions. A large ratio signalizes an edge response.</div></div></div>
<dl><dt>Input</dt>
<dd>Both plugins work with stacks of images.  The stacks might be virtual, which is strongly suggested for very large images.</dd></dl>
<dl><dt>Output</dt>
<dd>The plugins export their output as a file or a series of files respectively.  In addition, previously extracted <a href="Feature_Extraction" title="Feature Extraction">local features</a> and <a href="Feature_Extraction" title="Feature Extraction">feature correspondences</a> are saved for later re-use during parameter triggering (thanks to <a href="./User:Albertcardona" title="User:Albertcardona">Albert Cardona</a> for adding this).  Make sure that you work in a clear folder when changing the input data because both saved matches and features are identified by their stack index and parameters only.</dd>
<dd>You can choose whether to <b>interpolate</b> the result or not.  <b>Visualize</b> checked will render the spring mesh simulation into a 512&#215;512 pixel stack for demonstration purposes.  The result is rendered using a transform mesh similar to the spring mesh discussed above.  The parameter <b>resolution</b> specifies the number of vertices in a long row for this mesh, higher numbers give smoother results.  Usually, you will never need more than <b>128</b>.  Optionally, the result can be <b>rendered as RGB with green background</b> to explicitly mark empty background pixels e.g. when generating montages for successive series alignment.</dd></dl>
<dl><dt>Block Matching</dt>
<dd>Here you specify the <b>scale</b> factor at which matching is performed.  Choose this size such that 1) high level noise is effectively invisible, 2) for series alignment such that a pixel is approximately as large as (or larger than) the section thickness.  All further distances are specified in pixels of the <i>original image size</i>.  The <b>search radius</b> for block matching must include the largest non-linear deformation expected relative to an approximate linear alignment.  At the same time it should be kept as small as possible to increase speed and to prevent false matches.  Similarly, the <b>block radius</b> should be large enough to include recognizable texture but not too large.  Usually, setting both radii to similar values is a good choice. <b>Resolution</b> specifies the number of vertices in a long row of the spring mesh as depicted in Fig. 1.  <a href="TrakEM2" title="TrakEM2">TrakEM2</a> does not require all image tiles in a montage to have the same size.  Therefore, it asks for the side-length of a triangle in the mesh to be specified.  That way, the individual mesh resolution for each image can be calculated such that all tiles are modeled by meshes of approximately equal resolution.</dd></dl>
<div class="thumb tright"><div class="thumbinner" style="width:363px;"><a href="./File:Filters.jpg" class="image"><img alt="" src="_images/c/cb/Filters.jpg" width="361" height="260" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:Filters.jpg" class="internal" title="Enlarge"></a></div>Fig. 4: The effect of block matching filters. (ssTEM image data courtesy of <a rel="nofollow" class="external text" href="http://janelia.org/people/scientist/richard-fetter">Richard Fetter</a>)</div></div></div>
<dl><dt>Correlation Filters</dt>
<dd>The threshold for <b>minimal PMCC <i>r</i></b> can be higher for montaging (same signal) than for series alignment (changing signal).  Higher values will lead to more matches rejected and thus less false positives.  The <b>maximal curvature ratio</b> is the threshold for edge responses as depicted in Figs. 2 and 3.  The value must be &gt;1.0.  Higher values will accept more matches alongside elongated structures and thus lead to potentially more false positives.  <b>maximal second best <i>r</i> / best <i>r</i></b> is the maximal threshold for non-ambiguous detections.  Higher values will accept more potentially ambiguous matches that may be false positives.</dd>
<dt>Local Smoothness Filter</dt>
<dd>Particularly for <i>series alignment</i>, the local filters based on block correlation will either reject too many matches (false negatives) or let some false matches pass (false positives).  Since loosing correct matches will degrade the quality of the alignment, we want to keep the false negatives rate minimal by using only moderate local correlation filters.  The remaining false positives can be identified and rejected exploiting the smoothness of the deformation field (Fig. 4).  The local smoothness filter inspects each match and compares how well the estimated translational offset agrees with all other matches weighted by their distance to the inspected match.  To that end, a local linear transformation (typically rigid) is calculated using weighted least squares.  The weight for each match is defined by a <a href="http://en.wikipedia.org/wiki/Gaussian_function" class="extiw" title="wikipedia:Gaussian function">Gaussian</a> <a href="http://en.wikipedia.org/wiki/Radial_distribution_function" class="extiw" title="wikipedia:Radial distribution function">radial distribution function (RDF)</a> centered at the reference match.  A match is rejected if its transfer error relative to the estimated linear transformation is larger than an <b>absolute</b> threshold or larger than <i>k</i>&#215; the average transfer error of all weighted matches (<i>k</i> is specified in the <b>relative</b> field). </dd>
<dt>Miscellaneous</dt>
<dd>ImageJ lacking support for alpha-channels, you can choose to ignore perfectly green (rgb:0,255,0) pixels during <i>series alignment</i>.  This is a hack to cope with montages not covering the whole canvas.  <a href="TrakEM2" title="TrakEM2">TrakEM2</a> supports an alpha-channel such that background pixels are automatically treated correctly.  Both <a href="TrakEM2" title="TrakEM2">TrakEM2</a> and <b>Elastic Stack Alignment</b> offer to consider a series or montage as pre-aligned (<b>checkbox</b>) by which the feature based pre-alignment step can be skipped.  For series alignment, the range of layers to be compared can be limited.</dd></dl>
<dl><dt>Feature Extraction and Geometric Filters</dt>
<dd>Find documentation at the <a href="Feature_Extraction" title="Feature Extraction">Feature Extraction</a> page.  The <b>approximate transformation</b> model as specified in this dialog is used for geometric consensus filter and pairwise pre-alignment only.  It should approximate the actual deformation as closely as possible.  That is, you should select an <b>affine transformation</b> in almost any case. </dd>
<dd>In the <i>montage plugin</i>, all images are compared to each other.  This is prohibitive for montages consisting of a large number of images.  For large montages, use <a href="TrakEM2" title="TrakEM2">TrakEM2</a> where prior information can be used to limit the comparison to overlapping images only.  For <i>series alignment</i>, we try to establish matches not only to adjacent section in the series but to as many sections as possible in the previously constrained range (e.g. up to 10 sections).  Since the similarity of two section decreases with increasing distance, we can stop matching sections after we have failed to compare closer partners.  Still, we want to be able to bridge sections with bad quality or exceptional artifacts.  You can specify how many <b>failures</b> you may be willing to bridge.</dd></dl>
<dl><dt>Optimization</dt>
<dd><b>Approximate optimization</b> is used to initialize the alignment for block matching.  The iterative optimizer stops latest after <b>maximal iterations</b> but not earlier than <b>maximal plateauwidth</b> even if the square transfer error is not further decreasing, the same parameters need to be specified for <b>spring mesh</b> simulation.  Usually, only <i>translation</i> and <i>rigid</i> transformations will result in a reasonable configuration because both <i>similarity</i> and <i>affine</i> transforms have their global optimum at zero-size collapsed state.</dd>
<dd>For <i>series alignment</i>, springs across sections have a spring constant of 1.0/distance with distance being the absolute index difference in the series.  For <i>montage</i>, all springs between overlapping images have a spring constant of 1.0.  The constant of springs in the mesh is specified through <b>stiffness</b>.  Larger values will result in less deformed, less perfectly aligned results.  Very small values may result in unevenly distributed deformation due to premature `relaxation' of the simulation.  You can specify the maximal stretch of springs after which they will disrupt.  We have not experimented with this extensively yet.</dd></dl>
<h2><span class="mw-headline" id="Future_Work">Future Work</span></h2>
<p>We will further investigate in automatic selection of appropriate parameters depending on available meta-information of the data sets (resolution, section thickness, deformation priors,...).  We will investigate appropriate identification strategies for discontinuities in the transformation fields such as section ruptures or folds and cut springs at such places.  We will investigate strategies to solve the alignment through hierarchical stages of accuracy for gaining better local accuracy and improved speed.
</p>
<h2><span class="mw-headline" id="References">References</span></h2>
<ol class="references">
<li id="cite_note-SaalfeldAl2010-1"><span class="mw-cite-backlink"><a href="Elastic_Alignment_and_Montage.html#cite_ref-SaalfeldAl2010_1-0">↑</a></span> <span class="reference-text"><span class="Z3988" title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rft.genre=book&amp;rft.title=As-rigid-as-possible+mosaicking+and+serial+section+registration+of+large+ssTEM+datasets&amp;rft.aulast=S.+Saalfeld%2C+A.+Cardona%2C+V.+Hartenstein%2C+P.+Tomancak&amp;rft.date=2010&amp;rft.volume=26&amp;rft.issue=12&amp;rft.pages=i57%26ndash%3Bi63&amp;rft_id=info:doi/10.1093%2Fbioinformatics%2Fbtq219"><cite id="CITEREFS._Saalfeld.2C_A._Cardona.2C_V._Hartenstein.2C_P._Tomancak2010">S. Saalfeld, A. Cardona, V. Hartenstein, P. Tomancak&#32;(2010),&#32;"<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtq219">As-rigid-as-possible mosaicking and serial section registration of large ssTEM datasets</a>",&#32;<i>Bioinformatics</i>&#32;<b>26</b>&#xa0;(12):  i57&#8211;i63<span class="printonly">, DOI 10.1093/bioinformatics/btq219</span></cite></span>
</span>
</li>
<li id="cite_note-BrownL2002-2"><span class="mw-cite-backlink"><a href="Elastic_Alignment_and_Montage.html#cite_ref-BrownL2002_2-0">↑</a></span> <span class="reference-text"><cite style="font-style:normal">M. Brown and D. Lowe&#32;(2002). "Invariant Features from Interest Point Groups".&#32;<i>British Machine Vision Conference</i>: 656&#8211;665.</cite><span class="Z3988" title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rft.genre=proceeding&amp;rft.btitle=British+Machine+Vision+Conference&amp;rft.atitle=Invariant+Features+from+Interest+Point+Groups&amp;rft.au=M.+Brown+and+D.+Lowe&amp;rft.date=2002&amp;rft.pages=656%26ndash%3B665"><span style="display: none;">&#160;</span></span></span>
</li>
<li id="cite_note-Lowe2004-3"><span class="mw-cite-backlink">↑ <sup><a href="Elastic_Alignment_and_Montage.html#cite_ref-Lowe2004_3-0">3.0</a></sup> <sup><a href="Elastic_Alignment_and_Montage.html#cite_ref-Lowe2004_3-1">3.1</a></sup></span> <span class="reference-text"><span class="Z3988" title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rft.genre=book&amp;rft.title=Distinctive+Image+Features+from+Scale-Invariant+Keypoints&amp;rft.aulast=D.+Lowe&amp;rft.date=2004&amp;rft.volume=60&amp;rft.issue=2&amp;rft.pages=91%26ndash%3B110&amp;rft_id=info:doi/10.1109%2FICCV.1999.790410"><cite id="CITEREFD._Lowe2004">D. Lowe&#32;(2004),&#32;"<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1109%2FICCV.1999.790410">Distinctive Image Features from Scale-Invariant Keypoints</a>",&#32;<i>International Journal of Computer Vision</i>&#32;<b>60</b>&#xa0;(2):  91&#8211;110<span class="printonly">, DOI 10.1109/ICCV.1999.790410</span></cite></span>
</span>
</li>
</ol>

<!-- 
NewPP limit report
Cached time: 20200713065037
Cache expiry: 86400
Dynamic content: false
CPU time usage: 0.096 seconds
Real time usage: 0.101 seconds
Preprocessor visited node count: 1315/1000000
Preprocessor generated node count: 11359/1000000
Post‐expand include size: 8902/2097152 bytes
Template argument size: 3368/2097152 bytes
Highest expansion depth: 12/40
Expensive parser function count: 0/3
-->

<!-- 
Transclusion expansion time report (%,ms,calls,template)
100.00%   52.996      1 - -total
 48.08%   25.480      2 - Template:Cite_journal
 41.78%   22.140      2 - Template:Citation/core
 28.27%   14.980      1 - Template:Infobox
 22.87%   12.122      3 - Template:Person
 10.09%    5.345      1 - Template:Cite_conference
  3.23%    1.714      2 - Template:Link
  0.95%    0.502      2 - Template:Person:Albertcardona
-->
</div><div class="printfooter">
Retrieved from "<a dir="ltr" href="index.php?title=Elastic_Alignment_and_Montage&amp;oldid=41233">http://imagej.net/index.php?title=Elastic_Alignment_and_Montage&amp;oldid=41233</a>"</div>
							</div>

			<div id="footer">
				<p> This page was last modified on 24 January 2020, at 11:54.</p><ul><li><a href="./ImageJ:Privacy_policy" title="ImageJ:Privacy policy">Privacy policy</a></li><li><a href="./ImageJ:About" class="mw-redirect" title="ImageJ:About">About ImageJ</a></li><li><a href="Imprint" title="Imprint">Imprint</a></li></ul>			</div>

			<div id="catlinks" class="catlinks" data-mw="interface"><div id="mw-normal-catlinks" class="mw-normal-catlinks"><a href="./Special:Categories" title="Special:Categories">Categories</a>: <ul><li><a href="./Category:Plugins" title="Category:Plugins">Plugins</a></li><li><a href="./Category:Registration" title="Category:Registration">Registration</a></li><li><a href="./Category:TrakEM2" title="Category:TrakEM2">TrakEM2</a></li><li><a href="./Category:Citable" title="Category:Citable">Citable</a></li></ul></div></div>		</div>

		<div id="bottom-wrap">
			<div id="footer-wrap-inner">
				<div id="ternary" class="footer">
					<ul>
						<li class="widget">
							<img id="logo" src="skins/imagej-128.png" alt="">
						</li>
					</ul>
				</div>
			</div>
		</div>
		<script>(window.RLQ=window.RLQ||[]).push(function(){mw.loader.load(["ext.fancytree","ext.suckerfish","ext.cite.a11y","mediawiki.toc","mediawiki.action.view.postEdit","site","mediawiki.user","mediawiki.hidpi","mediawiki.page.ready","mediawiki.searchSuggest","ext.SimpleTooltip"]);});</script><script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgBackendResponseTime":285});});</script>
		</body>
		</html>
		