<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">
<head>
<meta charset="UTF-8"/>
<title>Stitching 2D/3D - ImageJ</title>
<script>document.documentElement.className = document.documentElement.className.replace( /(^|\s)client-nojs(\s|$)/, "$1client-js$2" );</script>
<script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":0,"wgPageName":"Stitching_2D/3D","wgTitle":"Stitching 2D/3D","wgCurRevisionId":32839,"wgRevisionId":32839,"wgArticleId":151,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Plugins","Registration","Stitching"],"wgBreakFrames":false,"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRelevantPageName":"Stitching_2D/3D","wgRelevantArticleId":151,"wgRequestId":"1214a760771b6fce43b7c63b","wgIsProbablyEditable":false,"wgRestrictionEdit":[],"wgRestrictionMove":[],"wgPreferredVariant":"en","fancytree_path":"/extensions/TreeAndMenu/fancytree"});mw.loader.state({"site.styles":"ready","noscript":"ready","user.styles":"ready","user.cssprefs":"ready","user":"ready","user.options":"loading","user.tokens":"loading","mediawiki.legacy.shared":"ready","mediawiki.legacy.commonPrint":"ready","mediawiki.sectionAnchor":"ready","skins.erudite":"ready"});mw.loader.implement("user.options@0j3lz3q",function($,jQuery,require,module){mw.user.options.set({"variant":"en"});});mw.loader.implement("user.tokens@1ku9xth",function ( $, jQuery, require, module ) {
mw.user.tokens.set({"editToken":"+\\","patrolToken":"+\\","watchToken":"+\\","csrfToken":"+\\"});/*@nomin*/;

});mw.loader.load(["mediawiki.page.startup"]);});</script>
<link rel="stylesheet" href="erudite14.css"/>
<link rel="stylesheet" href="extensions/TreeAndMenu/fancytree/fancytree.css"/><link rel="stylesheet" href="extensions/TreeAndMenu/suckerfish/suckerfish.css"/>
<meta name="ResourceLoaderDynamicStyles" content=""/>
<link rel="stylesheet" href="erudite15.css"/>
<meta name="generator" content="MediaWiki 1.28.0"/>
<meta name="description" content="Please note that the Stitching, as well as other plugins available through Fiji, is based on a publication. If you use it successfully for your research please be so kind to cite our work:"/>
<link rel="shortcut icon" href="skins/ij2.ico"/>
	<meta property="og:type" content="article"/>

	<meta property="og:site_name" content="ImageJ"/>

	<meta property="og:title" content="Stitching 2D/3D"/>

	<meta property="og:description" content="Please note that the Stitching, as well as other plugins available through Fiji, is based on a publication. If you use it successfully for your research please be so kind to cite our work:"/>


<meta name="viewport" content="width=device-width, initial-scale=1" />
</head>
<body class="mediawiki ltr sitedir-ltr mw-hide-empty-elt ns-0 ns-subject page-Stitching_2D_3D rootpage-Stitching_2D_3D skin-erudite action-view">
		<div id="top-wrap" role="banner">
			<h1><a href="Welcome" title="ImageJ" rel="home">ImageJ</a></h1>
			<div id="tagline">From ImageJ</div>

			<a id="menubutton" href="Stitch_Multiple_Series_or_Tile_Scan_File.html#menu">Menu</a>
			<div id="nav" role="navigation">
			<ul id='menu'>
<li id="menu-item-n-About"><a href="ImageJ">About</a></li>
<li id="menu-item-n-Downloads"><a href="Downloads">Downloads</a></li>
<li id="menu-item-n-Learn"><a href="Learn">Learn</a></li>
<li id="menu-item-n-Develop"><a href="Development">Develop</a></li>
<li id="menu-item-n-News"><a href="News">News</a></li>
<li id="menu-item-n-Events"><a href="Events">Events</a></li>
<li id="menu-item-n-Help"><a href="Help">Help</a></li>
</ul>
			</div>
		</div>

		<div id="mw-js-message"></div>
		
		<div id="main" role="main">

			<div id="bodyContent">
				<div style="font-size: large; border: 1px solid black; padding: 1em; margin-bottom: 1em; text-align: center; background-color: #fda;">
					This is an archive of the old MediaWiki-based ImageJ wiki. The current website can be found at <a href="https://imagej.net/">imagej.net</a>.
				</div>

				<h1>Stitching 2D/3D</h1>
				
				<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><div style="overflow:hidden"><table class="metadata plainlinks ambox ambox-content">
<tr>
<td class="ambox-image">
<p><a href="./File:Important-sign.png" class="image"><img alt="Important-sign.png" src="images/f/f0/Important-sign.png" width="40" height="40" /></a>
</p>
</td>
<td> <i>Please note that this version of the Stitching is deprecated. A new version is available here: <a href="Image_Stitching" title="Image Stitching">Image Stitching</a></i> </td>
<p><br />
</p>
</tr>
</table></div>
<p><br />
</p>
<table class="infobox" cellspacing="5" style="max-width: 30em; font-size: 80%; text-align: left; float: right; border: 1px solid #a0a0a0;">
<tr>
<th colspan="2" style="text-align: center; font-size: 130%;"> <div style="float: left"></div>Stitching<div style="clear: left;"></div>
</th></tr>
<tr>
<th> Project
</th>
<td> Fiji
</td></tr>
<tr>
<th> URL
</th>
<td> <a rel="nofollow" class="external free" href="Stitching">https://imagej.net/Stitching</a>
</td></tr>
<tr>
<th> Source
</th>
<td> <a rel="nofollow" class="external text" href="https://github.com/fiji/Stitching/releases/tag/Stitching_-3.1.3">on GitHub</a>
</td></tr>
<tr>
<th> License
</th>
<td> <a href="GPLv2" class="mw-redirect" title="GPLv2">GPLv2</a>
</td></tr>
<tr>
<th> Release
</th>
<td> <a rel="nofollow" class="external text" href="http://maven.imagej.net/index.html#nexus-search;gav~sc.fiji~Stitching_~3.1.3~~~kw,versionexpand">3.1.3</a>
</td></tr>
<tr>
<th> Date
</th>
<td> Tue May 02 10:36:26 CDT 2017
</td></tr>
<tr>
<th> <a href="Development_status" class="mw-redirect" title="Development status">Development status</a>
</th>
<td> Active
</td></tr>
<tr>
<th> <a href="Support_status" class="mw-redirect" title="Support status">Support status</a>
</th>
<td> Active
</td></tr>
<tr>
<th colspan="2" style="background: lightgray; font-variant: small-caps; text-align: center"> <a href="Team" class="mw-redirect" title="Team">Team</a>
</th></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Founders</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>
</td></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Leads</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>
</td></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Developers</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>
</td></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Debuggers</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>, <a href="./User:Rueden" title="User:Rueden">Curtis Rueden</a>
</td></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Reviewers</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>
</td></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Support</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>, <a href="./User:Rueden" title="User:Rueden">Curtis Rueden</a>
</td></tr>
<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Maintainers</a>
</th>
<td> <a href="./User:StephanP" title="User:StephanP">Stephan Preibisch</a>, <a href="./User:Rueden" title="User:Rueden">Curtis Rueden</a>
</td></tr>

<tr>
<th> <a href="Team" class="mw-redirect" title="Team">Contributors</a>
</th>
<td> <a href="./User:Schindelin" title="User:Schindelin">Johannes Schindelin</a>, <a href="./User:Iarganda" title="User:Iarganda">Ignacio Arganda-Carreras</a>, <a href="./User:Ehrenfeu" title="User:Ehrenfeu">Niko Ehrenfeuchter</a>, <a href="./User:Hinerm" title="User:Hinerm">Mark Hiner</a>, <a href="./User:Ksmader" title="User:Ksmader">Kevin Mader</a>, <a href="./User:Saalfeld" class="mw-redirect" title="User:Saalfeld">Stephan Saalfeld</a>, John Lapage
</td></tr>
</table><div class="project-info">
<span class="project-logo"><a href="Fiji" title="Fiji"><img class="simple-tooltip simple-tooltip-img" data-simple-tooltip="This page describes content which is part of the Fiji distribution of ImageJ. Click the logo for details." src="images/a/ae/Fiji-icon.png"></img></a></span></div>
<div id="toc" class="toc"><div id="toctitle"><h2>Contents</h2></div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Citation"><span class="tocnumber">1</span> <span class="toctext">Citation</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Introduction"><span class="tocnumber">2</span> <span class="toctext">Introduction</span></a></li>
<li class="toclevel-1 tocsection-3"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Overview_of_the_Stitching_Plugins"><span class="tocnumber">3</span> <span class="toctext">Overview of the Stitching Plugins</span></a></li>
<li class="toclevel-1 tocsection-4"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#2D_Stitching_and_3D_Stitching"><span class="tocnumber">4</span> <span class="toctext">2D Stitching and 3D Stitching</span></a></li>
<li class="toclevel-1 tocsection-5"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Grid_of_Images"><span class="tocnumber">5</span> <span class="toctext">Stitch Grid of Images</span></a></li>
<li class="toclevel-1 tocsection-6"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Image_Grid_Sequence"><span class="tocnumber">6</span> <span class="toctext">Stitch Image Grid Sequence</span></a></li>
<li class="toclevel-1 tocsection-7"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Collection_of_Images"><span class="tocnumber">7</span> <span class="toctext">Stitch Collection of Images</span></a></li>
<li class="toclevel-1 tocsection-8"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Directory_with_Images_.28unknown_configuration.29"><span class="tocnumber">8</span> <span class="toctext">Stitch Directory with Images (unknown configuration)</span></a></li>
<li class="toclevel-1 tocsection-9"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Multiple_Series_or_Tile_Scan_File"><span class="tocnumber">9</span> <span class="toctext">Stitch Multiple Series or Tile Scan File</span></a></li>
<li class="toclevel-1 tocsection-10"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Problems_and_Known_Issues"><span class="tocnumber">10</span> <span class="toctext">Problems and Known Issues</span></a></li>
<li class="toclevel-1 tocsection-11"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Results_.26_Computation_time"><span class="tocnumber">11</span> <span class="toctext">Results &amp; Computation time</span></a></li>
<li class="toclevel-1 tocsection-12"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Contact"><span class="tocnumber">12</span> <span class="toctext">Contact</span></a></li>
<li class="toclevel-1 tocsection-13"><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#See_Also"><span class="tocnumber">13</span> <span class="toctext">See Also</span></a></li>
</ul>
</div>

<h2><span class="mw-headline" id="Citation">Citation</span></h2>
<p>Please note that the Stitching, as well as other plugins available through Fiji, is based on a publication. If you use it successfully for your research please be so kind to cite our work:
</p>
<ul><li> S. Preibisch, S. Saalfeld, P. Tomancak (2009) "Globally optimal stitching of tiled 3D microscopic image acquisitions", <i>Bioinformatics</i>, <b>25</b>(11):1463-1465. <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btp184">Webpage</a> <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/reprint/25/11/1463.pdf">PDF</a></li></ul>
<h2><span class="mw-headline" id="Introduction">Introduction</span></h2>
<p>There is an increasing demand to image large biological specimen at high resolution. Typically those specimen do not fit in the field of view of the microscope. To overcome this drawback, motorized stages moving the sample are used to create a tiled scan of the whole specimen. The physical coordinates provided by the microscope stage are not precise enough to allow reconstruction (”Stitching”) of the whole image from individual image stacks.
</p><p>The 2D/3D Stitching Plugin is able to reconstruct big images/stacks from an arbitrary number of tiled input images/stacks, making use of the Fourier Shift Theorem that computes all possible translations (x, y[, z]) between two 2D/3D images at once, yielding the best overlap in terms of the cross correlation measure. If more than two input images/stacks are used the correct placement of all tiles is determined using a global optimization. To remove brightness differences at the tile borders, non-linear intensity blending is applied.
</p>
<h2><span class="mw-headline" id="Overview_of_the_Stitching_Plugins">Overview of the Stitching Plugins</span></h2>
<p>The Stitching package comes with 4 different plugins:
</p>
<ul><li> <b><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#2D_Stitching_and_3D_Stitching" title="Stitching 2D/3D"> 2D Stitching</a></b>: Stitch two 2-dimensional images, rectangular ROIs can be used to limit the area to search in.</li>
<li> <b><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#2D_Stitching_and_3D_Stitching" title="Stitching 2D/3D"> 3D Stitching</a></b>: Stitch two image stacks (3D), rectangular ROIs can be used to limit the area to search in as well.</li>
<li> <b><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Grid_of_Images" title="Stitching 2D/3D"> Stitch Grid of Images</a></b>: Stitch an arbitrary number of 2D or 3D images which are arranged in a regular grid (e.g. 4 x 3 images with roughly 10% overlap)</li>
<li><b><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Image_Grid_Sequence" title="Stitching 2D/3D"> Stitch Image Grid Sequence</a></b>: Stitch a sequence of 2D or 3D regular grids of images.</li>
<li> <b><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Collection_of_Images" title="Stitching 2D/3D"> Stitch Collection of Images</a></b>: Stitch an arbitrary number of 2D or 3D images which do not have to be in a regular grid. The rough initial configuration (which image is overlapping another one) is given in tile configuration file.</li>
<li> <b><a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Directory_with_Images_.28unknown_configuration.29" title="Stitching 2D/3D"> Stitch Directory with Images (unknown configuration)</a></b>: Stitch all images that are located in a given directory without any knowledge of the tile configuration.</li></ul>
<h2><span class="mw-headline" id="2D_Stitching_and_3D_Stitching">2D Stitching and 3D Stitching</span></h2>
<div class="thumb tright"><div class="thumbinner" style="width:232px;"><a href="./File:3dstitching.png" class="image"><img alt="" src="images/4/48/3dstitching.png" width="230" height="444" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:3dstitching.png" class="internal" title="Enlarge"></a></div>Shows the 3D Stitching Dialog.</div></div></div><br />
<p>The plugins 2D/3D Stitching take two 2D/3D images as input. They can contain rectangular ROIs which limit the search to those areas, however, the full images will be stitched together. If the input images are RGB images the user can select which channels will be used to determine the correct overlap. Note: If the stitching was not successful using ROIs on the obviously overlapping area might solve the problem.
</p><p>Windowing should be always used, except the user knows well about properties of the Fourier Transform and its drawbacks.
</p><p>The number of peaks defines the number of maxima in the Phase Correlation Matrix which are examined. If the stitching was not correct increasing this number might solve it.
</p><p>There are 5 fusion methods available right now.
</p>
<ul><li> <b>Average</b>: In the overlapping area the average intensity between both images is computed.</li>
<li> <b>Linear blending</b>: In the overlapping area the intensity are smoothly adjusted between the two images. The fusion alpha defines the blending function ( <b>alpha = 0</b> blending by averaging, <b>alpha = 1</b> linear blending, <b>alpha &gt; 1</b> non-linear blending ). </li>
<li> <b>Max. Intensity</b>: In the overlapping area the maximum intensity between both images is used int the output image.</li>
<li> <b>Min. Intensity</b>: In the overlapping area the minimum intensity between both images is used int the output image.</li>
<li> <b>Red/Cyan Overlay</b>: In the output image the first image is colored in red, the second one colored in cyan to visualize the registration. In the overlapping area the intensities should appear in gray scales (R + GB). This fusion method does not work with RGB input images for obvious reasons.</li></ul>
<h2><span class="mw-headline" id="Stitch_Grid_of_Images">Stitch Grid of Images</span></h2>
<div class="thumb tright"><div class="thumbinner" style="width:422px;"><a href="./File:StitchGrid.png" class="image"><img alt="" src="images/d/d8/StitchGrid.png" width="420" height="466" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:StitchGrid.png" class="internal" title="Enlarge"></a></div>Shows the Stitch Grid Dialog.</div></div></div><br />
<p>This plugin is able to stitch a grid of images, it does not matter if it is 2D or 3D images as long as all images are of the same type. In contrast to the 2D/3D stitching of two images, this plugins will load the images from harddisc.
</p><p>The user has to define the grid size, that means how the input tiles are arranged (e.g. 7 x 7 image tiles). The overlap is a rough estimate. Note: Smaller overlap reduces computation time, but if the correct alignment is not found try increasing this value first.
</p><p>The next entries in the dialog are used to tell the plugin how to load the images which are in the grid. The directory should point to the directory where the images are stored. In the file names the user has to tell the program how the files are named (e.g. 49 different file names). There are typically two ways how the files are named:
</p>
<ul><li> <b>Images have an increasing number</b>: Tile_001.lsm Tile_002.lsm .... Tile_049.lsm</li>
<li> <b>Images have the x and y coordinate in their filenames</b>: Tile_01_01.tif Tile_01_02.tif .... Tile_07_07.tif</li></ul>
<p>These two cases would be described as follows: 
</p>
<ul><li> <b>Images have an increasing number</b>: Tile_{iii}.lsm - The index number has always three digits and is just counted up starting at 1. The plugin knows that it has to count till 49 as there are 7 x 7 tiles.</li>
<li> <b>Images have the x and y coordinate in their filenames</b>: Tile_{xx}_{yy}.tif - The index numbers for x and y coordinate of the tile have always two digits and are counted up starting at one each. The plugin knows that it has to count to 7 each as there are 7 x 7 tiles.</li></ul>
<p>The starting numbers for x, y and i can be changed in the boxes below. The plugin will write a tileconfiguration.txt into the directory where the files are stored which can be investigated and changed if wanted. The plugin Stitch Collection of Images will take such a file as input. If you hit the checkbox "Save only tile configuration" the plugin will just create this file and stop.
</p><p>The RGB order can be used to define how the different channels (if there are more than one) are read into the channel Red Green and Blue.
</p><p>The channels for registration as well as the fusion methods work the same as described above in the 2D and 3D Stitching Plugin section. Note: Maximum intensity fusion is the most memory efficient stitching as it does not need to hold all images in memory at once!
</p><p>The last three entries describe the behaviour of the global optimization:
</p>
<ul><li> <b>regression threshold</b>: If the regression threshold between two images after the individual stitching are below that number they are assumed to be non-overlapping. Typical values for a good registration are over 0.7, however in noisy images with less content also small regression thresholds can be a correct overlap.</li>
<li> <b>max/avg displacement threshold</b>: After the overlap between all individual tiles has been computed the global optimization computes an optimal arrangement of all tiles. After that, some image pairs will be placed different to each other compared to the individual alignment, which we call displacement. If there are no major alignment errors the average and maximal displacement will be below or around 1 pixel. If one of the individual alignment between two images was wrong, this pair will be displaced a lot in the global optimization as all other connecting tiles pull it towards the correct global position. If the maximal displacement is much higher than the average one it means that this individual alignment is most likely wrong and will be removed. Note: An image will just be removed if there is no link to another image any more.</li>
<li> <b>absolute displacement threshold</b>: Removes links between images if the absolute displacement is higher than this value.</li></ul>
<p>If you check the Create only Preview checkbox the program will not register the input data but only show a the principal layout of the files to check if your input was correct.
</p>
<h2><span class="mw-headline" id="Stitch_Image_Grid_Sequence">Stitch Image Grid Sequence</span></h2>
<p>This plugin complements the previous one by stitching a sequence of grid of image that must have the same X-Y grid configuration. Apart from the previous parameters, the user must now select the number of sections (<i>grid size z</i>), the file name template (for example "Tile_Z{zzz}_Y{yyy}_X{xxx}.lsm"), the starting z value and the <i>output directory</i> where all resulting stitched images will be stored.
</p>
<h2><span class="mw-headline" id="Stitch_Collection_of_Images">Stitch Collection of Images</span></h2>
<div class="thumb tright"><div class="thumbinner" style="width:352px;"><a href="./File:StitchCollection.png" class="image"><img alt="" src="images/a/af/StitchCollection.png" width="350" height="235" class="thumbimage" /></a>  <div class="thumbcaption"><div class="magnify"><a href="./File:StitchCollection.png" class="internal" title="Enlarge"></a></div>Shows the Stitch Image Collection Dialog.</div></div></div><br />
<p>This plugin takes an arbitrary tile configuration as input which has to be stored in a file. 
</p><p>The plugin Stitch Grid of Images will for example create such a file. 
</p><p>All the other parameters are the same as explained in the section <a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Grid_of_Images" title="Stitching 2D/3D"> Stitch Grid of Images</a>.
</p>
<h2><span class="mw-headline" id="Stitch_Directory_with_Images_.28unknown_configuration.29">Stitch Directory with Images (unknown configuration)</span></h2>
<p>This plugin takes all images that are located in a given directory and tries to align them without any knowledge of the tile configuration. All other parameters are identical to those explained in the section <a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Grid_of_Images" title="Stitching 2D/3D"> Stitch Grid of Images</a>. If the plugin does not find the correct alignment, you can try to increase the cross correlation threshold so that only pairwise alignments with very high quality are used for the global optimization. This increases the chance that the optimizer is able to detect the correct pairwise alignments. This, however, requires images with high quality as otherwise correct pairwise alignments with relatively low correlation are excluded.
</p>
<h2><span class="mw-headline" id="Stitch_Multiple_Series_or_Tile_Scan_File">Stitch Multiple Series or Tile Scan File</span></h2>
<p>This plugin takes series of images contained in a single file, e.g. in the .lsm file format. It relies on <a href="Bio-Formats" title="Bio-Formats">Bio-Formats</a>'s series support. The parameters are as explained in the section <a href="Stitch_Multiple_Series_or_Tile_Scan_File.html#Stitch_Grid_of_Images" title="Stitching 2D/3D"> Stitch Grid of Images</a>.
</p>
<h2><span class="mw-headline" id="Problems_and_Known_Issues">Problems and Known Issues</span></h2>
<p>If the Stitch Grid, Stitch Collection or Stich with Unknown Configuration plugins are not able to create the correct output image you can do it yourself iteratively using the 2D/3D Stitching plugins. You just start with the first two images and fuse them. Using ROIs on clearly similar areas you can force a correct alignment. Afterwards you fuse the result with the third image and so on. The biggest drawback apart from the time consumption is that you can only use Maximum Intensity as fusion method, otherwise the image will look weird. Furthermore it will consume more memory.
</p><p>For loading microscopic images we use the LOCI Bioformats importer. If you experience problems loading files, convert them to TIFF before, it should be read fine.
</p><p>3D Stitching will not work for z-Stack size of smaller than 3 pixels. If you want to reconstruct such image from very thin tiles, duplicating of some of the stacks should solve the problem.
</p>
<h2><span class="mw-headline" id="Results_.26_Computation_time">Results &amp; Computation time</span></h2>
<p><a href="./File:ResultsStitching.jpg" class="image" title="Example of Stitching of 3D confocal stacks."><img alt="Example of Stitching of 3D confocal stacks." src="images/7/79/ResultsStitching.jpg" width="780" height="411" /></a><br />
</p><p>The figure shows stitched images of 3D confocal tiles. (A) shows a Drosophila melanogaster pupae expressing a GFP reporter under the regulation of the yellow gene, imaged few hours before eclosion using a 4× dry lens on an Optiphot confocal microscope (Nikon). It was stitched from three image stacks arranged in a 1 × 3 grid (Table 1 first row). The maximum intensity projection is shown. (B) shows the Drosophila larval nervous system stained with three dyes, stitched from a grid of 2 × 3 RGB images (see table 1 second row), the maximum intensity projection is shown. (C) shows a zone in the dorsal telencephalon of human embryonic tissue from week 17 post conception, incubated for 24 hours at 37°C in DiI. It was imaged using a 63×/1.4 objective on Zeiss LSM 510 equipped with a motorized stage. The final image was created from 24 image stacks arranged in a 4 × 6 grid (see table 1 third row), slice 18 is shown. Special thanks to Nicolas Gompel, James W. Truman, Simone Fietz and Wieland B. Huttner for providing the images.
</p><p><a rel="nofollow" class="external text" href="http://fly.mpi-cbg.de/~preibisch/stitching.html">For interactive examples of these datasets have a look here.</a>
</p>
<table cellpadding="4" cellspacing="0" border="1">

<tr>
<th>tiles
</th>
<th>individual tile dimension
</th>
<th>output image dimension
</th>
<th>output image size
</th>
<th>computation time
</th>
<th>min/avg/max displacement
</th></tr>
<tr>
<td>3
</td>
<td>1024&#215;1024&#215;42
</td>
<td>1097&#215;2345&#215;43
</td>
<td>108 MB (8 Bit)
</td>
<td>0:42 min
</td>
<td>0.00/0.00/0.00 px
</td></tr>
<tr>
<td>6
</td>
<td>512&#215;512&#215;86
</td>
<td>975&#215;1425&#215;86
</td>
<td>350 MB (RGB)
</td>
<td>1:20 min
</td>
<td>0.60/0.77/1.05 px
</td></tr>
<tr>
<td>24
</td>
<td>1024&#215;1024&#215;68
</td>
<td>3570&#215;5211&#215;70
</td>
<td>1200 MB (8 Bit)
</td>
<td>22:43 min
</td>
<td>0.49/0.76/0.99 px
</td></tr>
<tr>
<td>72
</td>
<td>512&#215;512&#215;122
</td>
<td>3391&#215;3847&#215;145
</td>
<td>1850 MB (8 Bit)
</td>
<td>43:10 min
</td>
<td>0.00/0.39/0.64 px
</td></tr>
<tr>
<td>63
</td>
<td>1024&#215;1024&#215;92
</td>
<td>6088&#215;7667&#215;119
</td>
<td>5424 MB (8 Bit)
</td>
<td>178:57 min
</td>
<td>0.00/0.66/1.18 px
</td></tr></table>
<p>The Table shows examples of stitched data computed on an Intel® Quad-Core CPU machine with 2.67GHz and 24GB of RAM. The global alignments of all stitchings have an average error below 1 px, the displacements in row 1 are zero because the two alignments are independent of each other. Note that the computation time scales roughly linearly with the output image size.
</p>
<h2><span class="mw-headline" id="Contact">Contact</span></h2>
<p>For any type of comment, questions or input please write to preibisch@mpi-cbg.de or visit my <a rel="nofollow" class="external text" href="http://fly.mpi-cbg.de/~preibisch">homepage</a>.
</p><p><br />
</p>
<h2><span class="mw-headline" id="See_Also">See Also</span></h2>
<ul><li> <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btp184">The Publication on the Stitching Plugin</a>, S. Preibisch, S. Saalfeld, P. Tomancak (2009) Globally optimal stitching of tiled 3D microscopic image acquisitions", <i>Bioinformatics</i>, <b>25</b>(11):1463-1465. <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/reprint/25/11/1463.pdf">PDF</a></li>
<li> <a href="TrakEM2" title="TrakEM2">TrakEM2</a> for non-destructive stitching with floating, adjustable images.</li>
<li> <a rel="nofollow" class="external text" href="http://www.xuvtools.org/doku.php">XuvTools</a> similar stitching software from the University of Freiburg and the <a rel="nofollow" class="external text" href="http://dx.doi.org/10.1111/j.1365-2818.2008.03094.x">abstract</a> of the accompying publication</li></ul>

<!-- 
NewPP limit report
Cached time: 20200713073516
Cache expiry: 86400
Dynamic content: false
CPU time usage: 0.164 seconds
Real time usage: 0.165 seconds
Preprocessor visited node count: 3616/1000000
Preprocessor generated node count: 4985/1000000
Post‐expand include size: 6709/2097152 bytes
Template argument size: 1984/2097152 bytes
Highest expansion depth: 12/40
Expensive parser function count: 0/3
-->

<!-- 
Transclusion expansion time report (%,ms,calls,template)
100.00%  127.838      1 - -total
 95.21%  121.715      1 - Template:ComponentStats:sc.fiji:Stitching
 93.91%  120.052      1 - Template:Component
 78.55%  100.421     16 - Template:Person
  4.19%    5.360      1 - Template:Warning
  3.11%    3.970      1 - Template:Ambox
  1.97%    2.515      1 - Template:GitHub
  1.52%    1.940      1 - Template:Maven
  1.34%    1.714      1 - Template:DevStatus
  1.17%    1.499      1 - Template:SupportStatus
-->
</div><div class="printfooter">
Retrieved from "<a dir="ltr" href="index.php?title=Stitching_2D/3D&amp;oldid=32839">http://imagej.net/index.php?title=Stitching_2D/3D&amp;oldid=32839</a>"</div>
							</div>

			<div id="footer">
				<p> This page was last modified on 3 June 2017, at 23:04.</p><ul><li><a href="./ImageJ:Privacy_policy" title="ImageJ:Privacy policy">Privacy policy</a></li><li><a href="./ImageJ:About" class="mw-redirect" title="ImageJ:About">About ImageJ</a></li><li><a href="Imprint" title="Imprint">Imprint</a></li></ul>			</div>

			<div id="catlinks" class="catlinks" data-mw="interface"><div id="mw-normal-catlinks" class="mw-normal-catlinks"><a href="./Special:Categories" title="Special:Categories">Categories</a>: <ul><li><a href="./Category:Plugins" title="Category:Plugins">Plugins</a></li><li><a href="./Category:Registration" title="Category:Registration">Registration</a></li><li><a href="./Category:Stitching" title="Category:Stitching">Stitching</a></li></ul></div></div>		</div>

		<div id="bottom-wrap">
			<div id="footer-wrap-inner">
				<div id="ternary" class="footer">
					<ul>
						<li class="widget">
							<img id="logo" src="skins/imagej-128.png" alt="">
						</li>
					</ul>
				</div>
			</div>
		</div>
		<script>(window.RLQ=window.RLQ||[]).push(function(){mw.loader.load(["ext.fancytree","ext.suckerfish","mediawiki.toc","mediawiki.action.view.postEdit","site","mediawiki.user","mediawiki.hidpi","mediawiki.page.ready","mediawiki.searchSuggest","ext.SimpleTooltip"]);});</script><script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgBackendResponseTime":337});});</script>
		</body>
		</html>
		